<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Module 5 Lab: GeneMANIA (web version) | Pathway and Network Analysis of -Omics Data ( June 2023 )</title>
  <meta name="description" content="Course covers the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information." />
  <meta name="generator" content="bookdown 0.33 and GitBook 2.6.7" />

  <meta property="og:title" content="Module 5 Lab: GeneMANIA (web version) | Pathway and Network Analysis of -Omics Data ( June 2023 )" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Course covers the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Module 5 Lab: GeneMANIA (web version) | Pathway and Network Analysis of -Omics Data ( June 2023 )" />
  
  <meta name="twitter:description" content="Course covers the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information." />
  

<meta name="author" content="Gary Bader, Ruth Isserlin, Chaitra Sarathy, Veronique Voisin" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/favicon.ico" type="image/x-icon" />
<link rel="prev" href="genemania_cytoscape.html"/>
<link rel="next" href="module-6-review-of-the-tools.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://bioinformatics.ca/workshops/current-workshops/"><img src="./images/bioinfologo.png" width="280"></a><a href="./">CBW Pathways Workshop</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Canadian Bioinformatics Workshops</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html"><i class="fa fa-check"></i>Welcome <a id="welcome"></a></a>
<ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#meet-your-faculty"><i class="fa fa-check"></i>Meet your Faculty</a>
<ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#gary-bader"><i class="fa fa-check"></i>Gary Bader</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#lincoln-stein"><i class="fa fa-check"></i>Lincoln Stein</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#veronique-voisin"><i class="fa fa-check"></i>Veronique Voisin</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#ruth-isserlin"><i class="fa fa-check"></i>Ruth Isserlin</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#chaitra-sarathy-phd"><i class="fa fa-check"></i>Chaitra Sarathy, PhD</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#nia-hughes"><i class="fa fa-check"></i>Nia Hughes</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#class-materials"><i class="fa fa-check"></i>Class Materials</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#schedule"><i class="fa fa-check"></i>Workshop Schedule</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#pre-workshop"><i class="fa fa-check"></i>Pre-Workshop Materials and Laptop Setup Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#laptop-setup-instructions"><i class="fa fa-check"></i>Laptop Setup Instructions</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#basic-programs"><i class="fa fa-check"></i>Basic programs</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#cytoscape-installation"><i class="fa fa-check"></i>Cytoscape Installation</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#gsea-installation"><i class="fa fa-check"></i>GSEA Installation</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#pre-workshop-tutorials"><i class="fa fa-check"></i>Pre-workshop Tutorials</a>
<ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#cytoscape-preparation-tutorials"><i class="fa fa-check"></i>Cytoscape Preparation tutorials</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#pre-workshop-readings-and-lectures"><i class="fa fa-check"></i>Pre-workshop Readings and Lectures</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#additional-tutorials"><i class="fa fa-check"></i>Additional tutorials <a id="additionaltutorials"></a></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i>Module 1 - Introduction to Pathway and Network Analysis (Gary Bader)</a></li>
<li class="chapter" data-level="" data-path="module-2-finding-over-represented-pathways-veronique-voisin.html"><a href="module-2-finding-over-represented-pathways-veronique-voisin.html"><i class="fa fa-check"></i>Module 2: Finding Over-represented Pathways (Veronique Voisin)</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html"><i class="fa fa-check"></i>Module 2 lab - g:Profiler</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#introduction"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#goal-of-the-exercise-1"><i class="fa fa-check"></i>Goal of the exercise 1</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#data"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#exercise-1"><i class="fa fa-check"></i>Exercise 1 - run g:Profiler</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-1---launch-gprofiler."><i class="fa fa-check"></i>Step 1 - Launch g:Profiler.</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-2---input-query"><i class="fa fa-check"></i>Step 2 - input query</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-3---adjust-parameters."><i class="fa fa-check"></i>Step 3 - Adjust parameters.</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-4---run-query"><i class="fa fa-check"></i>Step 4 - Run query</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-5---explore-the-results."><i class="fa fa-check"></i>Step 5 - Explore the results.</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-6-expand-the-stats-tab"><i class="fa fa-check"></i>Step 6: Expand the stats tab</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-7-save-the-results"><i class="fa fa-check"></i>Step 7: Save the results</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-8-optional-but-recommended"><i class="fa fa-check"></i>Step 8 (Optional but recommended)</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-9-optional-by-recommended"><i class="fa fa-check"></i>Step 9 (Optional by recommended)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#exercise-2-load-and-use-a-custom-.gmt-file-and-run-the-query"><i class="fa fa-check"></i>Exercise 2: Load and use a custom .gmt file and run the query</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#optional-steps"><i class="fa fa-check"></i>Optional steps</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#optional-1"><i class="fa fa-check"></i><strong>Optional 1</strong>:</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#option-2"><i class="fa fa-check"></i><strong>Option 2</strong>:</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#option-3"><i class="fa fa-check"></i><strong>Option 3</strong> :</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#bonus---automation."><i class="fa fa-check"></i>Bonus - Automation.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html"><i class="fa fa-check"></i>Module 2 lab - GSEA</a>
<ul>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#introduction-1"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#goal-of-the-exercise"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#data-1"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#background"><i class="fa fa-check"></i>Background</a>
<ul>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#how-to-generate-a-rank-file."><i class="fa fa-check"></i>How to generate a rank file.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#start-the-exercise"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step1."><i class="fa fa-check"></i>Step1.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-2."><i class="fa fa-check"></i>Step 2.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step3."><i class="fa fa-check"></i>Step3.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-4."><i class="fa fa-check"></i>Step 4.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-5."><i class="fa fa-check"></i>Step 5.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#additional_information"><i class="fa fa-check"></i>Additional information</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#bonus---automation.-1"><i class="fa fa-check"></i>Bonus - Automation.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-3-network-visualization-and-analysis-with-cytoscape.html"><a href="module-3-network-visualization-and-analysis-with-cytoscape.html"><i class="fa fa-check"></i>Module 3: Network Visualization and Analysis with Cytoscape</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html"><i class="fa fa-check"></i>Module 3 Lab Primer: Cytoscape Primer</a>
<ul>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#goal-of-the-exercise-2"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#data-2"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#start-the-exercise-1"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-1a---create-network-from-table"><i class="fa fa-check"></i>Exercise 1a - Create Network from table</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-1b---load-node-attributes"><i class="fa fa-check"></i>Exercise 1b - Load node attributes</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-1c---map-node-attributes-to-visual-style"><i class="fa fa-check"></i>Exercise 1c - Map node attributes to Visual Style</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-2---work-with-larger-networks"><i class="fa fa-check"></i>Exercise 2 - Work with larger networks</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-3---perform-basic-enrichment-analysis-using-enrichmenttable"><i class="fa fa-check"></i>Exercise 3 - Perform basic enrichment analysis using EnrichmentTable</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#enrichmenttabl-features"><i class="fa fa-check"></i>Exercise 3B - create Enrichment Map and Enhanced graphics nodes from EnrichmentTable</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-4---load-network-from-ndex"><i class="fa fa-check"></i>Exercise 4 - Load network from NDex</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html"><i class="fa fa-check"></i>Module 3 Lab: g:profiler Visualization</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#goal-of-the-exercise-3"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#data-3"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#enrichmentmap"><i class="fa fa-check"></i>EnrichmentMap</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#description-of-this-exercise"><i class="fa fa-check"></i>Description of this exercise</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#start-the-exercise-2"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#exercise-1a---compare-different-gprofiler-geneset-size-results"><i class="fa fa-check"></i>Exercise 1a - compare different gprofiler geneset size results</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#step-1"><i class="fa fa-check"></i>Step 1</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#step-2"><i class="fa fa-check"></i>Step 2</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#step3-explore-the-results"><i class="fa fa-check"></i>Step3: Explore the results:</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#explore-detailed-results"><i class="fa fa-check"></i>Explore Detailed results</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#exercise-1b---is-specifying-the-gmt-file-important"><i class="fa fa-check"></i>Exercise 1b - Is specifying the gmt file important?</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#exercise-1c---create-em-from-results-using-baderlab-genesets"><i class="fa fa-check"></i>Exercise 1c - create EM from results using Baderlab genesets</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#exercise-1d-optional---investigate-individual-pathways-in-genemania-or-string"><i class="fa fa-check"></i>Exercise 1d (optional) - investigate individual pathways in GeneMANIA or String</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#genemania"><i class="fa fa-check"></i>GeneMANIA</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#string"><i class="fa fa-check"></i>String</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#bonus---automation.-2"><i class="fa fa-check"></i>Bonus - Automation.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html"><i class="fa fa-check"></i>Module 3 Lab: GSEA Visualization</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#goal-of-the-exercise-4"><i class="fa fa-check"></i>Goal of the exercise:</a></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#data-4"><i class="fa fa-check"></i>Data</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#enrichmentmap-1"><i class="fa fa-check"></i>EnrichmentMap</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#exercise-1---gsea-output-and-enrichmentmap"><i class="fa fa-check"></i>Exercise 1 - GSEA output and EnrichmentMap</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-1-1"><i class="fa fa-check"></i>Step 1</a></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-2-1"><i class="fa fa-check"></i>Step 2</a></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-3"><i class="fa fa-check"></i>Step 3</a></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-4"><i class="fa fa-check"></i>Step 4</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#exercise-2---post-analysis-add-drug-target-gene-sets-to-the-network"><i class="fa fa-check"></i>Exercise 2 - Post analysis (add drug target gene-sets to the network)</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-5"><i class="fa fa-check"></i>Step 5</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#exercise-3---autoannotate-the-network"><i class="fa fa-check"></i>Exercise 3 - Autoannotate the Network</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-6"><i class="fa fa-check"></i>Step 6</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#exercise-4-optional---explore-results-in-genemania-or-string"><i class="fa fa-check"></i>Exercise 4 (Optional) - Explore results in GeneMANIA or STRING</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-7"><i class="fa fa-check"></i>Step 7</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#bonus---automation.-3"><i class="fa fa-check"></i>Bonus - Automation.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html"><i class="fa fa-check"></i>Module 3 Lab: (Bonus) Automation</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#goal-of-the-exercise-5"><i class="fa fa-check"></i>Goal of the exercise:</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#set-up---option-1---install-rrstudio"><i class="fa fa-check"></i>Set Up - Option 1 - Install R/Rstudio</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#set-up---option-2---docker-image-with-rrstudio"><i class="fa fa-check"></i>Set Up - Option 2 - Docker image with R/Rstudio</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#what-is-docker"><i class="fa fa-check"></i>What is docker?</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#docker---basic-term-definition"><i class="fa fa-check"></i>Docker - Basic term definition</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#container"><i class="fa fa-check"></i>Container</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#image"><i class="fa fa-check"></i>Image</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#docker-volumes"><i class="fa fa-check"></i>Docker Volumes</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#r_docker"><i class="fa fa-check"></i>Install Docker</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#windows"><i class="fa fa-check"></i>Windows</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#macos-linux"><i class="fa fa-check"></i>MacOS / Linux</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#create-your-first-notebook-using-docker"><i class="fa fa-check"></i>Create your first notebook using Docker</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#start-coding"><i class="fa fa-check"></i>Start coding!</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#start-using-automation"><i class="fa fa-check"></i>Start using automation</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-1---launch-rstudio"><i class="fa fa-check"></i>Step 1 - launch RStudio</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-2---create-a-new-project"><i class="fa fa-check"></i>Step 2 - create a new project</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-3---open-example-up-rnotebook"><i class="fa fa-check"></i>Step 3 - Open example up RNotebook</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-4---define-notebook-parameters"><i class="fa fa-check"></i>Step 4 - Define Notebook parameters</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-5---step-through-notebook-to-run-the-analysis"><i class="fa fa-check"></i>Step 5 - Step through notebook to run the analysis</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#exercises"><i class="fa fa-check"></i>Exercises</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#additional-resources"><i class="fa fa-check"></i>Additional resources</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-4-in-depth-analysis-of-networks-and-pathways.html"><a href="module-4-in-depth-analysis-of-networks-and-pathways.html"><i class="fa fa-check"></i>Module 4: In-depth Analysis of Networks and Pathways</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html"><i class="fa fa-check"></i>Module 4 Lab: ReactomeFI</a>
<ul>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#goal-of-this-practical-lab"><i class="fa fa-check"></i>Goal of this practical lab</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#data-download-the-following-files-on-your-computer-before-starting-the-practical-lab."><i class="fa fa-check"></i>Data: download the following files on your computer before starting the practical lab.</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#exercise-1-use-the-reactome-functional-interaction-fi-network"><i class="fa fa-check"></i>Exercise 1: Use the Reactome Functional Interaction (FI) Network</a>
<ul>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#question-1-describe-the-size-and-composition-of-the-network"><i class="fa fa-check"></i>Question 1: Describe the size and composition of the network?</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#question-2-after-clustering-how-many-modules-are-there"><i class="fa fa-check"></i>Question 2: After clustering, how many modules are there?</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#query-information-about-the-interaction-between-2-genes"><i class="fa fa-check"></i>Query information about the interaction between 2 genes:</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#question-3-what-are-the-most-significant-pathways-in-each-module"><i class="fa fa-check"></i>Question 3: What are the most significant pathways in each module?</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#set-the-size-of-the-nodes-proportional-to-the-mutation-frequencies-in-each-cancer"><i class="fa fa-check"></i>Set the size of the nodes proportional to the mutation frequencies in each cancer</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#play-around-with-the-styles-change-transparency-and-colors"><i class="fa fa-check"></i>Play around with the styles: change transparency and colors</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#create-a-pie-chart"><i class="fa fa-check"></i>Create a pie chart</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#create-a-subnetwork"><i class="fa fa-check"></i>Create a subnetwork</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#fetch-cancer-drugs-on-the-created-subnetwork"><i class="fa fa-check"></i>Fetch Cancer drugs on the created subnetwork</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#save-the-network-as-an-image-for-publication"><i class="fa fa-check"></i>Save the network as an image for publication</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#exercise-2a-explore-reactome-pathways"><i class="fa fa-check"></i>Exercise 2a: Explore Reactome Pathways</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#exercise-2b-pathway-enrichment-analysis-using-a-simple-gene-list"><i class="fa fa-check"></i>Exercise 2b: Pathway enrichment analysis using a simple gene list</a>
<ul>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#question-1-what-are-the-most-significant-biological-pathways-based-on-the-fdr"><i class="fa fa-check"></i>Question 1: What are the most significant biological pathways based on the FDR?</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#answer-to-question-1"><i class="fa fa-check"></i>Answer to Question 1</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#exercise-2c-pathway-based-analysis-using-a-rank-gene-list-gsea"><i class="fa fa-check"></i>Exercise 2c: Pathway-based analysis using a rank gene list (GSEA)</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#automation-for-advanced-users"><i class="fa fa-check"></i>Automation ( for advanced users)</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#reference-guide-bonus-exercises"><i class="fa fa-check"></i>Reference guide /bonus exercises:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-5-gene-function-prediction.html"><a href="module-5-gene-function-prediction.html"><i class="fa fa-check"></i>Module 5: Gene Function Prediction</a></li>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html"><i class="fa fa-check"></i>Module 5 Lab: GeneMANIA (Cytoscape version)</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#goal-of-this-practical-lab-1"><i class="fa fa-check"></i>Goal of this practical lab</a></li>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#exercise-1-questions-and-steps-to-follow"><i class="fa fa-check"></i>EXERCISE 1: QUESTIONS AND STEPS TO FOLLOW</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#steps"><i class="fa fa-check"></i>STEPS</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#exercise-1-answers"><i class="fa fa-check"></i>EXERCISE 1 ANSWERS</a></li>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#exercise-2-questions-and-steps-to-follow"><i class="fa fa-check"></i>EXERCISE 2: QUESTIONS AND STEPS TO FOLLOW</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#steps-1"><i class="fa fa-check"></i>STEPS</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#exercise-3-questions-and-steps-to-follow"><i class="fa fa-check"></i>EXERCISE 3: QUESTIONS AND STEPS TO FOLLOW</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#steps-2"><i class="fa fa-check"></i>STEPS</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#some-definitions"><i class="fa fa-check"></i>SOME DEFINITIONS:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html"><i class="fa fa-check"></i>Module 5 Lab: GeneMANIA (web version)</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#goal-of-this-practical-lab-2"><i class="fa fa-check"></i>Goal of this practical lab</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1-questions-and-steps-to-follow-1"><i class="fa fa-check"></i>EXERCISE 1: QUESTIONS AND STEPS TO FOLLOW</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#steps-3"><i class="fa fa-check"></i>STEPS</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1-answers-detailed-explanation-and-screenshots"><i class="fa fa-check"></i>EXERCISE 1 ANSWERS: DETAILED EXPLANATION AND SCREENSHOTS</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---steps-1-4"><i class="fa fa-check"></i>EXERCISE 1 - STEPS 1-4</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---step-5"><i class="fa fa-check"></i>EXERCISE 1 - STEP 5</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---step-6"><i class="fa fa-check"></i>EXERCISE 1 - STEP 6</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---step-7"><i class="fa fa-check"></i>Exercise 1 - STEP 7</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---step-8"><i class="fa fa-check"></i>Exercise 1 - STEP 8</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---step-9"><i class="fa fa-check"></i>Exercise 1 - STEP 9</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---step-10-layouts"><i class="fa fa-check"></i>Exercise 1 - STEP 10 (layouts)</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---step-11-save-an-image"><i class="fa fa-check"></i>Exercise 1 - STEP 11 (save an image)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2-questions-and-steps-to-follow-1"><i class="fa fa-check"></i>EXERCISE 2: QUESTIONS AND STEPS TO FOLLOW</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#steps-4"><i class="fa fa-check"></i>STEPS</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2-answers-detailed-steps-and-screenshots"><i class="fa fa-check"></i>EXERCISE 2 ANSWERS: DETAILED STEPS AND SCREENSHOTS</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---steps-1-to-4"><i class="fa fa-check"></i>Exercise 2 - STEPS 1 to 4</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-5"><i class="fa fa-check"></i>Exercise 2 - STEP 5</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-6."><i class="fa fa-check"></i>Exercise 2 - STEP 6.</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-7"><i class="fa fa-check"></i>Exercise 2 - STEP 7</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-8"><i class="fa fa-check"></i>Exercise 2 - STEP 8</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-9"><i class="fa fa-check"></i>Exercise 2 - STEP 9</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-10"><i class="fa fa-check"></i>Exercise 2 - STEP 10</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-11"><i class="fa fa-check"></i>Exercise 2 - STEP 11</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-12"><i class="fa fa-check"></i>Exercise 2 - STEP 12</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-13."><i class="fa fa-check"></i>Exercise 2 - STEP 13.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3-questions-and-steps-to-follow-1"><i class="fa fa-check"></i>EXERCISE 3: QUESTIONS AND STEPS TO FOLLOW</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#steps-5"><i class="fa fa-check"></i>STEPS</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3-more-details-and-screenshots"><i class="fa fa-check"></i>Exercise 3: MORE DETAILS AND SCREENSHOTS</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3---steps-1---3"><i class="fa fa-check"></i>Exercise 3 - STEPS 1 - 3</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3---step-4"><i class="fa fa-check"></i>Exercise 3 - STEP 4</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3---step-5"><i class="fa fa-check"></i>Exercise 3 - STEP 5</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3---steps-6"><i class="fa fa-check"></i>Exercise 3 - STEPS 6</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3---step-7"><i class="fa fa-check"></i>Exercise 3 - STEP 7</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3---step-8"><i class="fa fa-check"></i>Exercise 3 - STEP 8</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3---step-9"><i class="fa fa-check"></i>Exercise 3 - STEP 9</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#some-definitions-1"><i class="fa fa-check"></i>SOME DEFINITIONS:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-6-review-of-the-tools.html"><a href="module-6-review-of-the-tools.html"><i class="fa fa-check"></i>Module 6: Review of the tools</a>
<ul>
<li class="chapter" data-level="" data-path="module-6-review-of-the-tools.html"><a href="module-6-review-of-the-tools.html#final-slides"><i class="fa fa-check"></i>Final slides</a></li>
<li class="chapter" data-level="" data-path="module-6-review-of-the-tools.html"><a href="module-6-review-of-the-tools.html#scrna-lab-praticals"><i class="fa fa-check"></i>scRNA lab praticals</a></li>
<li class="chapter" data-level="" data-path="module-6-review-of-the-tools.html"><a href="module-6-review-of-the-tools.html#integrated-assignment"><i class="fa fa-check"></i>Integrated assignment</a></li>
<li class="chapter" data-level="" data-path="module-6-review-of-the-tools.html"><a href="module-6-review-of-the-tools.html#integrated-assignment-bonus"><i class="fa fa-check"></i>Integrated assignment bonus</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="scRNA-lab1.html"><a href="scRNA-lab1.html"><i class="fa fa-check"></i>Module 6 lab 1: scRNA data downstream analysis</a>
<ul>
<li class="chapter" data-level="" data-path="scRNA-lab1.html"><a href="scRNA-lab1.html#introduction-2"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="scRNA-lab1.html"><a href="scRNA-lab1.html#pmbc3k-seurat-pipeline"><i class="fa fa-check"></i>Pmbc3k Seurat Pipeline</a></li>
<li class="chapter" data-level="" data-path="scRNA-lab1.html"><a href="scRNA-lab1.html#load-libraries"><i class="fa fa-check"></i>load libraries</a></li>
<li class="chapter" data-level="" data-path="scRNA-lab1.html"><a href="scRNA-lab1.html#load-the-pbmc-dataset"><i class="fa fa-check"></i>Load the PBMC dataset</a></li>
<li class="chapter" data-level="" data-path="scRNA-lab1.html"><a href="scRNA-lab1.html#process-the-dataset"><i class="fa fa-check"></i>Process the dataset</a></li>
<li class="chapter" data-level="" data-path="scRNA-lab1.html"><a href="scRNA-lab1.html#assign-cell-type-identity-to-clusters"><i class="fa fa-check"></i>Assign cell type identity to clusters</a></li>
<li class="chapter" data-level="" data-path="scRNA-lab1.html"><a href="scRNA-lab1.html#find-differentially-expressed-features-cluster-biomarkers"><i class="fa fa-check"></i>Find differentially expressed features (cluster biomarkers)</a></li>
<li class="chapter" data-level="" data-path="scRNA-lab1.html"><a href="scRNA-lab1.html#create-gene-list-for-each-cluster-to-use-with-gprofiler-do-not-run-code"><i class="fa fa-check"></i>Create Gene list for each cluster to use with g:Profiler (Do not run code)</a></li>
<li class="chapter" data-level="" data-path="scRNA-lab1.html"><a href="scRNA-lab1.html#data-gene-lists-for-each-cluster"><i class="fa fa-check"></i>Data (gene lists for each cluster)</a></li>
<li class="chapter" data-level="" data-path="scRNA-lab1.html"><a href="scRNA-lab1.html#run-pathway-enrichment-analysis-using-gprofiler"><i class="fa fa-check"></i>Run pathway enrichment analysis using g:Profiler</a></li>
<li class="chapter" data-level="" data-path="scRNA-lab1.html"><a href="scRNA-lab1.html#create-an-enrichment-map-in-cytoscape"><i class="fa fa-check"></i>Create an enrichment map in Cytoscape</a>
<ul>
<li class="chapter" data-level="" data-path="scRNA-lab1.html"><a href="scRNA-lab1.html#gsea-from-pseudobulk"><i class="fa fa-check"></i>GSEA from pseudobulk</a></li>
<li class="chapter" data-level="" data-path="scRNA-lab1.html"><a href="scRNA-lab1.html#run-gsea"><i class="fa fa-check"></i>run GSEA:</a></li>
<li class="chapter" data-level="" data-path="scRNA-lab1.html"><a href="scRNA-lab1.html#create-an-enrichmentmap"><i class="fa fa-check"></i>Create an EnrichmentMap:</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><i class="fa fa-check"></i>Module 6 lab 2- scRNA pathway analysis using g:Profiler {#scRNA gprofiler-lab}</a>
<ul>
<li class="chapter" data-level="" data-path="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#introduction-3"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#goal-of-the-exercise-1-1"><i class="fa fa-check"></i>Goal of the exercise 1</a></li>
<li class="chapter" data-level="" data-path="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#data-5"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#can-module8-exercise-1"><i class="fa fa-check"></i>Exercise 1 - run g:Profiler</a>
<ul>
<li class="chapter" data-level="" data-path="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-1---launch-gprofiler.-1"><i class="fa fa-check"></i>Step 1 - Launch g:Profiler.</a></li>
<li class="chapter" data-level="" data-path="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-2---input-query-1"><i class="fa fa-check"></i>Step 2 - input query</a></li>
<li class="chapter" data-level="" data-path="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-3---adjust-parameters.-1"><i class="fa fa-check"></i>Step 3 - Adjust parameters.</a></li>
<li class="chapter" data-level="" data-path="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-4---run-query-1"><i class="fa fa-check"></i>Step 4 - Run query</a></li>
<li class="chapter" data-level="" data-path="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-5---explore-the-results.-1"><i class="fa fa-check"></i>Step 5 - Explore the results.</a></li>
<li class="chapter" data-level="" data-path="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-6-expand-the-stats-tab-1"><i class="fa fa-check"></i>Step 6: Expand the stats tab</a></li>
<li class="chapter" data-level="" data-path="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-7-save-the-results-1"><i class="fa fa-check"></i>Step 7: Save the results</a></li>
<li class="chapter" data-level="" data-path="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-8-optional-but-recommended-1"><i class="fa fa-check"></i>Step 8 (Optional but recommended)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#exercise-2"><i class="fa fa-check"></i>Exercise 2 - Cytoscape/EnrichmentMap</a></li>
<li class="chapter" data-level="" data-path="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#goal-of-the-exercise-6"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#data-6"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#enrichmentmap-2"><i class="fa fa-check"></i>EnrichmentMap</a></li>
<li class="chapter" data-level="" data-path="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#description-of-this-exercise-1"><i class="fa fa-check"></i>Description of this exercise</a></li>
<li class="chapter" data-level="" data-path="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#start-the-exercise-3"><i class="fa fa-check"></i>Start the exercise</a>
<ul>
<li class="chapter" data-level="" data-path="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-1-2"><i class="fa fa-check"></i>Step 1</a></li>
<li class="chapter" data-level="" data-path="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-2-2"><i class="fa fa-check"></i>Step 2</a></li>
<li class="chapter" data-level="" data-path="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-3-explore-detailed-results"><i class="fa fa-check"></i>Step 3: Explore Detailed results</a></li>
<li class="chapter" data-level="" data-path="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-4-optional-autoannotate-the-enrichment-map"><i class="fa fa-check"></i>Step 4 [OPTIONAL]: AutoAnnotate the enrichment map</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#exercise-3"><i class="fa fa-check"></i>Exercise 3 [OPTIONAL]- Cytoscape/EnrichmentMap using multiple datasets</a></li>
<li class="chapter" data-level="" data-path="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#goal-of-the-exercise-7"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#data-7"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#start-the-exercise-4"><i class="fa fa-check"></i>Start the exercise</a>
<ul>
<li class="chapter" data-level="" data-path="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-1-3"><i class="fa fa-check"></i>Step 1</a></li>
<li class="chapter" data-level="" data-path="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-6-lab-2--scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-2-3"><i class="fa fa-check"></i>Step 2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="scNetViz-lab.html"><a href="scNetViz-lab.html"><i class="fa fa-check"></i>Module 6 lab 3: ScNetViz with Single Cell RNASeq data</a>
<ul>
<li class="chapter" data-level="" data-path="scNetViz-lab.html"><a href="scNetViz-lab.html#introduction-4"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="scNetViz-lab.html"><a href="scNetViz-lab.html#goal-of-the-practical-lab"><i class="fa fa-check"></i>Goal of the practical lab</a></li>
<li class="chapter" data-level="" data-path="scNetViz-lab.html"><a href="scNetViz-lab.html#data-8"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="scNetViz-lab.html"><a href="scNetViz-lab.html#steps-6"><i class="fa fa-check"></i>Steps</a></li>
<li class="chapter" data-level="" data-path="scNetViz-lab.html"><a href="scNetViz-lab.html#calculate-differential-expression"><i class="fa fa-check"></i>Calculate differential expression</a></li>
<li class="chapter" data-level="" data-path="scNetViz-lab.html"><a href="scNetViz-lab.html#pathway-analysis-of-clusters"><i class="fa fa-check"></i>Pathway Analysis of clusters</a></li>
<li class="chapter" data-level="" data-path="scNetViz-lab.html"><a href="scNetViz-lab.html#automation-for-advanced-users-1"><i class="fa fa-check"></i>Automation (for advanced users)</a></li>
<li class="chapter" data-level="" data-path="scNetViz-lab.html"><a href="scNetViz-lab.html#scnetviz-references"><i class="fa fa-check"></i>scNetViz References</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html"><i class="fa fa-check"></i>Module 6 Integrated Assignment</a>
<ul>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#goal"><i class="fa fa-check"></i>Goal</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#dataset-1"><i class="fa fa-check"></i>DATASET 1</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#background-1"><i class="fa fa-check"></i>Background</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#data-processing"><i class="fa fa-check"></i>Data processing</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-1-run-gprofiler"><i class="fa fa-check"></i>PART 1: run g:Profiler</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-2-save-as-generic-enrichment-map-output-be"><i class="fa fa-check"></i>PART 2: save as Generic Enrichment Map output (BE)</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-3-save-as-generic-enrichment-map-output-ne"><i class="fa fa-check"></i>PART 3: save as Generic Enrichment Map output (NE)</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-4-create-an-enrichment-map"><i class="fa fa-check"></i>PART 4: create an enrichment map</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#answers-gprofiler"><i class="fa fa-check"></i>Answers g:Profiler</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-5-gsea-run-and-create-an-enrichment-map"><i class="fa fa-check"></i>PART 5: GSEA (run and create an enrichment map)</a>
<ul>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#answers-gsea"><i class="fa fa-check"></i>Answers GSEA</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-6-iregulon"><i class="fa fa-check"></i>PART 6: iRegulon</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#dataset-2"><i class="fa fa-check"></i>DATASET 2</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-1-reactomefi"><i class="fa fa-check"></i>PART 1: ReactomeFI</a>
<ul>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#answers-reactome-fi"><i class="fa fa-check"></i>Answers REACTOME FI</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-2-genemania"><i class="fa fa-check"></i>PART 2: GeneMANIA</a>
<ul>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#answers-genemania"><i class="fa fa-check"></i>Answers GeneMANIA</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="ass_automation.html"><a href="ass_automation.html"><i class="fa fa-check"></i>Module 6 Integrated Assignment Bonus - Automation</a>
<ul>
<li class="chapter" data-level="" data-path="ass_automation.html"><a href="ass_automation.html#goal-of-the-exercise-8"><i class="fa fa-check"></i>Goal of the exercise:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html"><i class="fa fa-check"></i>Optional Module 7: Regulatory Network Analysis</a>
<ul>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html#lecture"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html#practical-lab-1-chip_seq-data---great-and-meme-chip"><i class="fa fa-check"></i>Practical lab 1: chIP_seq data - GREAT and MEME-chIP</a></li>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html#practical-lab-2-gene-list---iregulon-and-enrichrenrichmentmap"><i class="fa fa-check"></i>Practical lab 2: gene list - iREgulon and enrichr/EnrichmentMap</a></li>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html#additional-slides-about-the-tools-segway-and-behst-presented-during-the-lecture"><i class="fa fa-check"></i>Additional slides about the tools Segway and BEHST presented during the lecture</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html"><i class="fa fa-check"></i>Optional Module 7 Lab 1: Gene Regulation and Motif Analysis Practical Lab /chIP-seq</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#goal-of-this-practical-lab-3"><i class="fa fa-check"></i>Goal of this practical lab</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#dataset-used-during-this-practical-lab"><i class="fa fa-check"></i>Dataset used during this practical lab</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-1---run-pathway-analysis-using-great"><i class="fa fa-check"></i>Exercise 1 - Run pathway analysis using GREAT</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#perform-pathway-enrichment"><i class="fa fa-check"></i>Perform pathway enrichment</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#explore-the-results."><i class="fa fa-check"></i>Explore the results.</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#perform-pathway-enrichment---proximal-approach"><i class="fa fa-check"></i>Perform pathway enrichment - Proximal approach</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#explore-the-results.---proximal-analysis"><i class="fa fa-check"></i>Explore the results. - proximal analysis</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-2---build-an-enrichment-map-to-visualize-great-results"><i class="fa fa-check"></i>Exercise 2 - Build an enrichment map to visualize GREAT results</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-3-optional-practice-building-enrichment-maps-and-auto-annotation"><i class="fa fa-check"></i>Exercise 3 (optional): Practice building enrichment maps and auto-annotation</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#optional-exercise-3a-autoannotate-the-enrichment-map"><i class="fa fa-check"></i>Optional exercise 3a: AutoAnnotate the enrichment map:</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#optional-exercise-3b-repeat-the-process-of-building-an-enrichment-map-using-the-proximal-data-proximal_gobp_greatexportall.tsv."><i class="fa fa-check"></i>Optional exercise 3b: Repeat the process of building an enrichment map using the proximal data (Proximal_GOBP_greatExportAll.tsv).</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#optional-exercise-3c-repeat-the-process-by-building-both-the-proximal-and-distal-enrichment-maps-at-the-same-time."><i class="fa fa-check"></i>Optional exercise 3c: Repeat the process by building both the Proximal and Distal enrichment maps at the same time.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-4-add-runx1-targets-and-runx1-ko-genes-on-the-distal-enrichment-map."><i class="fa fa-check"></i>Exercise 4: Add RUNX1 targets and RUNX1 KO genes on the distal enrichment map.</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#step-4a-post-analysis"><i class="fa fa-check"></i>step 4a: post analysis:</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#step-4b-optional-change-the-edge-style-of-the-signature-gene-sets"><i class="fa fa-check"></i>Step 4b Optional: Change the edge style of the signature gene-sets:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-5-learning-how-to-run-meme-chip-from-the-meme-suite-httpsmeme-suite.orgmemetoolsmeme-chip"><i class="fa fa-check"></i>Exercise 5: Learning how to run MEME-chip from the MEME suite (https://meme-suite.org/meme/tools/meme-chip)</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#format-the-data"><i class="fa fa-check"></i>Format the Data</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-5a-download-sequences-from-.bed-coordinates"><i class="fa fa-check"></i>Exercise 5a: Download sequences from .bed coordinates</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-5b-run-meme-chip"><i class="fa fa-check"></i>Exercise 5b: Run MEME-chIP</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-6-optional-get-the-iregulon-runx1-targets-and-find-the-mouse-orthologs-using-gorth-from-gprofiler-to-create-the-gmt-file-used-in-exercise-4."><i class="fa fa-check"></i>Exercise 6 (optional): Get the iRegulon RUNX1 targets and find the mouse orthologs using g:Orth (from g:Profiler) to create the gmt file used in Exercise 4.</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#end-of-lab"><i class="fa fa-check"></i>End of Lab</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html"><i class="fa fa-check"></i>Optional Module 7 Lab 2: Gene Regulation and Motif Analysis Practical Lab / iRegulon</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#iregulon-lab"><i class="fa fa-check"></i>iRegulon lab</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#goal-1"><i class="fa fa-check"></i>Goal</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#exercise-1.-detect-regulons-from-co-expressed-genes"><i class="fa fa-check"></i>Exercise 1. Detect regulons from co-expressed genes</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#skills-learned-in-this-exercise"><i class="fa fa-check"></i>Skills learned in this exercise:</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#steps-7"><i class="fa fa-check"></i>Steps</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#exercise-2.-create-a-metatargetome-using-iregulon-and-merge-2-networks-in-cytoscape."><i class="fa fa-check"></i>Exercise 2. Create a metatargetome using iRegulon and merge 2 networks in Cytoscape.</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#steps-8"><i class="fa fa-check"></i>Steps</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#use-or-precomputed-iregulon-results"><i class="fa fa-check"></i>Use or precomputed iRegulon results:</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#notes-about-iregulon"><i class="fa fa-check"></i>Notes about iRegulon:</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#notes-about-cytoscape"><i class="fa fa-check"></i>Notes about Cytoscape:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#exercise-3.-use-enrichr-with-the-prostate-gene-list."><i class="fa fa-check"></i>Exercise 3. Use Enrichr with the prostate gene list.</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#goal-2"><i class="fa fa-check"></i>Goal</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#steps-9"><i class="fa fa-check"></i>Steps</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#end-of-practical-lab"><i class="fa fa-check"></i>end of practical lab</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a  href=""><img src="./images/sponsors_2023.png" width="200"></a><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Pathway and Network Analysis of -Omics Data ( June 2023 )</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="genemania_web" class="section level1 hasAnchor">
<h1>Module 5 Lab: GeneMANIA (web version)<a href="genemania_web.html#genemania_web" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><strong>This work is licensed under a <a href="http://creativecommons.org/licenses/by-sa/3.0/deed.en_US">Creative Commons Attribution-ShareAlike 3.0 Unported License</a>. This means that you are able to copy, share and modify the work, as long as the result is distributed under the same license.</strong></p>
<p><em><font color="#827e9c">By Veronique Voisin </font></em></p>
<div id="goal-of-this-practical-lab-2" class="section level2 hasAnchor">
<h2>Goal of this practical lab<a href="genemania_web.html#goal-of-this-practical-lab-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Create GeneMANIA networks starting from a single gene to predict its function or starting from a gene list. Explore and understand the main output features of GeneMANIA such as the network composition or the enriched functions.</p>
<p>This practical consists of 3 exercises. You can choose to do these exercises using the questions as your only guide (section QUESTIONS AND STEPS TO FOLLOW) - or see the following pages for the step-by-step checklist to finding the answers (section ANSWERS: DETAILED STEPS AND SCREENSHOTS).</p>
<p>Before starting the exercises,download the files:</p>
<div class="rmd-datadownload">
<p>
Right click on link below and select Save Link As.
</p>
<p>
Place the file in your CBW work directory in the corresponding module
directory.
</p>
</div>
<ul>
<li><p><a href="./Module6/genemania/data/30_prostate_cancer_genes.txt">30_prostate_cancer_genes.txt</a></p></li>
<li><p><a href="./Module6/genemania/data/mixed_gene_list.txt">Mixed_gene_list.txt</a></p></li>
<li><p><a href="./Module6/genemania/data/CYB11B_pearson_correlation_prostate.txt">CYP11B_pearson_correlation_prostate.txt</a></p></li>
</ul>
<p>Optional exercise: try to redo these exercises using the Cytoscape GeneMANIA application.</p>
<div class="rmd-note">
<p>
Network layouts are flexible and can be rearranged. What you see when
you perform these exercises may not be identical to what you see in the
tutorial, or what you have seen other times that you have performed the
exercises. Exact layouts and predictions can also be affected by updates
to the networks database that GeneMANIA uses. However it is expected
that the network weights and predicted genes will be similar to those
shown here.
</p>
</div>
</div>
<div id="exercise-1-questions-and-steps-to-follow-1" class="section level2 hasAnchor">
<h2>EXERCISE 1: QUESTIONS AND STEPS TO FOLLOW<a href="genemania_web.html#exercise-1-questions-and-steps-to-follow-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ol style="list-style-type: decimal">
<li>Imagine that you are interested in exploring the function of the human GRN gene: GRN returned as the strongest hit from your omics experiment but not many information about this gene is available in functional databases. Use GeneMANIA to identify its predicted function as well as potential interaction partners.</li>
</ol>
<p><strong>Skills</strong>: GeneMANIA Single Gene search; Navigating Search Results; Exploring available Genes features; Rerun a new analysis using a single gene or multiple genes query from the network.</p>
<div id="steps-3" class="section level3 hasAnchor">
<h3>STEPS<a href="genemania_web.html#steps-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p>Go to GeneMANIAs homepage at <a href="http://www.genemania.org/" class="uri">http://www.genemania.org/</a></p></li>
<li><p>In the search window, ensure that the model organism is set to <em>Homo sapiens</em> <img src="Module6/genemania/images/Up.png" alt="homo" />.</p></li>
<li><p>Enter the following gene: GRN</p></li>
<li><p>Click on the search icon <img src="Module6/genemania/images/Search.png" alt="search" /> and wait for the results.</p></li>
<li><p>When your search results load, examine the network. Genes you searched with are indicated with stripes, related genes added by GeneMANIA are represented in black, and colored links represent the interactions that connect the nodes (genes). <strong>Tip</strong>: zoom in and zoom out using trackpad or mouse scrolling up and down using the mouse or trackpad or zoom in to the maximum using the specific icon <img src="Module6/genemania/images/arrow.png" alt="arrow" />.</p></li>
<li><p>Clicking on a node gives information about the name of gene, the possibility to add, remove this gene or search with this gene only. Click on the GRN node and explore the displayed information.</p></li>
<li><p>Locate the Functions summary tab (bottom left icon <img src="Module6/genemania/images/circle.png" alt="circle" />). What are the functions significantly associated with this network? GRN is the central node of this network: which function would you predict for GRN? How well did GeneMANIA perform (hints: use GeneCards (<a href="http://www.genecards.org/" class="uri">http://www.genecards.org/</a>) , PubMed (<a href="http://www.ncbi.nlm.nih.gov/pubmed/" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/</a>))?</p></li>
<li><p>Locate the gene with the strongest association with GRN. <strong>Tip</strong>: this gene is the largest node on the network.</p></li>
<li><p>Re-run an analysis by adding SORT1, HSPG2 to the search. Click on SORT1, click on Add <img src="Module6/genemania/images/add.png" alt="add" /> or type the name of the gene in the search box. Do the same for HSPG2. The genes are now added to the search box and a new network is automatically created. Which functions are associated with this new network <img src="Module6/genemania/images/circle.png" alt="circle" />?</p></li>
<li><p>On the left side of the window are located icons that we havent yet explored. The first 3 buttons are activating different network layouts. Try the circular <img src="Module6/genemania/images/circledot.png" alt="circular" />, the aligned <img src="Module6/genemania/images/twodown.png" alt="aligned" />, and the force_directed <img src="Module6/genemania/images/crossing.png" alt="force" /> layouts.</p></li>
<li><p>Choose your favorite layout and save the network as an image using the <em>Network image As Shown</em> option from the <em>save</em> menu <img src="Module6/genemania/images/save.png" alt="save" />.</p></li>
</ol>
</div>
</div>
<div id="exercise-1-answers-detailed-explanation-and-screenshots" class="section level2 hasAnchor">
<h2>EXERCISE 1 ANSWERS: DETAILED EXPLANATION AND SCREENSHOTS<a href="genemania_web.html#exercise-1-answers-detailed-explanation-and-screenshots" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="exercise-1---steps-1-4" class="section level3 hasAnchor">
<h3>EXERCISE 1 - STEPS 1-4<a href="genemania_web.html#exercise-1---steps-1-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure">
<img src="Module6/genemania/images/GM1.png" alt="" />
<p class="caption">GM1</p>
</div>
</div>
<div id="exercise-1---step-5" class="section level3 hasAnchor">
<h3>EXERCISE 1 - STEP 5<a href="genemania_web.html#exercise-1---step-5" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure">
<img src="Module6/genemania/images/GM2.png" alt="" />
<p class="caption">GM2</p>
</div>
</div>
<div id="exercise-1---step-6" class="section level3 hasAnchor">
<h3>EXERCISE 1 - STEP 6<a href="genemania_web.html#exercise-1---step-6" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure">
<img src="Module6/genemania/images/GM3.png" alt="" />
<p class="caption">GM3</p>
</div>
</div>
<div id="exercise-1---step-7" class="section level3 hasAnchor">
<h3>Exercise 1 - STEP 7<a href="genemania_web.html#exercise-1---step-7" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure">
<img src="Module6/genemania/images/GM4.png" alt="" />
<p class="caption">GM4</p>
</div>
<p><strong>Question</strong> What are the functions significantly associated with this network?
<strong>Answer</strong> the list of the functions associated with the network are listed in the above screenshot. lysosomal function, vacuolar lumen, vacuolar part are significant under a FDR threshold less than 0.0001.</p>
<p><strong>Question</strong> GRN is the central node of this network: which function would you predict for GRN?
<strong>Answer</strong> : a function related to lysosome and vacuole</p>
<p><strong>Question</strong> How well did GeneMANIA perform (hints: use GeneCards (<a href="http://www.genecards.org/" class="uri">http://www.genecards.org/</a>) , PubMed (<a href="http://www.ncbi.nlm.nih.gov/pubmed/" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/</a>))?
<strong>Answer</strong>
The top functions predicted by GeneMANIA for GRN were related to lysosome and vacuole. A pubmed search could confirm these results: We experimentally verified that granulin precursor (GRN) gene, whose mutations cause frontotemporal lobar degeneration, is involved in lysosome function. (Transcriptional gene network inference from a massive dataset elucidates transcriptome organization and gene function. Belcastro et al.Nucleic Acids Res. 2011 Nov 1;39(20):8677-88. 2011. <a href="PMID:21785136" class="uri">PMID:21785136</a>)</p>
</div>
<div id="exercise-1---step-8" class="section level3 hasAnchor">
<h3>Exercise 1 - STEP 8<a href="genemania_web.html#exercise-1---step-8" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Question</strong> Locate the gene with the strongest association with GRN (thick edge).
<strong>Answer is SORT1</strong></p>
</div>
<div id="exercise-1---step-9" class="section level3 hasAnchor">
<h3>Exercise 1 - STEP 9<a href="genemania_web.html#exercise-1---step-9" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure">
<img src="Module6/genemania/images/GM5.png" alt="" />
<p class="caption">GM5</p>
</div>
<div class="figure">
<img src="Module6/genemania/images/GM6.png" alt="" />
<p class="caption">GM6</p>
</div>
</div>
<div id="exercise-1---step-10-layouts" class="section level3 hasAnchor">
<h3>Exercise 1 - STEP 10 (layouts)<a href="genemania_web.html#exercise-1---step-10-layouts" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="circular-layout" class="section level4 hasAnchor">
<h4>Circular layout<a href="genemania_web.html#circular-layout" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="figure">
<img src="Module6/genemania/images/GM7.png" alt="" />
<p class="caption">GM7</p>
</div>
</div>
<div id="aligned-layout" class="section level4 hasAnchor">
<h4>Aligned layout<a href="genemania_web.html#aligned-layout" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="figure">
<img src="Module6/genemania/images/GM8.png" alt="" />
<p class="caption">GM8</p>
</div>
</div>
<div id="force-directed-layout" class="section level4 hasAnchor">
<h4>Force directed layout<a href="genemania_web.html#force-directed-layout" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="figure">
<img src="Module6/genemania/images/GM9.png" alt="" />
<p class="caption">GM9</p>
</div>
</div>
</div>
<div id="exercise-1---step-11-save-an-image" class="section level3 hasAnchor">
<h3>Exercise 1 - STEP 11 (save an image)<a href="genemania_web.html#exercise-1---step-11-save-an-image" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure">
<img src="Module6/genemania/images/GM10.png" alt="" />
<p class="caption">GM10</p>
</div>
<p><strong>Notes</strong> about biological interpretation of the results:</p>
<p><strong>A paper describing the interaction between GRN and SORT1 and demonstrates how finding related genes could be relevant for elaborating therapy:</strong></p>
<p>Targeted manipulation of the sortilinprogranulin axis rescues progranulin haploinsufficiency. Lee et al.Hum Mol Genet. 2014 March 15; 23(6): 14671478. Published online 2013 October 26. doi: 10.1093/hmg/ddt534. PMCID:PMC3929086
Progranulin (GRN) mutations causing haploinsufficiency are a major cause of frontotemporal lobar degeneration (FTLD-TDP). Recent discoveries demonstrating sortilin (SORT1) is a neuronal receptor for PGRN endocytosis and a determinant of plasma PGRN levels portend the development of enhancers targeting the SORT1PGRN axis. We demonstrate the preclinical efficacy of several approaches through which impairing PGRNs interaction with SORT1 restores extracellular PGRN levels. </p>
<div class="figure">
<img src="Module6/genemania/images/GM11.png" alt="" />
<p class="caption">GM11</p>
</div>
<hr />
</div>
</div>
<div id="exercise-2-questions-and-steps-to-follow-1" class="section level2 hasAnchor">
<h2>EXERCISE 2: QUESTIONS AND STEPS TO FOLLOW<a href="genemania_web.html#exercise-2-questions-and-steps-to-follow-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To start this exercise, you need to download the <a href="./Module6/genemania/data/0_prostate_cancer_genes.txt">30_prostate_cancer_genes.txt</a> file from the course wiki and save it on your computer.</p>
<p>For this exercise, you are working with a list of 30 prostate cancer genes. This list can be downloaded after the workshop from the cBioPortal website (<a href="http://www.cbioportal.org/" class="uri">http://www.cbioportal.org/</a>) under the download section, user defined list. The cBioPortal for Cancer Genomics stores genomic data from large scale, integrated cancer genomic data sets. During this exercise, you will explore the types of networks that have been used to create the GeneMANIA network from the prostate cancer gene list and you will see how changing input parameters can affect the results. The last step of the exercise consists of uploading a custom network which is a list of genes that are positively correlated with CYP11B1 in mRNA expression data of 94 prostate cancer samples (<a href="http://www.cbioportal.org/" class="uri">http://www.cbioportal.org/</a>) .</p>
<p><strong>Skills</strong>: GeneMANIA search using a gene list; Navigating Search Results; Exploring Networks and advanced options; Uploading a custom network.</p>
<div id="steps-4" class="section level3 hasAnchor">
<h3>STEPS<a href="genemania_web.html#steps-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p>Go to GeneMANIAs homepage at <a href="http://www.genemania.org/" class="uri">http://www.genemania.org/</a></p></li>
<li><p>In the search window, ensure that the model organism is set to <em>Homo sapiens</em> <img src="Module6/genemania/images/Up.png" alt="homo" /> .</p></li>
<li><p>Copy and paste genes in the file <a href="./Module6/genemania/data/30_prostate_cancer_genes.txt">30_prostate_cancer_genes.txt</a>. Make sure that the parameter Max resultant genes is set to 20 by clicking on the 3 menu buttons at the right side of the search box and selecting Customise advanced options.</p></li>
<li><p>Click on the search icon <img src="Module6/genemania/images/Search.png" alt="search" /> and wait for the results.</p></li>
<li><p>When your search results load, examine the network. Genes you searched with are indicated with stripes, related genes added by GeneMANIA are represented in black, and colored links represent the interactions that connect the nodes (genes). Move nodes around by selecting them with a mouse to investigate how they are connected.</p></li>
<li><p>Click any link (edge) connecting two nodes to highlight information about it.</p>
<div class="rmd-note">
<p>
Clicking on an edge between 2 nodes will display information about
all interaction networks that connect these 2 nodes. It indicates the
reference (publication) for these interactions. The colors indicate the
type of interaction (co-expression, shared protein domains,
co-localization, physical interactions and predicted).
</p>
</div></li>
<li><p>Locate and expand the Networks summary tab (on the right <img src="Module6/genemania/images/threelines.png" alt="lines" />) and look at what data has been used to create the network and predictions.</p>
<div class="rmd-note">
<p>
that Co-expression (purple colored lines, weight over 30%) and Shared
protein domains (lightgold colored lines, weight over 30%) influence the
results the most, but Co-localization (blue colored lines), Physical
interactions (salmon colored lines) and Predicted (orange) data are also
included. At the top of the Networks summary tab, try Expand none,
then top and all to get information about the sources of the
different networks.
</p>
</div></li>
<li><p>Highlight all connections corresponding to each network by clicking the name of each network category. Click on Shared protein domains and see which genes are connected by predicted protein protein interaction. You can do the same for Co-localization , Co-expression and Physical interactions. <strong>Tips</strong>: these observations of the number of connections make it easier to understand why co-expression and shared protein domains get the higher percent weight on this network: they are helping to connect more genes than physical interactions and predicted; A higher weight means that this network helped more to find related genes.</p></li>
<li><p>Locate the Functions summary tab (bottom left <img src="Module6/genemania/images/circle.png" alt="circle" />) and look at what functions were significantly enriched in this list of prostate genes.</p></li>
<li><p>Shared protein domains is an important part of the network. What would be the GeneMANIA results if we dont include this source when we run GSEA? Click on Show advanced option <img src="Module6/genemania/images/dotdotdot.png" alt="options" /> which is located at the right of the search box. Uncheck Shared protein domains and click on the search icon <img src="Module6/genemania/images/Search.png" alt="search" />. Explore the results.</p></li>
<li><p>Locate the Functions summary tab (bottom left <img src="Module6/genemania/images/circle.png" alt="circle" />) and look at what functions were significantly enriched with these new settings.</p></li>
<li><p>Upload a custom network to GeneMANIA: Go to the menu option at the right of the search box and go to the make sure you see the Networks tab, locate Uploaded, expand this option using the down arrow and click on Upload a network and browse your computer to locate and select the file <a href="./Module6/genemania/data/CYB11B_pearson_correlation_prostate.txt">CYP11B_pearson_correlation_prostate.txt</a>. Wait about a minute for the network to be uploaded. Click on the search icon to launch the query if necessaryand explore the results and locate the genes linked by the custom network (<strong>tip</strong>: click on Uploaded in the Networks tab).</p></li>
<li><p>Try additional parameters of the Customise advanced options <img src="Module6/genemania/images/dotdotdot.png" alt="options" /> tab and look at how the changes you made influenced the results. For example change Network weighting method or Max resultant genes:.</p></li>
</ol>
</div>
</div>
<div id="exercise-2-answers-detailed-steps-and-screenshots" class="section level2 hasAnchor">
<h2>EXERCISE 2 ANSWERS: DETAILED STEPS AND SCREENSHOTS<a href="genemania_web.html#exercise-2-answers-detailed-steps-and-screenshots" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="exercise-2---steps-1-to-4" class="section level3 hasAnchor">
<h3>Exercise 2 - STEPS 1 to 4<a href="genemania_web.html#exercise-2---steps-1-to-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure">
<img src="Module6/genemania/images/GM12.png" alt="" />
<p class="caption">GM12</p>
</div>
<div class="rmd-tip">
<p>
check that the parameter Max resultant genes is set to 20: <img
src="Module6/genemania/images/GM12_v2.png" alt="GM12" />
</p>
</div>
</div>
<div id="exercise-2---step-5" class="section level3 hasAnchor">
<h3>Exercise 2 - STEP 5<a href="genemania_web.html#exercise-2---step-5" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure">
<img src="Module6/genemania/images/GM13.png" alt="" />
<p class="caption">GM13</p>
</div>
</div>
<div id="exercise-2---step-6." class="section level3 hasAnchor">
<h3>Exercise 2 - STEP 6.<a href="genemania_web.html#exercise-2---step-6." class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure">
<img src="Module6/genemania/images/GM14.png" alt="" />
<p class="caption">GM14</p>
</div>
</div>
<div id="exercise-2---step-7" class="section level3 hasAnchor">
<h3>Exercise 2 - STEP 7<a href="genemania_web.html#exercise-2---step-7" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure">
<img src="Module6/genemania/images/GM15.png" alt="" />
<p class="caption">GM15</p>
</div>
<div class="figure">
<img src="Module6/genemania/images/GM16.png" alt="" />
<p class="caption">GM16</p>
</div>
</div>
<div id="exercise-2---step-8" class="section level3 hasAnchor">
<h3>Exercise 2 - STEP 8<a href="genemania_web.html#exercise-2---step-8" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure">
<img src="Module6/genemania/images/GM17.png" alt="" />
<p class="caption">GM17</p>
</div>
<div class="figure">
<img src="Module6/genemania/images/GM18.png" alt="" />
<p class="caption">GM18</p>
</div>
</div>
<div id="exercise-2---step-9" class="section level3 hasAnchor">
<h3>Exercise 2 - STEP 9<a href="genemania_web.html#exercise-2---step-9" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The top pathways with the strongest enrichments are: cellular hormone metabolic process with 21 genes in the list overlapping with this pathway.
The FDR is equal to 6.75e-36.</p>
<div class="figure">
<img src="Module6/genemania/images/GM19.png" alt="" />
<p class="caption">GM19</p>
</div>
</div>
<div id="exercise-2---step-10" class="section level3 hasAnchor">
<h3>Exercise 2 - STEP 10<a href="genemania_web.html#exercise-2---step-10" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Question</strong> Shared protein domains is an important part of the network. What would be the GeneMANIA results if we dont include this source when we run GSEA? <strong>Answer</strong> If shared protein domain is removed, the relationships between the nodes are from the Co-expression, Co-localization, Predicted and Physical interactions networks.The genes added to the network are different compared to the first network created with Shared protein domain.</p>
<div class="figure">
<img src="Module6/genemania/images/GM20.png" alt="" />
<p class="caption">GM20</p>
</div>
<div class="figure">
<img src="Module6/genemania/images/GM21.png" alt="" />
<p class="caption">GM21</p>
</div>
</div>
<div id="exercise-2---step-11" class="section level3 hasAnchor">
<h3>Exercise 2 - STEP 11<a href="genemania_web.html#exercise-2---step-11" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Question</strong> what functions were significantly enriched with these new settings. <strong>Answer</strong> With the new settings, hormone biosynthetic process is the new top enriched pathway.</p>
<div class="figure">
<img src="Module6/genemania/images/GM22.png" alt="" />
<p class="caption">GM22</p>
</div>
</div>
<div id="exercise-2---step-12" class="section level3 hasAnchor">
<h3>Exercise 2 - STEP 12<a href="genemania_web.html#exercise-2---step-12" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure">
<img src="Module6/genemania/images/GM23.png" alt="" />
<p class="caption">GM23</p>
</div>
<div class="figure">
<img src="Module6/genemania/images/GM24.png" alt="" />
<p class="caption">GM24</p>
</div>
<div class="figure">
<img src="Module6/genemania/images/GM25.png" alt="" />
<p class="caption">GM25</p>
</div>
</div>
<div id="exercise-2---step-13." class="section level3 hasAnchor">
<h3>Exercise 2 - STEP 13.<a href="genemania_web.html#exercise-2---step-13." class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure">
<img src="Module6/genemania/images/GM26.png" alt="" />
<p class="caption">GM26</p>
</div>
<p><strong>How to get the 30 prostate cancer gene list</strong>:</p>
<div class="figure">
<img src="Module6/genemania/images/GM27.png" alt="" />
<p class="caption">GM27</p>
</div>
<hr />
</div>
</div>
<div id="exercise-3-questions-and-steps-to-follow-1" class="section level2 hasAnchor">
<h2>EXERCISE 3: QUESTIONS AND STEPS TO FOLLOW<a href="genemania_web.html#exercise-3-questions-and-steps-to-follow-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To start this exercise, you need to download the <a href="./Module6/genemania/data/mixed_gene_list.txt">Mixed_gene_list.txt</a> file and save it on your computer.</p>
<p>For this exercise, you are working on a gene list created by combining 3 user defined gene lists available from the cBioportal (<a href="http://www.cbioportal.org" class="uri">http://www.cbioportal.org</a>). It contains genes implicated in the DNA damage response, the PI3K-AKT-mTOR signaling pathway and Folate transport. This list is representative of a gene list obtained from transcriptomics data. During this exercise, we will first characterize our gene list based on functions and then we will add potential drug and microRNAs targeting genes in the network, and we will save the report.</p>
<p><strong>Skills</strong>: GeneMANIA search using a gene list; Navigating Search Results; Exploring Functions; Adding attributes; Create a report.</p>
<div id="steps-5" class="section level3 hasAnchor">
<h3>STEPS<a href="genemania_web.html#steps-5" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p>Go to GeneMANIAs homepage at <a href="http://www.genemania.org/" class="uri">http://www.genemania.org/</a>.</p></li>
<li><p>In the search window, ensure that the model organism is set to <em>Homo sapiens</em> <img src="Module6/genemania/images/Up.png" alt="homo" /> .</p></li>
<li><p>Copy and paste genes in the file <a href="./Module6/genemania/data/mixed_gene_list.txt">Mixed_gene_list.txt</a>. Click on the search icon <img src="Module6/genemania/images/Search.png" alt="search" /> and wait for the results.</p></li>
<li><p>Locate the Functions summary tab (bottom left <img src="Module6/genemania/images/circle.png" alt="circle" />) and look at functions returned by GeneMANIA</p></li>
<li><p>In the functions summary tab, check some functions to color genes included in these functions. To follow this tutorial, you can for example color the response to insulin , DNA recombination andvitamin transport functions. <strong>Tip</strong>: You need to scroll down to found DNA recombination and vitamin transport as these pathways have an FDR greater than the one for response to insulin.</p></li>
<li><p>We are going next to add miRs and drug interaction networks. Click on Show advanced option <img src="Module6/genemania/images/dotdotdot.png" alt="options" /> which is located at the right of the search box. In the Networks tab, expand Attributes and check Drug-interactions-2013 and miRNA-target-predictions-2013. Check Physical interactions and Co-expression . Click on Customise advanced options. Set Max resultant genes to 20 and Max resultant attributes to 40. Click on the search icon <img src="Module6/genemania/images/Search.png" alt="search" /> and wait for the results. Explore the network. <strong>Tips</strong>: the Drug-interactions and miRNA-target-predictions nodes are displayed in gray. The nodes connected to a drug are genes that are targeted by the drug and nodes connected to a microRNA (miR) are genes predicted to be targeted by this miR.</p></li>
<li><p>Locate our favorite gene PDPK1 on the network, select it by moving the mouse cursor to its node and wait there for a second. It will highlight this gene and all its connections. <strong>Tip</strong>: make sure that the black box displaying the details for PDPK1 is closed.</p></li>
<li><p>Generate and save a report of your results by locating the save menu <img src="Module6/genemania/images/save.png" alt="save" />, and selecting Report. The PDF report provides a detailed description of your search and results.</p></li>
<li><p>Investigate the history function by clicking on the related icon <img src="Module6/genemania/images/redo.png" alt="redo" /> located at the bottom of the window. A panel pops up showing the past networks generated by GeneMANIA. Clicking on one panel will relaunch the search for this network.</p></li>
</ol>
</div>
</div>
<div id="exercise-3-more-details-and-screenshots" class="section level2 hasAnchor">
<h2>Exercise 3: MORE DETAILS AND SCREENSHOTS<a href="genemania_web.html#exercise-3-more-details-and-screenshots" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="exercise-3---steps-1---3" class="section level3 hasAnchor">
<h3>Exercise 3 - STEPS 1 - 3<a href="genemania_web.html#exercise-3---steps-1---3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure">
<img src="Module6/genemania/images/GM28.png" alt="" />
<p class="caption">GM28</p>
</div>
<div class="figure">
<img src="Module6/genemania/images/GM29.png" alt="" />
<p class="caption">GM29</p>
</div>
</div>
<div id="exercise-3---step-4" class="section level3 hasAnchor">
<h3>Exercise 3 - STEP 4<a href="genemania_web.html#exercise-3---step-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure">
<img src="Module6/genemania/images/GM30.png" alt="" />
<p class="caption">GM30</p>
</div>
</div>
<div id="exercise-3---step-5" class="section level3 hasAnchor">
<h3>Exercise 3 - STEP 5<a href="genemania_web.html#exercise-3---step-5" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure">
<img src="Module6/genemania/images/GM31.png" alt="" />
<p class="caption">GM31</p>
</div>
</div>
<div id="exercise-3---steps-6" class="section level3 hasAnchor">
<h3>Exercise 3 - STEPS 6<a href="genemania_web.html#exercise-3---steps-6" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure">
<img src="Module6/genemania/images/GM32.png" alt="" />
<p class="caption">GM32</p>
</div>
<div class="figure">
<img src="Module6/genemania/images/GM33.png" alt="" />
<p class="caption">GM33</p>
</div>
</div>
<div id="exercise-3---step-7" class="section level3 hasAnchor">
<h3>Exercise 3 - STEP 7<a href="genemania_web.html#exercise-3---step-7" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure">
<img src="Module6/genemania/images/GM34.png" alt="" />
<p class="caption">GM34</p>
</div>
</div>
<div id="exercise-3---step-8" class="section level3 hasAnchor">
<h3>Exercise 3 - STEP 8<a href="genemania_web.html#exercise-3---step-8" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure">
<img src="Module6/genemania/images/GM35.png" alt="" />
<p class="caption">GM35</p>
</div>
<div class="figure">
<img src="Module6/genemania/images/GM36.png" alt="" />
<p class="caption">GM36</p>
</div>
</div>
<div id="exercise-3---step-9" class="section level3 hasAnchor">
<h3>Exercise 3 - STEP 9<a href="genemania_web.html#exercise-3---step-9" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure">
<img src="Module6/genemania/images/GM37.png" alt="" />
<p class="caption">GM37</p>
</div>
<p></p>
</div>
</div>
<div id="some-definitions-1" class="section level2 hasAnchor">
<h2>SOME DEFINITIONS:<a href="genemania_web.html#some-definitions-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>What are the networks: Definition of the types of interaction:</strong></p>
<ul>
<li><p><strong>Shared domains</strong>: Protein domain data. Two gene products are linked if they have the same protein domain. These data are collected from domain databases, such as InterPro, SMART and Pfam.</p></li>
<li><p><strong>Co-localization</strong>: Genes expressed in the same tissue, or proteins found in the same location. Two genes are linked if they are both expressed in the same tissue or if their gene products are both identified in the same cellular location.</p></li>
<li><p><strong>Co-expression</strong>: Gene expression data. Two genes are linked if their expression levels are similar across conditions in a gene expression study. Most of these data are collected from the Gene Expression Omnibus (GEO); we only collect data associated with a publication.</p></li>
<li><p><strong>Predicted</strong>: Predicted functional relationships between genes, often protein interactions. A major source of predicted data is mapping known functional relationships from another organism via orthology.</p></li>
</ul>
<p><strong>What is defined by evidence sources?:</strong></p>
<ul>
<li><strong>Evidence sources</strong> are the information contained in the multiple databases that GeneMANIA uses to establish interaction between two genes.</li>
</ul>
<p><strong>Network:</strong></p>
<ul>
<li><p><strong>Node</strong> : circle representing the genes</p></li>
<li><p><strong>Edge</strong>: line that links two nodes and represent an interaction between two genes (multiple lines correspond to multiple sources</p></li>
<li><p><strong>Node size</strong>: Mapped to gene score, i.e.the degree to which GeneMANIA predicts the genes are related</p></li>
<li><p><strong>Thickness of edge</strong>: Strength/weight of interaction</p></li>
</ul>
<p><strong>Layout</strong> : The layout is different each time so the user can request the layout run multiple times until the user is satisfied with the result.</p>
<p><strong>in Networks tab:</strong></p>
<ul>
<li><strong>Percent weight (score)</strong> : a higher weight means that this network helped more to find related genes.</li>
</ul>
<p><strong>in Functions tab</strong> :</p>
<ul>
<li><p><strong>FDR</strong> : False discovery rate (FDR) is greater than or equal to the probability that this is a false positive.</p></li>
<li><p><strong>Coverage</strong> : (number of genes in the network with a given function) / (all genes in the genome with the function)</p></li>
</ul>
<div id="in-advanced-options-1" class="section level4 hasAnchor">
<h4>In advanced options:<a href="genemania_web.html#in-advanced-options-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li><p><strong>Network weighting?</strong> GeneMANIA can use a few different methods to weight networks when combining all networks to form the final composite network that results from a search. The default settings are usually appropriate, but you can choose a weighting method in the advanced option panel. (more details at <a href="http://pages.genemania.org/help/" class="uri">http://pages.genemania.org/help/</a>).</p></li>
<li><p><strong>Related genes</strong> : are genes added by GeneMANIA in addition to the genes from the query. It helps to grow the network and then to predict function of the query gene(s).</p></li>
<li><p><strong>The attributes</strong> represent the differences sources of evidence that can be used to build the network.</p></li>
</ul>
<p><strong>Notes</strong> :</p>
<ul>
<li><p>prostate cancer gene list is AKR1C3 AR CYB5A CYP11A1 CYP11B1 CYP11B2 CYP17A1 CYP19A1 CYP21A2 HSD17B1 HSD17B10 HSD17B11 HSD17B12 HSD17B13 HSD17B14 HSD17B2 HSD17B3 HSD17B4 HSD17B6 HSD17B7 HSD17B8 HSD3B1 HSD3B2 HSD3B7 RDH5 SHBG SRD5A1 SRD5A3 STAR.</p></li>
<li><p>mixed gene list is AKT1 AKT1S1 AKT2 ATM ATR BRCA1 BRCA2 CHEK1 CHEK2 FANCF FOLR1 FOLR2 FOLR3 FOXO1 FOXO3 MDC1 MLH1 MLST8 MSH2 MTOR PARP1 PDPK1 PIK3CA PIK3R1 PIK3R2 PTEN RAD51 RHEB RICTOR RPTOR SLC19A1 TSC1 TSC2</p></li>
<li><p><strong>Tip</strong>: look at GeneMANIA help pages when you run an analysis on your own after the workshop: <a href="http://pages.genemania.org/help/" class="uri">http://pages.genemania.org/help/</a>.</p></li>
</ul>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="genemania_cytoscape.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="module-6-review-of-the-tools.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": "https://github.com/BaderLab/CBW_Pathways_2023/commit/master/5.2-Module5_genenmania_web_lab.Rmd",
"text": null
},
"view": {
"link": "https://github.com/BaderLab/CBW_Pathways_2023/master/5.2-Module5_genenmania_web_lab.Rmd",
"text": null
},
"download": ["CBW_Pathways.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true
}
});
});
</script>

</body>

</html>
