<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Module 2 lab - g:Profiler | Pathways</title>
  <meta name="description" content="Course covers the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information." />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Module 2 lab - g:Profiler | Pathways" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Course covers the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Module 2 lab - g:Profiler | Pathways" />
  
  <meta name="twitter:description" content="Course covers the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information." />
  

<meta name="author" content="Gary Bader, Veronique Voisin, Ruth Isserlin, Robin Haw, Michael Hoffman" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/favicon.ico" type="image/x-icon" />
<link rel="prev" href="module-2-finding-over-represented-pathways-veronique-voisin.html"/>
<link rel="next" href="gsea-lab.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>




<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">CBW Pathways Workshop</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Pathway and Network Analysis of -Omics Data ( May 2021 )</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#meet-your-faculty"><i class="fa fa-check"></i>Meet your Faculty</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#gary-bader"><i class="fa fa-check"></i>Gary Bader</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#robin-haw"><i class="fa fa-check"></i>Robin Haw</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#michael-hoffman"><i class="fa fa-check"></i>Michael Hoffman</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#veronique-voisin"><i class="fa fa-check"></i>Veronique Voisin</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#ruth-isserlin"><i class="fa fa-check"></i>Ruth Isserlin</a></li>
</ul></li>
</ul></li>
<li><a href="welcome.html#welcome">Welcome <a id="welcome"></a></a><ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html"><i class="fa fa-check"></i>Class Materials</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#schedule"><i class="fa fa-check"></i>Workshop Schedule</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#pre-workshop"><i class="fa fa-check"></i>Pre-Workshop Materials and Laptop Setup Instructions</a><ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#laptop-setup-instructions"><i class="fa fa-check"></i>Laptop Setup Instructions</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#basic-programs"><i class="fa fa-check"></i>Basic programs</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#java-11-installation"><i class="fa fa-check"></i>Java 11 Installation</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#cytoscape-installation"><i class="fa fa-check"></i>Cytoscape Installation</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#gsea-installation"><i class="fa fa-check"></i>GSEA Installation</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#pre-workshop-tutorials"><i class="fa fa-check"></i>Pre-workshop Tutorials</a><ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#cytoscape-preparation-tutorials"><i class="fa fa-check"></i>Cytoscape Preparation tutorials</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#pre-workshop-readings-and-lectures"><i class="fa fa-check"></i>Pre-workshop Readings and Lectures</a></li>
<li><a href="welcome.html#additional-tutorials">Additional tutorials <a id="additionaltutorials"></a></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i>Module 1 - Introduction to Pathway and Network Analysis (Gary Bader)</a></li>
<li class="chapter" data-level="" data-path="module-2-finding-over-represented-pathways-veronique-voisin.html"><a href="module-2-finding-over-represented-pathways-veronique-voisin.html"><i class="fa fa-check"></i>Module 2: Finding Over-represented Pathways (Veronique Voisin)</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html"><i class="fa fa-check"></i>Module 2 lab - g:Profiler</a><ul>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#introduction"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#goal-of-the-exercise-1"><i class="fa fa-check"></i>Goal of the exercise 1</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#data"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#exercise-1"><i class="fa fa-check"></i>Exercise 1 - run g:Profiler</a><ul>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-1---launch-gprofiler."><i class="fa fa-check"></i>Step 1 - Launch g:Profiler.</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-2---input-query"><i class="fa fa-check"></i>Step 2 - input query</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-3---adjust-parameters."><i class="fa fa-check"></i>Step 3 - Adjust parameters.</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-4---run-query"><i class="fa fa-check"></i>Step 4 - Run query</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-5---explore-the-results."><i class="fa fa-check"></i>Step 5 - Explore the results.</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-6-expand-the-stats-tab"><i class="fa fa-check"></i>Step 6: Expand the stats tab</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-7-save-the-results"><i class="fa fa-check"></i>Step 7: Save the results</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-8-optional-but-recommended"><i class="fa fa-check"></i>Step 8 (Optional but recommended)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#exercise-2-load-and-use-a-custom-.gmt-file-and-run-the-query"><i class="fa fa-check"></i>Exercise 2: Load and use a custom .gmt file and run the query</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#optional-steps"><i class="fa fa-check"></i>Optional steps</a><ul>
<li><a href="gprofiler-lab.html#optional-1"><strong>Optional 1</strong>:</a></li>
<li><a href="gprofiler-lab.html#option-2"><strong>Option 2</strong>:</a></li>
<li><a href="gprofiler-lab.html#option-3"><strong>Option 3</strong> :</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html"><i class="fa fa-check"></i>Module 2 lab - GSEA</a><ul>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#introduction-1"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#goal-of-the-exercise"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#data-1"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#background"><i class="fa fa-check"></i>Background</a><ul>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#how-to-generate-a-rank-file."><i class="fa fa-check"></i>How to generate a rank file.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#start-the-exercise"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step1."><i class="fa fa-check"></i>Step1.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-2."><i class="fa fa-check"></i>Step 2.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step3."><i class="fa fa-check"></i>Step3.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-4."><i class="fa fa-check"></i>Step 4.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-5."><i class="fa fa-check"></i>Step 5.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#additional_information"><i class="fa fa-check"></i>Additional information</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-3-network-visualization-and-analysis-with-cytoscape.html"><a href="module-3-network-visualization-and-analysis-with-cytoscape.html"><i class="fa fa-check"></i>Module 3: Network Visualization and Analysis with Cytoscape</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html"><i class="fa fa-check"></i>Module 3 Lab: g:profiler Visualization</a><ul>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#goal-of-the-exercise-2"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#data-2"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#enrichmentmap"><i class="fa fa-check"></i>EnrichmentMap</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#description-of-this-exercise"><i class="fa fa-check"></i>Description of this exercise</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#start-the-exercise-1"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#exercise-1a---compare-different-gprofiler-geneset-size-results"><i class="fa fa-check"></i>Exercise 1a - compare different gprofiler geneset size results</a><ul>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#step-1"><i class="fa fa-check"></i>Step 1</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#step-2"><i class="fa fa-check"></i>Step 2</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#step3-explore-the-results"><i class="fa fa-check"></i>Step3: Explore the results:</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#explore-detailed-results"><i class="fa fa-check"></i>Explore Detailed results</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#exercise-1b---is-specifying-the-gmt-file-important"><i class="fa fa-check"></i>Exercise 1b - Is specifying the gmt file important?</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#exercise-1c---create-em-from-results-using-baderlab-genesets"><i class="fa fa-check"></i>Exercise 1c - create EM from results using Baderlab genesets</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#exercise-1c-optional---investigate-individual-pathways-in-genemania-or-string"><i class="fa fa-check"></i>Exercise 1c (optional) - investigate individual pathways in GeneMANIA or String</a><ul>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#genemania"><i class="fa fa-check"></i>GeneMANIA</a></li>
<li class="chapter" data-level="" data-path="gprofiler-mod3.html"><a href="gprofiler-mod3.html#string"><i class="fa fa-check"></i>String</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea-mod3.html"><a href="gsea-mod3.html"><i class="fa fa-check"></i>Module 3 Lab: GSEA Visualization</a><ul>
<li class="chapter" data-level="" data-path="gsea-mod3.html"><a href="gsea-mod3.html#goal-of-the-exercise-3"><i class="fa fa-check"></i>Goal of the exercise:</a></li>
<li class="chapter" data-level="" data-path="gsea-mod3.html"><a href="gsea-mod3.html#data-3"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="gsea-mod3.html"><a href="gsea-mod3.html#gsea-output-and-enrichmentmap"><i class="fa fa-check"></i>GSEA output and EnrichmentMap</a><ul>
<li class="chapter" data-level="" data-path="gsea-mod3.html"><a href="gsea-mod3.html#enrichmentmap-1"><i class="fa fa-check"></i>EnrichmentMap</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea-mod3.html"><a href="gsea-mod3.html#start-the-exercise-2"><i class="fa fa-check"></i>Start the exercise</a><ul>
<li class="chapter" data-level="" data-path="gsea-mod3.html"><a href="gsea-mod3.html#step-1-1"><i class="fa fa-check"></i>Step 1</a></li>
<li class="chapter" data-level="" data-path="gsea-mod3.html"><a href="gsea-mod3.html#step-2-1"><i class="fa fa-check"></i>Step 2</a></li>
<li class="chapter" data-level="" data-path="gsea-mod3.html"><a href="gsea-mod3.html#step-3"><i class="fa fa-check"></i>Step 3</a></li>
<li class="chapter" data-level="" data-path="gsea-mod3.html"><a href="gsea-mod3.html#step-4"><i class="fa fa-check"></i>Step 4</a></li>
<li class="chapter" data-level="" data-path="gsea-mod3.html"><a href="gsea-mod3.html#step-5"><i class="fa fa-check"></i>Step 5</a></li>
<li class="chapter" data-level="" data-path="gsea-mod3.html"><a href="gsea-mod3.html#step-6"><i class="fa fa-check"></i>Step 6</a></li>
<li class="chapter" data-level="" data-path="gsea-mod3.html"><a href="gsea-mod3.html#step-7"><i class="fa fa-check"></i>Step 7</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html"><i class="fa fa-check"></i>Module 3 Lab: (Bonus) Automation</a><ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#goal-of-the-exercise-4"><i class="fa fa-check"></i>Goal of the exercise:</a><ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#set-up"><i class="fa fa-check"></i>Set Up</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-1---launch-rstudio"><i class="fa fa-check"></i>Step 1 - launch RStudio</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-2---create-a-new-project"><i class="fa fa-check"></i>Step 2 - create a new project</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-3---open-example-up-rnotebook"><i class="fa fa-check"></i>Step 3 - Open example up RNotebook</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-4---define-notebook-parameters"><i class="fa fa-check"></i>Step 4 - Define Notebook parameters</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-5---step-through-notebook-to-run-the-analysis"><i class="fa fa-check"></i>Step 5 - Step through notebook to run the analysis</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#exercises"><i class="fa fa-check"></i>Exercises</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#additional-resources"><i class="fa fa-check"></i>Additional resources</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-4-in-depth-analysis-of-networks-and-pathways.html"><a href="module-4-in-depth-analysis-of-networks-and-pathways.html"><i class="fa fa-check"></i>Module 4: In-depth Analysis of Networks and Pathways</a></li>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html"><i class="fa fa-check"></i>Module 5: Regulatory Network Analysis</a><ul>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html#lecture"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html#practical-lab-1-chip_seq-data---great-and-meme-chip"><i class="fa fa-check"></i>Practical lab 1: chIP_seq data - GREAT and MEME-chIP</a></li>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html#practical-lab-2-gene-list---iregulon-and-enrichrenrichmentmap"><i class="fa fa-check"></i>Practical lab 2: gene list - iREgulon and enrichr/EnrichmentMap</a></li>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html#additional-slides-about-the-tools-segway-and-behst-presented-during-the-lecture"><i class="fa fa-check"></i>Additional slides about the tools Segway and BEHST presented during the lecture</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html"><i class="fa fa-check"></i>Module 5 Lab 1: Gene Regulation and Motif Analysis Practical Lab /chIP-seq</a><ul>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#goal-of-this-practical-lab"><i class="fa fa-check"></i>Goal of this practical lab</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#dataset-used-during-this-practical-lab"><i class="fa fa-check"></i>Dataset used during this practical lab</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#exercise-1---run-pathway-analysis-using-great"><i class="fa fa-check"></i>Exercise 1 - Run pathway analysis using GREAT</a><ul>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#perform-pathway-enrichment"><i class="fa fa-check"></i>Perform pathway enrichment</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#explore-the-results."><i class="fa fa-check"></i>Explore the results.</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#perform-pathway-enrichment---proximal-approach"><i class="fa fa-check"></i>Perform pathway enrichment - Proximal approach</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#explore-the-results.---proximal-analysis"><i class="fa fa-check"></i>Explore the results. - proximal analysis</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#exercise-2---build-an-enrichment-map-to-visualize-great-results"><i class="fa fa-check"></i>Exercise 2 - Build an enrichment map to visualize GREAT results</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#exercise-3-optional-auto-annotate-the-distal-enrichment-map-build-the-proximal-map-and-compare-both-the-proximal-and-distal-maps"><i class="fa fa-check"></i>Exercise 3 (optional): Auto-annotate the distal enrichment map, build the proximal map and compare both the proximal and distal maps</a><ul>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#optional-exercise-3a-autoannotate-the-enrichment-map"><i class="fa fa-check"></i>Optional exercise 3a: AutoAnnotate the enrichment map:</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#optional-exercise-3b-repeat-the-process-of-building-an-enrichment-map-using-the-proximal-data-proximal_gobp_greatexportall.tsv."><i class="fa fa-check"></i>Optional exercise 3b: Repeat the process of building an enrichment map using the proximal data (Proximal_GOBP_greatExportAll.tsv).</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#optional-exercise-3c-repeat-the-process-by-building-both-the-proximal-and-distal-enrichment-maps-at-the-same-time."><i class="fa fa-check"></i>Optional exercise 3c: Repeat the process by building both the Proximal and Distal enrichment maps at the same time.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#exercise-4-add-runx1-targets-and-runx1-ko-up-and-down-regulated-genes-on-the-distal-enrichment-map-as-a-new-gene-set."><i class="fa fa-check"></i>Exercise 4: Add RUNX1 targets and RUNX1 KO up and down regulated genes on the distal enrichment map as a new gene-set.</a><ul>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#step-4a-add-the-.gmt-file-to-the-enrichment-map"><i class="fa fa-check"></i>step 4a: Add the .gmt file to the enrichment map:</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#step-4b-optional-change-the-edge-style-of-the-signature-gene-sets"><i class="fa fa-check"></i>Step 4b Optional: Change the edge style of the signature gene-sets:</a></li>
</ul></li>
<li><a href="regulatory-network-chipseq-lab.html#exercise-5-learning-how-to-run-meme-chip-from-the-meme-suite-httpsmeme-suite.orgmemetoolsmeme-chip">Exercise 5: Learning how to run MEME-chip from the MEME suite (<span>https://meme-suite.org/meme/tools/meme-chip</span>)</a><ul>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#exercise-5a-downloading-sequences-from-.bed-coordinates"><i class="fa fa-check"></i>Exercise 5a: Downloading sequences from .bed coordinates</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#exercise-5b-run-meme-chip"><i class="fa fa-check"></i>Exercise 5b: Run MEME-chIP</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#exercise-6-optional-get-the-iregulon-runx1-targets-and-find-the-mouse-orthologs-using-gorth-from-gprofiler-to-create-the-gmt-file-used-in-exercise-4."><i class="fa fa-check"></i>Exercise 6 (optional): Get the iRegulon RUNX1 targets and find the mouse orthologs using g:Orth (from g:Profiler) to create the gmt file used in Exercise 4.</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-chipseq-lab.html"><a href="regulatory-network-chipseq-lab.html#end-of-lab"><i class="fa fa-check"></i>End of Lab</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html"><i class="fa fa-check"></i>Module 5 Lab 2: Gene Regulation and Motif Analysis Practical Lab / iRegulon</a><ul>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html#iregulon-lab"><i class="fa fa-check"></i>iRegulon lab</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html#goal"><i class="fa fa-check"></i>Goal</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html#exercise-1.-detect-regulons-from-co-expressed-genes"><i class="fa fa-check"></i>Exercise 1. Detect regulons from co-expressed genes</a><ul>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html#skills-learned-in-this-exercise"><i class="fa fa-check"></i>Skills learned in this exercise:</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html#steps"><i class="fa fa-check"></i>Steps</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html#exercise-2.-create-a-metatargetome-using-iregulon-and-merge-2-networks-in-cytoscape."><i class="fa fa-check"></i>Exercise 2. Create a metatargetome using iRegulon and merge 2 networks in Cytoscape.</a><ul>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html#steps-1"><i class="fa fa-check"></i>Steps</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html#use-or-precomputed-iregulon-results"><i class="fa fa-check"></i>Use or precomputed iRegulon results:</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html#notes-about-iregulon"><i class="fa fa-check"></i>Notes about iRegulon:</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html#notes-about-cytoscape"><i class="fa fa-check"></i>Notes about Cytoscape:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html#exercise-3.-use-enrichr-with-the-prostate-gene-list."><i class="fa fa-check"></i>Exercise 3. Use Enrichr with the prostate gene list.</a><ul>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html#goal-1"><i class="fa fa-check"></i>Goal</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html#steps-2"><i class="fa fa-check"></i>Steps</a></li>
<li class="chapter" data-level="" data-path="regulatory-network-lab.html"><a href="regulatory-network-lab.html#end-of-practical-lab"><i class="fa fa-check"></i>end of practical lab</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-6-gene-function-prediction.html"><a href="module-6-gene-function-prediction.html"><i class="fa fa-check"></i>Module 6: Gene Function Prediction</a></li>
<li class="chapter" data-level="" data-path="genemania-cytoscape.html"><a href="genemania-cytoscape.html"><i class="fa fa-check"></i>Module 6 Lab: GeneMANIA (Cytoscape version)</a><ul>
<li class="chapter" data-level="" data-path="genemania-cytoscape.html"><a href="genemania-cytoscape.html#goal-of-this-practical-lab-1"><i class="fa fa-check"></i>Goal of this practical lab</a></li>
<li class="chapter" data-level="" data-path="genemania-cytoscape.html"><a href="genemania-cytoscape.html#exercise-1-questions-and-steps-to-follow"><i class="fa fa-check"></i>EXERCISE 1: QUESTIONS AND STEPS TO FOLLOW</a><ul>
<li class="chapter" data-level="" data-path="genemania-cytoscape.html"><a href="genemania-cytoscape.html#steps-3"><i class="fa fa-check"></i>STEPS</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania-cytoscape.html"><a href="genemania-cytoscape.html#exercise-1-answers"><i class="fa fa-check"></i>EXERCISE 1 ANSWERS</a></li>
<li class="chapter" data-level="" data-path="genemania-cytoscape.html"><a href="genemania-cytoscape.html#exercise-2-questions-and-steps-to-follow"><i class="fa fa-check"></i>EXERCISE 2: QUESTIONS AND STEPS TO FOLLOW</a><ul>
<li class="chapter" data-level="" data-path="genemania-cytoscape.html"><a href="genemania-cytoscape.html#steps-4"><i class="fa fa-check"></i>STEPS</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania-cytoscape.html"><a href="genemania-cytoscape.html#exercise-3-questions-and-steps-to-follow"><i class="fa fa-check"></i>EXERCISE 3: QUESTIONS AND STEPS TO FOLLOW</a><ul>
<li class="chapter" data-level="" data-path="genemania-cytoscape.html"><a href="genemania-cytoscape.html#steps-5"><i class="fa fa-check"></i>STEPS</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania-cytoscape.html"><a href="genemania-cytoscape.html#some-definitions"><i class="fa fa-check"></i>SOME DEFINITIONS:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html"><i class="fa fa-check"></i>Module 6 Lab: GeneMANIA (web version)</a><ul>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#goal-of-this-practical-lab-2"><i class="fa fa-check"></i>Goal of this practical lab</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-1-questions-and-steps-to-follow-1"><i class="fa fa-check"></i>EXERCISE 1: QUESTIONS AND STEPS TO FOLLOW</a><ul>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#steps-6"><i class="fa fa-check"></i>STEPS</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-1-answers-detailed-explanation-and-screenshots"><i class="fa fa-check"></i>EXERCISE 1 ANSWERS: DETAILED EXPLANATION AND SCREENSHOTS</a><ul>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-1---steps-1-4"><i class="fa fa-check"></i>EXERCISE 1 - STEPS 1-4</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-1---step-5"><i class="fa fa-check"></i>EXERCISE 1 - STEP 5</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-1---step-6"><i class="fa fa-check"></i>EXERCISE 1 - STEP 6</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-1---step-7"><i class="fa fa-check"></i>Exercise 1 - STEP 7</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-1---step-8"><i class="fa fa-check"></i>Exercise 1 - STEP 8</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-1---step-9"><i class="fa fa-check"></i>Exercise 1 - STEP 9</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-1---step-10-layouts"><i class="fa fa-check"></i>Exercise 1 - STEP 10 (layouts)</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-1---step-11-save-an-image"><i class="fa fa-check"></i>Exercise 1 - STEP 11 (save an image)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-2-questions-and-steps-to-follow-1"><i class="fa fa-check"></i>EXERCISE 2: QUESTIONS AND STEPS TO FOLLOW</a><ul>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#steps-7"><i class="fa fa-check"></i>STEPS</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-2-answers-detailed-steps-and-screenshots"><i class="fa fa-check"></i>EXERCISE 2 ANSWERS: DETAILED STEPS AND SCREENSHOTS</a><ul>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-2---steps-1-to-4"><i class="fa fa-check"></i>Exercise 2 - STEPS 1 to 4</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-2---step-5"><i class="fa fa-check"></i>Exercise 2 - STEP 5</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-2---step-6."><i class="fa fa-check"></i>Exercise 2 - STEP 6.</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-2---step-7"><i class="fa fa-check"></i>Exercise 2 - STEP 7</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-2---step-8"><i class="fa fa-check"></i>Exercise 2 - STEP 8</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-2---step-9"><i class="fa fa-check"></i>Exercise 2 - STEP 9</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-2---step-10"><i class="fa fa-check"></i>Exercise 2 - STEP 10</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-2---step-11"><i class="fa fa-check"></i>Exercise 2 - STEP 11</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-2---step-12"><i class="fa fa-check"></i>Exercise 2 - STEP 12</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-2---step-13."><i class="fa fa-check"></i>Exercise 2 - STEP 13.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-3-questions-and-steps-to-follow-1"><i class="fa fa-check"></i>EXERCISE 3: QUESTIONS AND STEPS TO FOLLOW</a><ul>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#steps-8"><i class="fa fa-check"></i>STEPS</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-3-more-details-and-screenshots"><i class="fa fa-check"></i>Exercise 3: MORE DETAILS AND SCREENSHOTS</a><ul>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-3---steps-1---3"><i class="fa fa-check"></i>Exercise 3 - STEPS 1 - 3</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-3---step-4"><i class="fa fa-check"></i>Exercise 3 - STEP 4</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-3---step-5"><i class="fa fa-check"></i>Exercise 3 - STEP 5</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-3---steps-6"><i class="fa fa-check"></i>Exercise 3 - STEPS 6</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-3---step-7"><i class="fa fa-check"></i>Exercise 3 - STEP 7</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-3---step-8"><i class="fa fa-check"></i>Exercise 3 - STEP 8</a></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#exercise-3---step-9"><i class="fa fa-check"></i>Exercise 3 - STEP 9</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania-web.html"><a href="genemania-web.html#some-definitions-1"><i class="fa fa-check"></i>SOME DEFINITIONS:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html"><i class="fa fa-check"></i>Integrated Assignment</a><ul>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#final-slides"><i class="fa fa-check"></i>Final slides</a></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#goal-2"><i class="fa fa-check"></i>Goal</a></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#dataset-1"><i class="fa fa-check"></i>DATASET 1</a></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#background-1"><i class="fa fa-check"></i>Background</a></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#data-processing"><i class="fa fa-check"></i>Data processing</a></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#part-1-run-gprofiler"><i class="fa fa-check"></i>PART 1: run g:Profiler</a></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#part-2-save-as-generic-enrichment-map-output-be"><i class="fa fa-check"></i>PART 2: save as Generic Enrichment Map output (BE)</a></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#part-3-save-as-generic-enrichment-map-output-ne"><i class="fa fa-check"></i>PART 3: save as Generic Enrichment Map output (NE)</a></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#part-4-create-an-enrichment-map"><i class="fa fa-check"></i>PART 4: create an enrichment map</a></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#answers-gprofiler"><i class="fa fa-check"></i>Answers g:Profiler</a></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#part-5-gsea-run-and-create-an-enrichment-map"><i class="fa fa-check"></i>PART 5: GSEA (run and create an enrichment map)</a><ul>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#answers-gsea"><i class="fa fa-check"></i>Answers GSEA</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#part-6-iregulon"><i class="fa fa-check"></i>PART 6: iRegulon</a></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#dataset-2"><i class="fa fa-check"></i>DATASET 2</a></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#part-1-reactomefi"><i class="fa fa-check"></i>PART 1: ReactomeFI</a><ul>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#answers-reactome-fi"><i class="fa fa-check"></i>Answers REACTOME FI</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="integrated-assignment.html"><a href="integrated-assignment.html#part-2-genemania"><i class="fa fa-check"></i>PART 2: GeneMANIA</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="ass-automation.html"><a href="ass-automation.html"><i class="fa fa-check"></i>Integrated Assignment Bonus - Automation</a><ul>
<li class="chapter" data-level="" data-path="ass-automation.html"><a href="ass-automation.html#goal-of-the-exercise-5"><i class="fa fa-check"></i>Goal of the exercise:</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Pathways</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="gprofiler-lab" class="section level1">
<h1>Module 2 lab - g:Profiler</h1>
<p><strong>This work is licensed under a <a href="http://creativecommons.org/licenses/by-sa/3.0/deed.en_US">Creative Commons Attribution-ShareAlike 3.0 Unported License</a>. This means that you are able to copy, share and modify the work, as long as the result is distributed under the same license.</strong></p>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Performing Over-Representation Analysis (ORA) with <a href="https://biit.cs.ut.ee/gprofiler/gost">g:Profiler</a>.</p>
<p>The practical lab contains 2 exercises. The first exercise uses <a href="https://biit.cs.ut.ee/gprofiler/gost">g:Profiler</a> to perform gene-set enrichment analysis.</p>
</div>
<div id="goal-of-the-exercise-1" class="section level2">
<h2>Goal of the exercise 1</h2>
<p>Learn how to run <em>g:GOSt Functional profiling</em> from the g:Profiler website and explore the results.</p>
</div>
<div id="data" class="section level2">
<h2>Data</h2>
<p>g:Profiler requires a list of genes, one per line, in a text file or spreadsheet,
ready to copy and paste into a web page: for this, we use genes with frequent somatic SNVs identified in TCGA exome sequencing data of 3,200 tumors of 12 types. The MuSiC cancer driver mutation detection software was used to find 127 cancer driver genes that displayed higher than expected mutation frequencies in cancer samples (Supplementary Table 1, which is derived from column B of Supplementary Table 4 in <a href="https://www.nature.com/articles/nature12634">Kandoth C. et al.</a>. Genes are ranked in decreasing order of significance (FDR Q value) and mutation frequency (not shown).</p>
</div>
<div id="exercise-1" class="section level2">
<h2>Exercise 1 - run g:Profiler</h2>
<p>For this exercise, our goal is to copy and paste the list of genes into g:Profiler, adjust some parameters (e.g selecting the pathway databases), run the query and explore the results.</p>
<p>g:Profiler performs a gene-set enrichment analysis using a hypergeometric test (Fisher’s exact test) with the option to consider the ranking of the genes in the calculation of the enrichment significance (minimum hypergeometric test). The Gene Ontology Biological Process and Reactome are going to be used as pathway databases. The results are displayed as a table or downloadable as an Generic Enrichment Map (GEM) output file.</p>
<p>Before starting this exercise, download the required files:</p>
<div class="rmd-datadownload">
<p>
Right click on link below and select “Save Link As…”.
</p>
<p>
Place it in the corresponding module directory of your CBW work directory.
</p>
</div>
<ul>
<li><a href="./Module2/gprofiler/data/Supplementary_Table1_Cancer_drivers.txt">Supplementary_Tabe1_Cancer_drivers.txt</a></li>
</ul>
<p>We recommend saving all these files in a personal project data folder before starting. We also recommend creating an additional result data folder to save the files generated while performing the protocol.</p>
<div id="step-1---launch-gprofiler." class="section level3">
<h3>Step 1 - Launch g:Profiler.</h3>
<p>Open the g:Profiler website at <a href="https://biit.cs.ut.ee/gprofiler/gost">g:Profiler</a> in your web browser.</p>
</div>
<div id="step-2---input-query" class="section level3">
<h3>Step 2 - input query</h3>
<p>Paste the gene list (<a href="./Module2/gprofiler/data/Supplementary_Table1_Cancer_drivers.txt">Supplementary_Table1_Cancer_drivers.txt</a>) into the Query field in top-left corner of the screen.</p>
<p><img src="Module2/gprofiler/images/gp1.png" /></p>
<div class="rmd-note">
<p>
The gene list can be space-separated or one per line.<br> The organism for the analysis, Homo sapiens, is selected by default.<br> The input list can contain a mix of gene and protein IDs, symbols and accession numbers.<br> Duplicated and unrecognized IDs will be removed automatically, and ambiguous symbols can be refined in an interactive dialogue after submitting the query.
</p>
</div>
<div class="rmd-tip">
<p>
open the file in a simple text editor such as Notepad or Textedit to copy the list of genes. <br> Or right click on the file name above and select <strong>Open link in new tab</strong>
</p>
</div>
</div>
<div id="step-3---adjust-parameters." class="section level3">
<h3>Step 3 - Adjust parameters.</h3>
<p>3a. Click on the <em>Advanced options</em> tab (black rectangle) to expand it.</p>
<ul>
<li><p>Set <em>Significance threshold</em> to “Benjamini-Hochberg FDR”</p></li>
<li><p><em>User threshold</em> - select 0.05 if you want g:Profiler to return only pathways that are significant (FDR &lt; 0.05).</p></li>
</ul>
<div class="rmd-tip">
<p>
if g:Profiler does not return any results increase the threshold (0.1, then 1) to check that g:Profiler is running successfully but there are simply no significant results for your query.
</p>
</div>
<p align="center">
<img src="Module2/gprofiler/images/gp2.png" alt="workflow" width="50%" />
</p>
<p>3b. Click on the <em>Data sources</em> tab (black rectangle) to expand it.</p>
<ul>
<li>UnSelect all gene-set databases by clicking the “clear all” button.</li>
<li>In the <em>Gene Ontology</em> category, check <em>GO Biological Process</em> and <em>No electronic GO annotations</em>.</li>
</ul>
<div class="rmd-note">
<p>
<em>No electronic GO annotations</em> option will discard less reliable GO annotations (inferred from electronic annotations (IEAs)) that are not manually reviewed.
</p>
</div>
<div class="rmd-tip">
<p>
if g:Profiler does not return any results uncheck the <em>No electronic GO annotation</em> option to expand annotations used in the test.
</p>
</div>
<ul>
<li>In the <em>biological pathways</em> category, check <em>Reactome</em>.</li>
</ul>
<p align="center">
<img src="Module2/gprofiler/images/gp3.png" alt="workflow" width="50%" />
</p>
</div>
<div id="step-4---run-query" class="section level3">
<h3>Step 4 - Run query</h3>
<p>Click on the <em>Run query</em> button to run g:Profiler.</p>
<p><img src="Module2/gprofiler/images/gp3a.png" alt="workflow" width="100%" /></p>
<p>Scroll down page to see results.</p>
<div class="rmd-tip">
<p>
If after clicking on <em>Run query</em> button the analysis completes but there is the following message above results file - <em>Select the Ensembl ID with the most GO annotations (all)</em>. For each ambiguous gene select its correct mapping. Ambiguous mapping is often caused by multiple ensembl ids for a given gene and are easy to resolve as a user. Rerun query.
</p>
</div>
<p><img src="Module2/gprofiler/images/gp4.png" alt="workflow" width="100%" /></p>
</div>
<div id="step-5---explore-the-results." class="section level3">
<h3>Step 5 - Explore the results.</h3>
<p>Step 5a:</p>
<ul>
<li>After the query has run, the results are displayed at the bottom of the page, below the input parameters.</li>
<li>By default, the “Results” tab is selected. A global graph displays gene-sets that passed the significance threshold of 0.05 for the 2 gene-set databases that we have selected, GO Biological Process(<a href="GO:BP" class="uri">GO:BP</a>) and Reactome(REAC). Numbers in parentheses are indicating the number of gene-sets that passed the threshold.</li>
</ul>
<p><img src="Module2/gprofiler/images/gp5.png" alt="workflow" width="100%" /></p>
<p>Step5b:</p>
<ul>
<li>Click on “Detailed Results” to view the results in more depth. Two tables are displayed, one for each of the data sources selected. (If more than 2 data sources are selected there will be additional tables for each datasource) Each row of the table contains:
<ul>
<li><strong>Term name</strong> - gene-set name</li>
<li><strong>Term ID</strong> - gene-est identifier</li>
<li><strong>Padj</strong> - FDR value</li>
<li><strong>-log10(Padj)</strong> - enrichment score calculated using the formula -log10(padj)</li>
<li>variable number of gene columns (One for each gene in the query set) - If the gene is present in the current gene-set its cell is colored. For any data source besides GO the cell is colored black if the gene is found in the gene-set. For the GO data source cells are colored according to the annotation evidence code. Expand the legend tab for detailed coloring mapping of GO evidence codes.</li>
</ul></li>
</ul>
<p>The first table displays the gene-sets significantly enriched at FDR 0.05 for the <a href="GO:BP" class="uri">GO:BP</a> database.</p>
<p><img src="Module2/gprofiler/images/gp6.png" alt="workflow" width="100%" /></p>
<p>The second table displays the results corresponding to the Reactome database.</p>
<p><img src="Module2/gprofiler/images/gp7.png" alt="workflow" width="100%" /></p>
</div>
<div id="step-6-expand-the-stats-tab" class="section level3">
<h3>Step 6: Expand the stats tab</h3>
<p>Expand the <em>stats</em> tab by clicking on the double arrow located at the right of the tab.</p>
<p align="center">
<img src="Module2/gprofiler/images/gp7a.png" alt="workflow" width="75%" />
</p>
<p>It displays the gene set size (T), the size of our gene list (Q) , the number of genes that overlap between our gene list and the tested gene-set (TnQ) as well as the number of genes in the background (U).</p>
<ul>
<li>Above the <a href="GO:BP" class="uri">GO:BP</a> result table, locate the slide bar that enables to select for the minimum and maximum number of genes in the tested gene-sets (Term size).</li>
<li>Change the maximum <em>Term size</em> from 10000 to <strong>250</strong> and observe the results in the detailed stats panel:</li>
</ul>
<p><img src="Module2/gprofiler/images/gp8a.png" alt="workflow" width="100%" /></p>
<ul>
<li>Without filtering term size, the top terms were GO terms containing that could contain 4000 or 5000 genes and that were located high in the GO hierarchy (parent term).</li>
<li>With filtering the maximum term size to 250, the top list contains pathways of larger interpretative values. However, please note that the adjusted pvalues was calculated using all gene-sets without size filtering.</li>
</ul>
</div>
<div id="step-7-save-the-results" class="section level3">
<h3>Step 7: Save the results</h3>
<p>7a. In the <em>Detailed Results</em> panel, select “GEM” . It will save the results in a text file in the “Generic Enrichment Map” format that we will use to visualize using Cytoscape.</p>
<ul>
<li>select max term size to 10000 ( = no filtering by gene-set size) and click on the GEM button. A file is downloaded on your computer. (change the name to gProfiler_hsapiens_lab2_results_GEM_maxterm10000.gem.txt)
<p align="center">
<img src="Module2/gprofiler/images/gp13.png" alt="workflow" width="75%" />
</p></li>
<li>select max term size to 1000 ( = filter by gene-set size) and click on the GEM button. A file is downloaded on your computer. (change the name to gProfiler_hsapiens_lab2_results_GEM_maxterm1000.gem.txt)
<p align="center">
<img src="Module2/gprofiler/images/gp14.png" alt="workflow" width="75%" />
</p></li>
<li>select max term size to 250 ( = filter by gene-set size) and click on the GEM button. A file is downloaded on your computer. (change the name to gProfiler_hsapiens_lab2_results_GEM_maxterm250.gem.txt)
<p align="center">
<img src="Module2/gprofiler/images/gp14a.png" alt="workflow" width="75%" />
</p></li>
</ul>
<p>7b: Open the file that you saved using the geneset threshold of 250 using Microsoft Office Excel or in an equivalent software.</p>
<p>Observe the results included in this file:</p>
<ol style="list-style-type: decimal">
<li>Name of each gene-set</li>
<li>Description of each gene-set</li>
<li>significance of the overlap (pvalue)</li>
<li>significance of the overlap (adjusted pvalue/qvalue)</li>
<li>Phenotype</li>
<li>Genes included in each gene-set</li>
</ol>
<div class="rmd-question">
<p>
Which term has the best corrected p-value? <br> Which genes in our list are included in this term? <br> Observe that same genes can be present in several lines (pathways are related when they contain a lof of genes in common).
</p>
</div>
<div class="rmd-note">
<p>
The table is formatted for the input into Cytoscape EnrichmentMap. It is called the <a href="https://enrichmentmap.readthedocs.io/en/latest/FileFormats.html#generic-results-files"><em>generic format</em></a>. The p-value and FDR columns contain identical values because g:Profiler directly outputs the FDR (= corrected p-value) meaning that the p-value column is already the FDR. Phenotype 1 means that each pathway will be represented by red nodes on the enrichment map (presented during next module).
</p>
</div>
<p><img src="Module2/gprofiler/images/gp15.png" alt="workflow" width="100%" /></p>
<p>The term <em>regulation of cell cycle G1/S phase transition</em> is the most significant gene-set (=the lowest FDR value). Many gene-sets from the top of this list are related to each other and have genes in common.</p>
<hr />
</div>
<div id="step-8-optional-but-recommended" class="section level3">
<h3>Step 8 (Optional but recommended)</h3>
<p>8a. Download the pathway database files.</p>
<ul>
<li>Go to the top of the page and expand the “Data sources” tab. Click on the ‘combined name.gmt’ link located at bottom of this tab. It will download a file named <em>combined name.gmt</em> containing a pathway database gmt file with all the available sources.</li>
</ul>
<p align="center">
<img src="Module2/gprofiler/images/gp12.png" alt="workflow" width="60%" />
</p>
<p>8b. concatenate the <a href="GO:BP" class="uri">GO:BP</a> and Reactome gmt files:</p>
<p>If you want to create a smaller gmt file that doesn’t contain all of the gprofiler datasources you can instead download <em>name.gmt.zip</em> that contains each datasource as its own gmt file. You will need to concatenate the sources you require into one gmt file to use for later.</p>
<div id="option-1-manually-if-you-are-not-familiar-with-unix-commands" class="section level4">
<h4>Option 1: manually if you are not familiar with unix commands</h4>
<ul>
<li>open a text editor such a Notepad or equivalent</li>
<li>open hsapiens.GO:BP.name.gmt using the text editor</li>
<li>open gmt hsapiens.REAC.name.gmt using the text editor</li>
<li>copy-paste all the rows from REAC file together with all the rows in <a href="GO:BP" class="uri">GO:BP</a> file and save it as hsapiens.pathways.name.gmt .</li>
</ul>
</div>
<div id="option-2-using-the-cat-command-if-you-are-familiar-with-unix-commands" class="section level4">
<h4>Option 2: using the cat command if you are familiar with unix commands</h4>
<ul>
<li>open your terminal window</li>
<li>cd to the unzipped gprofiler_hsapiens.name folder</li>
<li>type the following command:</li>
</ul>
<pre><code> cat hsapiens.GO:BP.name.gmt hsapiens.REAC.name.gmt &gt; hsapiens.pathways.name.gmt</code></pre>
<div class="rmd-note">
<p>
you will be using this optional hsapiens.pathways.name.gmt file in Cytoscape EnrichmentMap.
</p>
</div>
<hr />
</div>
</div>
</div>
<div id="exercise-2-load-and-use-a-custom-.gmt-file-and-run-the-query" class="section level2">
<h2>Exercise 2: Load and use a custom .gmt file and run the query</h2>
<p>For this exercise, our goal is to copy and paste the list of genes into g:Profiler, upload a custom gmt file, adjust some parameters (e.g selecting the pathway databases), run the query and explore the results. Uploading a custom gmt file enables us to use alternate pathway data sources not available in g:Profiler.</p>
<p>We are going to use a gmt file that contains a database of pathway gene sets used for pathway enrichment analysis in the standard GMT format downloaded from <a href="http://baderlab.org/GeneSets" class="uri">http://baderlab.org/GeneSets</a> and updated monthly.</p>
<p>This file contains pathways from eight data sources:</p>
<ul>
<li>GO,</li>
<li>Reactome,</li>
<li>Panther,</li>
<li>NetPath,</li>
<li>NCI,</li>
<li>MSigDB curated gene sets (C2 collection, excluding Reactome and
KEGG),</li>
<li>MSigDB Hallmark (H collection) and</li>
<li>HumanCyc.</li>
</ul>
<p>A GMT file is a text file in which each line represents a gene set for a single pathway. Each line includes a pathway ID, a name and the list of associated genes in a tab-separated format. This file has been filtered to exlclude gene-sets that contained more than 250 genes as these gene-sets are associated with more general terms.</p>
<p>Before starting this exercise, download the required files:</p>
<div class="rmd-datadownload">
<p>
Right click on link below and select “Save Link As…”.
</p>
<p>
Place it in the corresponding module directory of your CBW work directory.
</p>
</div>
<ul>
<li><p><a href="./Module2/gprofiler/data/Supplementary_Table1_Cancer_drivers.txt">Supplementary_Tabe1_Cancer_drivers.txt</a></p></li>
<li><p><a href="./Module2/gprofiler/data/Human_GOBP_AllPathways_no_GO_iea_September_01_2020_symbol_maxgssize250.gmt">Baderlab_genesets.gmt (from September 2020)</a>.</p></li>
</ul>
<p>We recommend saving all these files in a personal project data folder before starting. We also recommend creating an additional result data folder to save the files generated while performing the protocol.</p>
<p>STEPS:</p>
<ul>
<li><p>Repeat step 1 to 3a from <a href="gprofiler-lab.html#exercise-1">Exercise 1</a> (go back to exercise 1 to get detailed instructions) Briefly:</p>
<ul>
<li>Step 1:
<ul>
<li>Open g:profiler</li>
</ul></li>
<li>Step 2a :
<ul>
<li>Copy and paste the gene list in the Query field</li>
</ul></li>
<li>Step 2b: Click on the <em>Advanced options</em> tab (black rectangle) to expand it.
<ul>
<li>Set <em>Significance threshold</em> to “Benjamini-Hochberg FDR”.</li>
</ul></li>
<li>Step 3a: Click on the <em>Data sources</em> tab (black rectangle) to expand it.
<ul>
<li><strong>Unselect all choices by clicking the “clear all” button.</strong></li>
</ul></li>
<li>Step 4: Click on the <em>Custom GMT</em> tab (black rectangle) to expand it.
<ul>
<li>Drag in the box the Baderlab gmt file <a href="./Module2/gprofiler/data/Human_GOBP_AllPathways_no_GO_iea_September_01_2020_symbol_maxgssize250.gmt">Baderlab_genesets.gmt</a>.</li>
<li>Once uploaded successfully, the name of the file is displayed in the “File name used” box.</li>
</ul></li>
</ul>
<p><img src="Module2/gprofiler/images/gp16.png" alt="workflow" width="100%" /></p>
<ul>
<li><p>Step 5: Click on <em>Run query</em> .</p></li>
<li><p>Step 6: Explore the detailed results</p>
<p><img src="Module2/gprofiler/images/gp17a.png" alt="workflow" width="100%" /></p></li>
<li><p>Step 7: Save the file as GEM</p></li>
</ul></li>
</ul>
<hr />
</div>
<div id="optional-steps" class="section level2">
<h2>Optional steps</h2>
<p>Please follow these optional steps if time permits and/or to explore more g:Profiler parameters.</p>
<p>Here below are 3 optional steps that cover several options offered by g:Profiler:</p>
<ol style="list-style-type: decimal">
<li>test different data sources,</li>
<li>take the order of the gene list into account,</li>
<li>use different types of multiple hypothesis correction methods.</li>
</ol>
<p>Use the same gene list as used in <a href="gprofiler-lab.html#exercise-1">exercise 1</a> and modify paramters listed above. Observe the results.</p>
<p align="center">
<img src="Module2/gprofiler/images/gp18.png" alt="workflow" width="60%" />
</p>
<div id="optional-1" class="section level3">
<h3><strong>Optional 1</strong>:</h3>
<p>If time permits, play with input parameters, e.g. add <em>TRANSFAC </em> and <em>miRTarBase</em> databases, rerun the query and explore the new results.</p>
<p align="center">
<img src="Module2/gprofiler/images/gp21.png" alt="workflow" width="60%" />
</p>
<div class="rmd-note">
<p>
<strong>Transfac</strong> putative transcription factor binding sites (TFBSs) from TRANSFAC database are retrieved into g:GOSt through a special prediction pipeline. First, TFBSs are found by matching TRANSFAC position specific matrices using the program Match on range +/-1kb from TSS as provided by APPRIS (Annotating principal splice isoforms) via Ensembl biomart. For genes with multiple primary TSS annotations we selected one with most TF matches. The matching range for C. elegans, D. melanogaster and S. cerevisiae is 1kb upstream from ATG (translation start site). A cut-off value to minimize the number of false positive matches (provided by TRANSFAC) is then applied to remove spurious motifs. Remaining matches are split into two inclusive groups based on the amount of matches, i.e TFBSs that have at least 1 match are classified as match class 0 and TFBSs that have at least 2 matches per gene are classified as match class 1.<br><br> <strong>mirTarBase</strong> is a database that holds experimentally validated information about genes that are targetted by miRNAs. We include all the organisms that are covered by mirTarBase.
</p>
</div>
</div>
<div id="option-2" class="section level3">
<h3><strong>Option 2</strong>:</h3>
<p>Re-run the g:Profiler using the <strong>ordered</strong> query checked.<br> This will run the minimum hypergeometric test. g:Profiler then performs incremental enrichment analysis with increasingly larger numbers of genes starting from the top of the list. When this option is checked, <strong>it assumes that the genes were preordered by significance with the most significant gene at the top of the list</strong>.<br> Compare the results between “ordered” and non ordered query.</p>
<div class="rmd-note">
<p>
for this practical lab, the genes were ordered by the number of mutations found in these genes for all samples. <br>For example, TP53, a highly mutated genes is listed at the top.
</p>
</div>
<p align="center">
<img src="Module2/gprofiler/images/gp20.png" alt="workflow" width="75%" />
</p>
</div>
<div id="option-3" class="section level3">
<h3><strong>Option 3</strong> :</h3>
<p>Re-run g:Profiler and select g:SCS or Bonferonni as method to correct for multiple hypothesis testing. Do you get any significant results?</p>
<p align="center">
<img src="Module2/gprofiler/images/gp19.png" alt="workflow" width="75%" />
</p>
<div class="rmd-note">
<p>
you can get detailed information about these methods at <a href="https://biit.cs.ut.ee/gprofiler/page/docs" class="uri">https://biit.cs.ut.ee/gprofiler/page/docs</a> in the section <em>Significance threshold</em>.
</p>
</div>
<hr />
<div class="rmd-tip">
<p>
Instead of using the g:Profiler website g:profiler can be run directly from R or python see g:Profiler document for more info at <a href="https://biit.cs.ut.ee/gprofiler/page/r" class="uri">https://biit.cs.ut.ee/gprofiler/page/r</a>
</p>
</div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="module-2-finding-over-represented-pathways-veronique-voisin.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="gsea-lab.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": "https://github.com/risserlin/CBW_Pathways_bookdown/commit/master/2.1-Module2lab_gprofiler.Rmd",
"text": null
},
"view": {
"link": "https://github.com/risserlin/CBW_Pathways_bookdown/blob/master/2.1-Module2lab_gprofiler.Rmd",
"text": null
},
"download": ["CBW_Pathways.epub"],
"toc": {
"collapse": "section",
"scroll_highlight": true
}
});
});
</script>

</body>

</html>
