<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Module 3 Lab: GSEA Visualization | Pathway and Network Analysis of -Omics Data ( June 2024 )</title>
  <meta name="description" content="Course covers the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information." />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="Module 3 Lab: GSEA Visualization | Pathway and Network Analysis of -Omics Data ( June 2024 )" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Course covers the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Module 3 Lab: GSEA Visualization | Pathway and Network Analysis of -Omics Data ( June 2024 )" />
  
  <meta name="twitter:description" content="Course covers the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information." />
  

<meta name="author" content="Gary Bader, Ruth Isserlin, Chaitra Sarathy, Veronique Voisin" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/favicon.ico" type="image/x-icon" />
<link rel="prev" href="gprofiler_mod3.html"/>
<link rel="next" href="automation.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://bioinformatics.ca/workshops/current-workshops/"><img src="./images/bioinfologo.png" width="280"></a><a href="./">CBW Pathways Workshop</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Canadian Bioinformatics Workshops</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html"><i class="fa fa-check"></i>Welcome <a id="welcome"></a></a>
<ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#meet-your-faculty"><i class="fa fa-check"></i>Meet your Faculty</a>
<ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#gary-bader"><i class="fa fa-check"></i>Gary Bader</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#lincoln-stein"><i class="fa fa-check"></i>Lincoln Stein</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#gregory-schwartz"><i class="fa fa-check"></i>Gregory Schwartz</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#veronique-voisin"><i class="fa fa-check"></i>Veronique Voisin</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#ruth-isserlin"><i class="fa fa-check"></i>Ruth Isserlin</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#chaitra-sarathy-phd"><i class="fa fa-check"></i>Chaitra Sarathy, PhD</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#nia-hughes"><i class="fa fa-check"></i>Nia Hughes</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#class-materials"><i class="fa fa-check"></i>Class Materials</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#schedule"><i class="fa fa-check"></i>Workshop Schedule</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#pre-workshop"><i class="fa fa-check"></i>Pre-Workshop Materials and Laptop Setup Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#laptop-setup-instructions"><i class="fa fa-check"></i>Laptop Setup Instructions</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#basic-programs"><i class="fa fa-check"></i>Basic programs</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#cytoscape-installation"><i class="fa fa-check"></i>Cytoscape Installation</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#gsea-installation"><i class="fa fa-check"></i>GSEA Installation</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#docker-installation"><i class="fa fa-check"></i>Docker Installation</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#pre-workshop-tutorials"><i class="fa fa-check"></i>Pre-workshop Tutorials</a>
<ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#cytoscape-preparation-tutorials"><i class="fa fa-check"></i>Cytoscape Preparation tutorials</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#r-tutorial"><i class="fa fa-check"></i>R Tutorial</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#pre-workshop-readings-and-lectures"><i class="fa fa-check"></i>Pre-workshop Readings and Lectures</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#additional-tutorials"><i class="fa fa-check"></i>Additional tutorials <a id="additionaltutorials"></a></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i>Module 1 - Introduction to Pathway and Network Analysis (Gary Bader)</a></li>
<li class="chapter" data-level="" data-path="module-2-finding-over-represented-pathways-veronique-voisin.html"><a href="module-2-finding-over-represented-pathways-veronique-voisin.html"><i class="fa fa-check"></i>Module 2: Finding Over-represented Pathways (Veronique Voisin)</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html"><i class="fa fa-check"></i>Module 2 lab - g:Profiler</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#introduction"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#goal-of-the-exercise-1"><i class="fa fa-check"></i>Goal of the exercise 1</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#data"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#exercise-1"><i class="fa fa-check"></i>Exercise 1 - run g:Profiler</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-1---launch-gprofiler."><i class="fa fa-check"></i>Step 1 - Launch g:Profiler.</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-2---input-query"><i class="fa fa-check"></i>Step 2 - input query</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-3---adjust-parameters."><i class="fa fa-check"></i>Step 3 - Adjust parameters.</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-4---run-query"><i class="fa fa-check"></i>Step 4 - Run query</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-5---explore-the-results."><i class="fa fa-check"></i>Step 5 - Explore the results.</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-6-expand-the-stats-tab"><i class="fa fa-check"></i>Step 6: Expand the stats tab</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-7-save-the-results"><i class="fa fa-check"></i>Step 7: Save the results</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-8-optional-but-recommended"><i class="fa fa-check"></i>Step 8 (Optional but recommended)</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-9-optional-by-recommended"><i class="fa fa-check"></i>Step 9 (Optional by recommended)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#exercise-2-load-and-use-a-custom-.gmt-file-and-run-the-query"><i class="fa fa-check"></i>Exercise 2: Load and use a custom .gmt file and run the query</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#optional-steps"><i class="fa fa-check"></i>Optional steps</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#optional-1"><i class="fa fa-check"></i><strong>Optional 1</strong>:</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#option-2"><i class="fa fa-check"></i><strong>Option 2</strong>:</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#option-3"><i class="fa fa-check"></i><strong>Option 3</strong> :</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#bonus---automation."><i class="fa fa-check"></i>Bonus - Automation.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html"><i class="fa fa-check"></i>Module 2 lab - GSEA</a>
<ul>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#introduction-1"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#goal-of-the-exercise"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#data-1"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#background"><i class="fa fa-check"></i>Background</a>
<ul>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#how-to-generate-a-rank-file."><i class="fa fa-check"></i>How to generate a rank file.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#start-the-exercise"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step1."><i class="fa fa-check"></i>Step1.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-2."><i class="fa fa-check"></i>Step 2.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step3."><i class="fa fa-check"></i>Step3.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-4."><i class="fa fa-check"></i>Step 4.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-5."><i class="fa fa-check"></i>Step 5.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#additional_information"><i class="fa fa-check"></i>Additional information</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#bonus---automation.-1"><i class="fa fa-check"></i>Bonus - Automation.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-3-network-visualization-and-analysis-with-cytoscape.html"><a href="module-3-network-visualization-and-analysis-with-cytoscape.html"><i class="fa fa-check"></i>Module 3: Network Visualization and Analysis with Cytoscape</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html"><i class="fa fa-check"></i>Module 3 Lab Primer: Cytoscape Primer</a>
<ul>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#goal-of-the-exercise-2"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#data-2"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#start-the-exercise-1"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-1a---create-network-from-table"><i class="fa fa-check"></i>Exercise 1a - Create Network from table</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-1b---load-node-attributes"><i class="fa fa-check"></i>Exercise 1b - Load node attributes</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-1c---map-node-attributes-to-visual-style"><i class="fa fa-check"></i>Exercise 1c - Map node attributes to Visual Style</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-2---work-with-larger-networks"><i class="fa fa-check"></i>Exercise 2 - Work with larger networks</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-3---perform-basic-enrichment-analysis-using-enrichmenttable"><i class="fa fa-check"></i>Exercise 3 - Perform basic enrichment analysis using EnrichmentTable</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#enrichmenttabl-features"><i class="fa fa-check"></i>Exercise 3B - create Enrichment Map and Enhanced graphics nodes from EnrichmentTable</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-4---load-network-from-ndex"><i class="fa fa-check"></i>Exercise 4 - Load network from NDex</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html"><i class="fa fa-check"></i>Module 3 Lab: g:profiler Visualization</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#goal-of-the-exercise-3"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#data-3"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#enrichmentmap"><i class="fa fa-check"></i>EnrichmentMap</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#description-of-this-exercise"><i class="fa fa-check"></i>Description of this exercise</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#start-the-exercise-2"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#exercise-1a---compare-different-gprofiler-geneset-size-results"><i class="fa fa-check"></i>Exercise 1a - compare different gprofiler geneset size results</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#step-1"><i class="fa fa-check"></i>Step 1</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#step-2"><i class="fa fa-check"></i>Step 2</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#step3-explore-the-results"><i class="fa fa-check"></i>Step3: Explore the results:</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#explore-detailed-results"><i class="fa fa-check"></i>Explore Detailed results</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#exercise-1b---is-specifying-the-gmt-file-important"><i class="fa fa-check"></i>Exercise 1b - Is specifying the gmt file important?</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#exercise-1c---create-em-from-results-using-baderlab-genesets"><i class="fa fa-check"></i>Exercise 1c - create EM from results using Baderlab genesets</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#exercise-1d-optional---investigate-individual-pathways-in-genemania-or-string"><i class="fa fa-check"></i>Exercise 1d (optional) - investigate individual pathways in GeneMANIA or String</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#genemania"><i class="fa fa-check"></i>GeneMANIA</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#string"><i class="fa fa-check"></i>String</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#bonus---automation.-2"><i class="fa fa-check"></i>Bonus - Automation.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html"><i class="fa fa-check"></i>Module 3 Lab: GSEA Visualization</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#goal-of-the-exercise-4"><i class="fa fa-check"></i>Goal of the exercise:</a></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#data-4"><i class="fa fa-check"></i>Data</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#enrichmentmap-1"><i class="fa fa-check"></i>EnrichmentMap</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#exercise-1---gsea-output-and-enrichmentmap"><i class="fa fa-check"></i>Exercise 1 - GSEA output and EnrichmentMap</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-1-1"><i class="fa fa-check"></i>Step 1</a></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-2-1"><i class="fa fa-check"></i>Step 2</a></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-3"><i class="fa fa-check"></i>Step 3</a></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-4"><i class="fa fa-check"></i>Step 4</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#exercise-2---post-analysis-add-drug-target-gene-sets-to-the-network"><i class="fa fa-check"></i>Exercise 2 - Post analysis (add drug target gene-sets to the network)</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-5"><i class="fa fa-check"></i>Step 5</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#exercise-3---autoannotate-the-network"><i class="fa fa-check"></i>Exercise 3 - Autoannotate the Network</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-6"><i class="fa fa-check"></i>Step 6</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#exercise-4-optional---explore-results-in-genemania-or-string"><i class="fa fa-check"></i>Exercise 4 (Optional) - Explore results in GeneMANIA or STRING</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-7"><i class="fa fa-check"></i>Step 7</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#bonus---automation.-3"><i class="fa fa-check"></i>Bonus - Automation.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html"><i class="fa fa-check"></i>Module 3 Lab: (Bonus) Automation</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#goal-of-the-exercise-5"><i class="fa fa-check"></i>Goal of the exercise:</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#set-up---option-1---install-rrstudio"><i class="fa fa-check"></i>Set Up - Option 1 - Install R/Rstudio</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#set-up---option-2---docker-image-with-rrstudio"><i class="fa fa-check"></i>Set Up - Option 2 - Docker image with R/Rstudio</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#what-is-docker"><i class="fa fa-check"></i>What is docker?</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#docker---basic-term-definition"><i class="fa fa-check"></i>Docker - Basic term definition</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#container"><i class="fa fa-check"></i>Container</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#image"><i class="fa fa-check"></i>Image</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#docker-volumes"><i class="fa fa-check"></i>Docker Volumes</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#r_docker"><i class="fa fa-check"></i>Install Docker</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#windows"><i class="fa fa-check"></i>Windows</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#macos-linux"><i class="fa fa-check"></i>MacOS / Linux</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#create-your-first-notebook-using-docker"><i class="fa fa-check"></i>Create your first notebook using Docker</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#start-coding"><i class="fa fa-check"></i>Start coding!</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#start-using-automation"><i class="fa fa-check"></i>Start using automation</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-1---launch-rstudio"><i class="fa fa-check"></i>Step 1 - launch RStudio</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-2---create-a-new-project"><i class="fa fa-check"></i>Step 2 - create a new project</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-3---open-example-up-rnotebook"><i class="fa fa-check"></i>Step 3 - Open example up RNotebook</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-4---define-notebook-parameters"><i class="fa fa-check"></i>Step 4 - Define Notebook parameters</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-5---step-through-notebook-to-run-the-analysis"><i class="fa fa-check"></i>Step 5 - Step through notebook to run the analysis</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#exercises"><i class="fa fa-check"></i>Exercises</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#additional-resources"><i class="fa fa-check"></i>Additional resources</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-4-in-depth-analysis-of-networks-and-pathways.html"><a href="module-4-in-depth-analysis-of-networks-and-pathways.html"><i class="fa fa-check"></i>Module 4: In-depth Analysis of Networks and Pathways</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html"><i class="fa fa-check"></i>Module 4 Lab: ReactomeFI</a>
<ul>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#goal-of-this-practical-lab"><i class="fa fa-check"></i>Goal of this practical lab</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#data-download-the-following-files-on-your-computer-before-starting-the-practical-lab."><i class="fa fa-check"></i>Data: download the following files on your computer before starting the practical lab.</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#exercise-1-use-the-reactome-functional-interaction-fi-network"><i class="fa fa-check"></i>Exercise 1: Use the Reactome Functional Interaction (FI) Network</a>
<ul>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#question-1-describe-the-size-and-composition-of-the-network"><i class="fa fa-check"></i>Question 1: Describe the size and composition of the network?</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#question-2-after-clustering-how-many-modules-are-there"><i class="fa fa-check"></i>Question 2: After clustering, how many modules are there?</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#query-information-about-the-interaction-between-2-genes"><i class="fa fa-check"></i>Query information about the interaction between 2 genes:</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#question-3-what-are-the-most-significant-pathways-in-each-module"><i class="fa fa-check"></i>Question 3: What are the most significant pathways in each module?</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#set-the-size-of-the-nodes-proportional-to-the-mutation-frequencies-in-each-cancer"><i class="fa fa-check"></i>Set the size of the nodes proportional to the mutation frequencies in each cancer</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#play-around-with-the-styles-change-transparency-and-colors"><i class="fa fa-check"></i>Play around with the styles: change transparency and colors</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#create-a-pie-chart"><i class="fa fa-check"></i>Create a pie chart</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#create-a-subnetwork"><i class="fa fa-check"></i>Create a subnetwork</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#fetch-cancer-drugs-on-the-created-subnetwork"><i class="fa fa-check"></i>Fetch Cancer drugs on the created subnetwork</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#save-the-network-as-an-image-for-publication"><i class="fa fa-check"></i>Save the network as an image for publication</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#exercise-2a-explore-reactome-pathways"><i class="fa fa-check"></i>Exercise 2a: Explore Reactome Pathways</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#exercise-2b-pathway-enrichment-analysis-using-a-simple-gene-list"><i class="fa fa-check"></i>Exercise 2b: Pathway enrichment analysis using a simple gene list</a>
<ul>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#question-1-what-are-the-most-significant-biological-pathways-based-on-the-fdr"><i class="fa fa-check"></i>Question 1: What are the most significant biological pathways based on the FDR?</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#answer-to-question-1"><i class="fa fa-check"></i>Answer to Question 1</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#exercise-2c-pathway-based-analysis-using-a-rank-gene-list-gsea"><i class="fa fa-check"></i>Exercise 2c: Pathway-based analysis using a rank gene list (GSEA)</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#automation-for-advanced-users"><i class="fa fa-check"></i>Automation ( for advanced users)</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#reference-guide-bonus-exercises"><i class="fa fa-check"></i>Reference guide /bonus exercises:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-5-gene-function-prediction.html"><a href="module-5-gene-function-prediction.html"><i class="fa fa-check"></i>Module 5: Gene Function Prediction</a></li>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html"><i class="fa fa-check"></i>Module 5 Lab: GeneMANIA (Cytoscape version)</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#goal-of-this-practical-lab-1"><i class="fa fa-check"></i>Goal of this practical lab</a></li>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#exercise-1-searching-genemania-with-single-gene"><i class="fa fa-check"></i>EXERCISE 1: Searching GeneMANIA with single gene</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#answers"><i class="fa fa-check"></i>ANSWERS</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#exercise-2-searching-genemania-with-gene-list"><i class="fa fa-check"></i>EXERCISE 2: Searching GeneMANIA with gene list</a></li>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#exercise-3-searching-genemania-with-mixed-gene-list"><i class="fa fa-check"></i>EXERCISE 3: Searching GeneMANIA with mixed gene list</a></li>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#genemania-definitions"><i class="fa fa-check"></i>GeneMANIA DEFINITIONS:</a></li>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#exercise-4-optional-discover-the-stringapp"><i class="fa fa-check"></i>EXERCISE 4 (OPTIONAL): Discover the stringApp</a></li>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#more-string-information-and-tutorials"><i class="fa fa-check"></i>More STRING information and tutorials:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html"><i class="fa fa-check"></i>Module 5 Lab: GeneMANIA (web version)</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#goal-of-this-practical-lab-2"><i class="fa fa-check"></i>Goal of this practical lab</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1-questions-and-steps-to-follow"><i class="fa fa-check"></i>EXERCISE 1: QUESTIONS AND STEPS TO FOLLOW</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1-answers-detailed-explanation-and-screenshots"><i class="fa fa-check"></i>EXERCISE 1 ANSWERS: DETAILED EXPLANATION AND SCREENSHOTS</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---steps-1-4"><i class="fa fa-check"></i>EXERCISE 1 - STEPS 1-4</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---step-5"><i class="fa fa-check"></i>EXERCISE 1 - STEP 5</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---step-6"><i class="fa fa-check"></i>EXERCISE 1 - STEP 6</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---step-7"><i class="fa fa-check"></i>Exercise 1 - STEP 7</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---step-8"><i class="fa fa-check"></i>Exercise 1 - STEP 8</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---step-9"><i class="fa fa-check"></i>Exercise 1 - STEP 9</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---step-10-layouts"><i class="fa fa-check"></i>Exercise 1 - STEP 10 (layouts)</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---step-11-save-an-image"><i class="fa fa-check"></i>Exercise 1 - STEP 11 (save an image)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2-questions-and-steps-to-follow"><i class="fa fa-check"></i>EXERCISE 2: QUESTIONS AND STEPS TO FOLLOW</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2-answers-detailed-steps-and-screenshots"><i class="fa fa-check"></i>EXERCISE 2 ANSWERS: DETAILED STEPS AND SCREENSHOTS</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---steps-1-to-4"><i class="fa fa-check"></i>Exercise 2 - STEPS 1 to 4</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-5"><i class="fa fa-check"></i>Exercise 2 - STEP 5</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-6."><i class="fa fa-check"></i>Exercise 2 - STEP 6.</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-7"><i class="fa fa-check"></i>Exercise 2 - STEP 7</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-8"><i class="fa fa-check"></i>Exercise 2 - STEP 8</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-9"><i class="fa fa-check"></i>Exercise 2 - STEP 9</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-10"><i class="fa fa-check"></i>Exercise 2 - STEP 10</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-11"><i class="fa fa-check"></i>Exercise 2 - STEP 11</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-12"><i class="fa fa-check"></i>Exercise 2 - STEP 12</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-13."><i class="fa fa-check"></i>Exercise 2 - STEP 13.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3-questions-and-steps-to-follow"><i class="fa fa-check"></i>EXERCISE 3: QUESTIONS AND STEPS TO FOLLOW</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3-more-details-and-screenshots"><i class="fa fa-check"></i>Exercise 3: MORE DETAILS AND SCREENSHOTS</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3---steps-1---3"><i class="fa fa-check"></i>Exercise 3 - STEPS 1 - 3</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3---step-4-step5"><i class="fa fa-check"></i>Exercise 3 - STEP 4/ STEP5</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3---steps-6"><i class="fa fa-check"></i>Exercise 3 - STEPS 6</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3---step-7"><i class="fa fa-check"></i>Exercise 3 - STEP 7</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3---step-8"><i class="fa fa-check"></i>Exercise 3 - STEP 8</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3---step-9"><i class="fa fa-check"></i>Exercise 3 - STEP 9</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#some-definitions"><i class="fa fa-check"></i>SOME DEFINITIONS:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-6-cell-cell-communication.html"><a href="module-6-cell-cell-communication.html"><i class="fa fa-check"></i>Module 6: Cell Cell Communication</a>
<ul>
<li class="chapter" data-level="" data-path="module-6-cell-cell-communication.html"><a href="module-6-cell-cell-communication.html#module-6-lecture-cell-cell-communication."><i class="fa fa-check"></i>Module 6 lecture : Cell-Cell Communication.</a></li>
<li class="chapter" data-level="" data-path="module-6-cell-cell-communication.html"><a href="module-6-cell-cell-communication.html#scrna-lab-praticals"><i class="fa fa-check"></i>scRNA lab praticals</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-6-review-of-the-tools.html"><a href="module-6-review-of-the-tools.html"><i class="fa fa-check"></i>Module 6: Review of the tools</a>
<ul>
<li class="chapter" data-level="" data-path="module-6-review-of-the-tools.html"><a href="module-6-review-of-the-tools.html#final-slides"><i class="fa fa-check"></i>Final slides</a></li>
<li class="chapter" data-level="" data-path="module-6-review-of-the-tools.html"><a href="module-6-review-of-the-tools.html#scrna-lab-praticals-1"><i class="fa fa-check"></i>scRNA lab praticals</a></li>
<li class="chapter" data-level="" data-path="module-6-review-of-the-tools.html"><a href="module-6-review-of-the-tools.html#integrated-assignment"><i class="fa fa-check"></i>Integrated assignment</a></li>
<li class="chapter" data-level="" data-path="module-6-review-of-the-tools.html"><a href="module-6-review-of-the-tools.html#integrated-assignment-bonus"><i class="fa fa-check"></i>Integrated assignment bonus</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html"><i class="fa fa-check"></i>Module 6 lab 1: scRNA PBMC</a>
<ul>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#introduction-2"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#pmbc3k-seurat-pipeline"><i class="fa fa-check"></i>Pmbc3k Seurat Pipeline</a></li>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#load-libraries"><i class="fa fa-check"></i>load libraries</a></li>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#load-the-pbmc-dataset"><i class="fa fa-check"></i>Load the PBMC dataset</a></li>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#process-the-dataset"><i class="fa fa-check"></i>Process the dataset</a></li>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#assign-cell-type-identity-to-clusters"><i class="fa fa-check"></i>Assign cell type identity to clusters</a></li>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#find-differentially-expressed-features-cluster-biomarkers"><i class="fa fa-check"></i>Find differentially expressed features (cluster biomarkers)</a></li>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#create-gene-list-for-each-cluster-to-use-with-gprofiler"><i class="fa fa-check"></i>Create Gene list for each cluster to use with g:Profiler</a></li>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#data-gene-lists-for-each-cluster"><i class="fa fa-check"></i>Data (gene lists for each cluster)</a></li>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#run-pathway-enrichment-analysis-using-gprofiler"><i class="fa fa-check"></i>Run pathway enrichment analysis using g:Profiler</a></li>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#create-an-enrichment-map-in-cytoscape"><i class="fa fa-check"></i>Create an enrichment map in Cytoscape</a></li>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#gsea-from-pseudobulk"><i class="fa fa-check"></i>GSEA from pseudobulk</a>
<ul>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#pseudobulk-creation-differential-expression-and-rank-file"><i class="fa fa-check"></i>pseudobulk creation, differential expression and rank file</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#run-gsea"><i class="fa fa-check"></i>run GSEA:</a></li>
<li class="chapter" data-level="" data-path="scRNA_PBMC.html"><a href="scRNA_PBMC.html#create-an-enrichmentmap"><i class="fa fa-check"></i>Create an EnrichmentMap:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html"><i class="fa fa-check"></i>Module 6 lab 2- scRNA Glioblastoma</a>
<ul>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#introduction-3"><i class="fa fa-check"></i>Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#goal"><i class="fa fa-check"></i>Goal</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#data-5"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#overview"><i class="fa fa-check"></i>Overview</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#part-1---run-gprofiler-can-module8-exercise-1-optional"><i class="fa fa-check"></i>Part 1 - run g:Profiler {#can-module8-exercise-1} [OPTIONAL]</a>
<ul>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#step-1---launch-gprofiler.-1"><i class="fa fa-check"></i>Step 1 - Launch g:Profiler.</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#step-2---input-query-1"><i class="fa fa-check"></i>Step 2 - input query</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#step-3---adjust-parameters.-1"><i class="fa fa-check"></i>Step 3 - Adjust parameters.</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#step-4---run-query-1"><i class="fa fa-check"></i>Step 4 - Run query</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#step-5---explore-the-results.-1"><i class="fa fa-check"></i>Step 5 - Explore the results.</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#step-6-expand-the-stats-tab-1"><i class="fa fa-check"></i>Step 6: Expand the stats tab</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#step-7-save-the-results-1"><i class="fa fa-check"></i>Step 7: Save the results</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#step-8-optional-but-recommended-1"><i class="fa fa-check"></i>Step 8 (Optional but recommended)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#part-2---cytoscapeenrichmentmap-exercise-2-optional"><i class="fa fa-check"></i>Part 2 - Cytoscape/EnrichmentMap {#exercise-2} [OPTIONAL]</a>
<ul>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#goal-of-the-exercise-6"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#data-6"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#enrichmentmap-2"><i class="fa fa-check"></i>EnrichmentMap</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#description-of-this-exercise-1"><i class="fa fa-check"></i>Description of this exercise</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#start-the-exercise-3"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#step-1-2"><i class="fa fa-check"></i>Step 1</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#step-2-2"><i class="fa fa-check"></i>Step 2</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#step-3-explore-detailed-results"><i class="fa fa-check"></i>Step 3: Explore Detailed results</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#step-4-optional-autoannotate-the-enrichment-map"><i class="fa fa-check"></i>Step 4 [OPTIONAL]: AutoAnnotate the enrichment map</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#exercise-3"><i class="fa fa-check"></i>Part 3 - Master map using multiple datasets</a>
<ul>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#goal-1"><i class="fa fa-check"></i>Goal</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#data-7"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#start-the-exercise-4"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#step-1-3"><i class="fa fa-check"></i>Step 1</a></li>
<li class="chapter" data-level="" data-path="scRNA_glioblastoma.html"><a href="scRNA_glioblastoma.html#step-2-3"><i class="fa fa-check"></i>Step 2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html"><i class="fa fa-check"></i>Module 7 Integrated Assignment</a>
<ul>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#goal-2"><i class="fa fa-check"></i>Goal</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#dataset-1"><i class="fa fa-check"></i>DATASET 1</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#background-1"><i class="fa fa-check"></i>Background</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#data-processing"><i class="fa fa-check"></i>Data processing</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-1-run-gprofiler"><i class="fa fa-check"></i>PART 1: run g:Profiler</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-2-save-as-generic-enrichment-map-output-be"><i class="fa fa-check"></i>PART 2: save as Generic Enrichment Map output (BE)</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-3-save-as-generic-enrichment-map-output-ne"><i class="fa fa-check"></i>PART 3: save as Generic Enrichment Map output (NE)</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-4-create-an-enrichment-map"><i class="fa fa-check"></i>PART 4: create an enrichment map</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#answers-gprofiler"><i class="fa fa-check"></i>Answers g:Profiler</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-5-gsea-run-and-create-an-enrichment-map"><i class="fa fa-check"></i>PART 5: GSEA (run and create an enrichment map)</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-6-iregulon"><i class="fa fa-check"></i>PART 6: iRegulon</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#dataset-2"><i class="fa fa-check"></i>DATASET 2</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-1-reactomefi"><i class="fa fa-check"></i>PART 1: ReactomeFI</a>
<ul>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#answers-reactome-fi"><i class="fa fa-check"></i>Answers REACTOME FI</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-2-genemania"><i class="fa fa-check"></i>PART 2: GeneMANIA</a>
<ul>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#answers-genemania"><i class="fa fa-check"></i>Answers GeneMANIA</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="ass_automation.html"><a href="ass_automation.html"><i class="fa fa-check"></i>Module 7 Integrated Assignment Bonus - Automation</a>
<ul>
<li class="chapter" data-level="" data-path="ass_automation.html"><a href="ass_automation.html#goal-of-the-exercise-7"><i class="fa fa-check"></i>Goal of the exercise:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="scRNA_cellPhoneDB.html"><a href="scRNA_cellPhoneDB.html"><i class="fa fa-check"></i>Module 6 lab part2: cellPhoneDB</a>
<ul>
<li class="chapter" data-level="" data-path="scRNA_cellPhoneDB.html"><a href="scRNA_cellPhoneDB.html#cell-cell-communication-in-scrna-cellphonedb"><i class="fa fa-check"></i>Cell-Cell communication in scRNA: CellPhoneDB</a>
<ul>
<li class="chapter" data-level="" data-path="scRNA_cellPhoneDB.html"><a href="scRNA_cellPhoneDB.html#presentation"><i class="fa fa-check"></i>Presentation</a></li>
<li class="chapter" data-level="" data-path="scRNA_cellPhoneDB.html"><a href="scRNA_cellPhoneDB.html#method"><i class="fa fa-check"></i>Method</a></li>
<li class="chapter" data-level="" data-path="scRNA_cellPhoneDB.html"><a href="scRNA_cellPhoneDB.html#dataset-and-references"><i class="fa fa-check"></i>Dataset and references</a></li>
<li class="chapter" data-level="" data-path="scRNA_cellPhoneDB.html"><a href="scRNA_cellPhoneDB.html#dataset-preprocessing-and-running-cellphonedb"><i class="fa fa-check"></i>Dataset preprocessing and running CellPhoneDB</a></li>
<li class="chapter" data-level="" data-path="scRNA_cellPhoneDB.html"><a href="scRNA_cellPhoneDB.html#examining-the-results"><i class="fa fa-check"></i>Examining the results</a></li>
<li class="chapter" data-level="" data-path="scRNA_cellPhoneDB.html"><a href="scRNA_cellPhoneDB.html#visualization-using-cytoscape"><i class="fa fa-check"></i>Visualization using Cytoscape</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="scRNA_NEST.html"><a href="scRNA_NEST.html"><i class="fa fa-check"></i>Module 6 lab part3: NEST</a>
<ul>
<li class="chapter" data-level="" data-path="scRNA_NEST.html"><a href="scRNA_NEST.html#cell-cell-communication-ccc-in-spatial-transcriptomics-using-nest"><i class="fa fa-check"></i>Cell-Cell Communication (CCC) in spatial transcriptomics using NEST</a>
<ul>
<li class="chapter" data-level="" data-path="scRNA_NEST.html"><a href="scRNA_NEST.html#presentation-of-nest-neural-network-on-spatial-transcriptomics"><i class="fa fa-check"></i>Presentation of NEST (NEural network on Spatial Transcriptomics)</a></li>
<li class="chapter" data-level="" data-path="scRNA_NEST.html"><a href="scRNA_NEST.html#how-to-run-nest"><i class="fa fa-check"></i>How to run NEST</a></li>
<li class="chapter" data-level="" data-path="scRNA_NEST.html"><a href="scRNA_NEST.html#practical-lab-pancreatic-ductal-adenocarcinoma-pdac"><i class="fa fa-check"></i>Practical lab : Pancreatic Ductal Adenocarcinoma (PDAC)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html"><i class="fa fa-check"></i>Optional Module 7: Regulatory Network Analysis</a>
<ul>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html#lecture"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html#practical-lab-1-chip_seq-data---great-and-meme-chip"><i class="fa fa-check"></i>Practical lab 1: chIP_seq data - GREAT and MEME-chIP</a></li>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html#practical-lab-2-gene-list---iregulon-and-enrichrenrichmentmap"><i class="fa fa-check"></i>Practical lab 2: gene list - iREgulon and enrichr/EnrichmentMap</a></li>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html#additional-slides-about-the-tools-segway-and-behst-presented-during-the-lecture"><i class="fa fa-check"></i>Additional slides about the tools Segway and BEHST presented during the lecture</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html"><i class="fa fa-check"></i>Optional Module 7 Lab 1: Gene Regulation and Motif Analysis Practical Lab /chIP-seq</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#goal-of-this-practical-lab-3"><i class="fa fa-check"></i>Goal of this practical lab</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#dataset-used-during-this-practical-lab"><i class="fa fa-check"></i>Dataset used during this practical lab</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-1---run-pathway-analysis-using-great"><i class="fa fa-check"></i>Exercise 1 - Run pathway analysis using GREAT</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#perform-pathway-enrichment"><i class="fa fa-check"></i>Perform pathway enrichment</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#explore-the-results."><i class="fa fa-check"></i>Explore the results.</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#perform-pathway-enrichment---proximal-approach"><i class="fa fa-check"></i>Perform pathway enrichment - Proximal approach</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#explore-the-results.---proximal-analysis"><i class="fa fa-check"></i>Explore the results. - proximal analysis</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-2---build-an-enrichment-map-to-visualize-great-results"><i class="fa fa-check"></i>Exercise 2 - Build an enrichment map to visualize GREAT results</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-3-optional-practice-building-enrichment-maps-and-auto-annotation"><i class="fa fa-check"></i>Exercise 3 (optional): Practice building enrichment maps and auto-annotation</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#optional-exercise-3a-autoannotate-the-enrichment-map"><i class="fa fa-check"></i>Optional exercise 3a: AutoAnnotate the enrichment map:</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#optional-exercise-3b-repeat-the-process-of-building-an-enrichment-map-using-the-proximal-data-proximal_gobp_greatexportall.tsv."><i class="fa fa-check"></i>Optional exercise 3b: Repeat the process of building an enrichment map using the proximal data (Proximal_GOBP_greatExportAll.tsv).</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#optional-exercise-3c-repeat-the-process-by-building-both-the-proximal-and-distal-enrichment-maps-at-the-same-time."><i class="fa fa-check"></i>Optional exercise 3c: Repeat the process by building both the Proximal and Distal enrichment maps at the same time.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-4-add-runx1-targets-and-runx1-ko-genes-on-the-distal-enrichment-map."><i class="fa fa-check"></i>Exercise 4: Add RUNX1 targets and RUNX1 KO genes on the distal enrichment map.</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#step-4a-post-analysis"><i class="fa fa-check"></i>step 4a: post analysis:</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#step-4b-optional-change-the-edge-style-of-the-signature-gene-sets"><i class="fa fa-check"></i>Step 4b Optional: Change the edge style of the signature gene-sets:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-5-learning-how-to-run-meme-chip-from-the-meme-suite-httpsmeme-suite.orgmemetoolsmeme-chip"><i class="fa fa-check"></i>Exercise 5: Learning how to run MEME-chip from the MEME suite (https://meme-suite.org/meme/tools/meme-chip)</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#format-the-data"><i class="fa fa-check"></i>Format the Data</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-5a-download-sequences-from-.bed-coordinates"><i class="fa fa-check"></i>Exercise 5a: Download sequences from .bed coordinates</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-5b-run-meme-chip"><i class="fa fa-check"></i>Exercise 5b: Run MEME-chIP</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-6-optional-get-the-iregulon-runx1-targets-and-find-the-mouse-orthologs-using-gorth-from-gprofiler-to-create-the-gmt-file-used-in-exercise-4."><i class="fa fa-check"></i>Exercise 6 (optional): Get the iRegulon RUNX1 targets and find the mouse orthologs using g:Orth (from g:Profiler) to create the gmt file used in Exercise 4.</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#end-of-lab"><i class="fa fa-check"></i>End of Lab</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html"><i class="fa fa-check"></i>Optional Module 7 Lab 2: Gene Regulation and Motif Analysis Practical Lab / iRegulon</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#iregulon-lab"><i class="fa fa-check"></i>iRegulon lab</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#goal-3"><i class="fa fa-check"></i>Goal</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#exercise-1.-detect-regulons-from-co-expressed-genes"><i class="fa fa-check"></i>Exercise 1. Detect regulons from co-expressed genes</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#skills-learned-in-this-exercise"><i class="fa fa-check"></i>Skills learned in this exercise:</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#steps"><i class="fa fa-check"></i>Steps</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#exercise-2.-create-a-metatargetome-using-iregulon-and-merge-2-networks-in-cytoscape."><i class="fa fa-check"></i>Exercise 2. Create a metatargetome using iRegulon and merge 2 networks in Cytoscape.</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#use-our-precomputed-iregulon-results"><i class="fa fa-check"></i>Use our precomputed iRegulon results:</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#notes-about-iregulon"><i class="fa fa-check"></i>Notes about iRegulon:</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#notes-about-cytoscape"><i class="fa fa-check"></i>Notes about Cytoscape:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#exercise-3.-use-enrichr-with-the-prostate-gene-list."><i class="fa fa-check"></i>Exercise 3. Use Enrichr with the prostate gene list.</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#goal-4"><i class="fa fa-check"></i>Goal</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#steps-1"><i class="fa fa-check"></i>Steps</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#end-of-practical-lab"><i class="fa fa-check"></i>end of practical lab</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a  href=""><img src="./images/sponsors_2023.png" width="200"></a><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Pathway and Network Analysis of -Omics Data ( June 2024 )</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="gsea_mod3" class="section level1 hasAnchor">
<h1>Module 3 Lab: GSEA Visualization<a href="gsea_mod3.html#gsea_mod3" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><strong>This work is licensed under a <a href="http://creativecommons.org/licenses/by-sa/3.0/deed.en_US">Creative Commons Attribution-ShareAlike 3.0 Unported License</a>. This means that you are able to copy, share and modify the work, as long as the result is distributed under the same license.</strong></p>
<p><em><font color="#827e9c">By Veronique Voisin, Ruth Isserlin, Gary Bader</font></em></p>
<div id="goal-of-the-exercise-4" class="section level2 hasAnchor">
<h2>Goal of the exercise:<a href="gsea_mod3.html#goal-of-the-exercise-4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Exercise 1 - Create an enrichment map and navigate through the network</strong></p>
<p>During this exercise, you will learn how to create an EnrichmentMap from gene-set enrichment results. The enrichment tool chosen for this exercise is <a href="http://software.broadinstitute.org/gsea/index.jsp">GSEA</a> but an enrichment map can be created from output from <a href="http://software.broadinstitute.org/gsea/index.jsp">GSEA</a>,
<a href="https://biit.cs.ut.ee/gprofiler/gost">g:Profiler</a>,
<a href="http://great.stanford.edu/public/html/">GREAT</a>,
<a href="http://apps.cytoscape.org/apps/bingo">BinGo</a>, <a href="https://amp.pharm.mssm.edu/Enrichr/">Enrichr</a> or alternately from any gene-set tool using the generic enrichment results format.</p>
<p><strong>Exercise 2 - Post analysis (add drug target gene-sets to the network)</strong></p>
<p>As second part of the exercise, you will learn how to expand the network by adding an extra layer of information.</p>
<p><strong>Exercise 3 - Autoannotate</strong></p>
<p>A last optional exercise guides you through the creation of automatically generated cluster labels to the network.</p>
</div>
<div id="data-4" class="section level2 hasAnchor">
<h2>Data<a href="gsea_mod3.html#data-4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The data used in this exercise is gene expression data obtained from high throughput RNA sequencing.
The data correspond to Pancreatic Ductal Adenocarcinoma samples (TCGA-PAAD). We use precomputed results of the GSEA analysis <a href="gsea-lab.html#gsea-lab">Module 2 lab - gsea</a> to create an enrichment map with the aim to transform the tabular format to a network so we can better visualize the relationships between the significant gene-sets:</p>
<p><img src="Module3/gsea/images/workflow.png" alt="workflow" width="750" /></p>
<p>GSEA outpus an entire directory of files and results. For the purpose of this analysis we only need two tables found in the output directory. The output result tables are:</p>
<ul>
<li><p>One table (<em>pos</em>) contains all pathways with an enrichment score (significant or not) related to enrichment of the basal category (positive score). (By default called - gsea_report_for_na_pos_#############.tsv)</p></li>
<li><p>One table (<em>neg</em>) contains all pathways with an enrichment score (significant or not) related to enrichment of the classical category (negative score). (By default called - gsea_report_for_na_neg_#############.tsv)</p></li>
<li><p>These 2 tables are uploaded using the EnrichmentMap App which will create a network of basal and classical pathways that have a significant score (FDR &lt;= 0.05) for clearer visualization of the results.</p></li>
</ul>
<div id="enrichmentmap-1" class="section level3 hasAnchor">
<h3>EnrichmentMap<a href="gsea_mod3.html#enrichmentmap-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><p>A <font color="red">red circle (node)</font> is a pathway specific of the mesenchymal type. (or pathway with mostly positively ranked genes)</p></li>
<li><p>A <font color="blue">blue circle (node)</font> is a pathway specific of the immunoreactive type. (or pathway with mostly negatively ranked genes)</p></li>
<li><p>An edge represents genes in common between 2 pathways (nodes).</p></li>
<li><p>A cluster of nodes represent overlapping and related pathways and may represent a common biological process or theme.</p></li>
<li><p>Clicking on a node will display the genes included in each pathway.</p></li>
</ul>
</div>
</div>
<div id="exercise-1---gsea-output-and-enrichmentmap" class="section level2 hasAnchor">
<h2>Exercise 1 - GSEA output and EnrichmentMap<a href="gsea_mod3.html#exercise-1---gsea-output-and-enrichmentmap" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To start the lab practical section, first download the files.</p>
<div class="rmd-datadownload">
<p>
Right click on link below and select Save Link As.
</p>
<p>
Place it in the corresponding module directory of your CBW work
directory.
</p>
</div>
<p>7 Files are needed to create the enrichment map for this exercise (please download these files on your computer or alternately use the GSEA directory created in <a href="gsea-lab.html#gsea-lab">module 2 lab - gsea</a> for files 1,2,3) :</p>
<ol style="list-style-type: decimal">
<li><p>GMT (file containing all pathways and corresponding genes) - <a href="./Module3/gsea/data/Human_GOBP_AllPathways_noPFOCR_no_GO_iea_June_01_2024_symbol.gmt">Human_GOBP_AllPathways_noPFOCR_no_GO_iea_June_01_2024_symbol.gmt</a></p></li>
<li><p>Enrichments 1 (GSEA results for the pos basal subtype) - <a href="./Module3/gsea/data/gsea_report_for_na_pos_1717773429384.tsv">gsea_report_for_na_pos_1717773429384.tsv</a></p></li>
<li><p>Enrichments 2 (GSEA results for the neg Classical subtype) - <a href="./Module3/gsea/data/gsea_report_for_na_neg_1717773429384.tsv">gsea_report_for_na_neg_1717773429384.tsv</a></p></li>
<li><p>Expression (file containing the RNAseq data for all samples and all genes) - <a href="./Module3/gsea/data/TCGA-PAAD_GDC_BasalvsClassical_normalized_rnaseq.txt">TCGA-PAAD_GDC_BasalvsClassical_normalized_rnaseq.txt</a></p></li>
<li><p>Rank file (file that has been used as input to GSEA) - <a href="./Module3/gsea/data/TCGA-PAAD_GDC_Subtype_Moffitt_BasalvsClassical_ranks.rnk">TCGA-PAAD_GDC_Subtype_Moffitt_BasalvsClassical_ranks.rnk</a></p></li>
<li><p>Classes (define which samples are basal and which samples are classical) - <a href="./Module3/gsea/data/TCGA-PAAD_Subtype_Moffitt_BasalvsClassical_RNAseq_classes.cls">TCGA-PAAD_Subtype_Moffitt_BasalvsClassical_RNAseq_classes.cls</a></p></li>
<li><p>Drug target database (preselection of 7 drugs and their target genes in the post analysis exercise, <a href="http://www.drugbank.ca/" class="uri">http://www.drugbank.ca/</a>) - <a href="./Module3/gsea/data/Human_DrugBank_all_symbol_June_01_2024_selected.gmt">Human_DrugBank_all_symbol_June_01_2024_selected.gmt</a></p></li>
</ol>
<p>Follow the steps described below at your own pace:</p>
<div id="step-1-1" class="section level3 hasAnchor">
<h3>Step 1<a href="gsea_mod3.html#step-1-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Launch Cytoscape and open EnrichmentMap App</p>
<p><strong>1a</strong>. Double click on the Cytoscape icon</p>
<p><strong>1b</strong>. Open EnrichmentMap App</p>
<ul>
<li><p>In the top menu bar:</p>
<ul>
<li>Click on Apps -&gt; EnrichmentMap</li>
</ul></li>
</ul>
<p><img src="Module3/gsea/images/EM1.png"  /></p>
<p>A Create EnrichmentMap window is now opened.</p>
</div>
<div id="step-2-1" class="section level3 hasAnchor">
<h3>Step 2<a href="gsea_mod3.html#step-2-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Create an enrichment map</p>
<p><strong>2a</strong>. In the Create EnrichmentMap window, add a dataset of the GSEA type by clicking on the +ADD &gt; + add data set manually.</p>
<p><img src="Module3/gsea/images/EMinput.png" width="750" /></p>
<p><strong>2b</strong>. Specify the following parameters and upload the specified files:</p>
<ul>
<li><p><em>Name</em>: leave default or a name of your choice like GSEAmapPAAD_Basal_vs_Classical</p></li>
<li><p><em>Analysis Type</em>: GSEA</p></li>
<li><p><em>Enrichments Pos</em>: gsea_report_for_na_pos_1717773429384.tsv</p></li>
<li><p><em>Enrichments Neg</em>: gsea_report_for_na_neg_1717773429384.tsv</p></li>
<li><p><em>GMT</em> : Human_GOBP_AllPathways_noPFOCR_no_GO_iea_June_01_2024_symbol.gmt</p></li>
<li><p><em>Ranks</em>: TCGA-PAAD_GDC_Subtype_Moffitt_BasalvsClassical_ranks.rnk</p></li>
<li><p><em>Expressions</em> : TCGA-PAAD_GDC_BasalvsClassical_normalized_rnaseq.txt</p>
<div class="rmd-tip">
<p>
This field is optional but recommended.
</p>
</div></li>
<li><p><em>Classes</em>: TCGA-PAAD_Subtype_Moffitt_BasalvsClassical_RNAseq_classes.cls</p>
<div class="rmd-tip">
<p>
This field is optional.
</p>
</div></li>
<li><p><em>Phenotypes</em>: In the text boxes place <em>Basal</em> as the Positive phenotype <em>Classical</em> as the Negative phenotype. Basal will be associated with red nodes because it corresponds to the positive phenotype and Classical will be associated with the blue nodes because it corresponds to the negative phenotype.</p></li>
<li><p>Set FDR q-value cutoff to 0.05 (= only gene-sets significantly enriched at a value of 0.05 or less will be displayed on the map).</p>
<div class="rmd-tip">
<p>
If the cutoff is set to a very small number, for exaxmple 0.0001, it
will be displayed as 1E-04 in the scientific notation.
</p>
</div></li>
</ul>
<p><strong>2c</strong>. Click on <em>Build</em></p>
<p><img src="Module3/gsea/images/EMinput2.png" alt="EM" width="750" /></p>
<div class="rmd-caution">
<p>
We populated the fields manually. If you work with your own data, a
way to populate automatically the fields is to drag and drop your GSEA
folder in the Data Set window. You are encouraged to give it a try
once you finished the lab with your own GSEA results.
</p>
</div>
<p><strong>Unformatted results</strong>:</p>
<div class="rmd-note">
<p>
layout will be different for each user (there is a random seed in the
layout algorithm) but it does not change the results or interpretation
(the connections are the same, only the display is different).
</p>
</div>
<p><img src="Module3/gsea/images/EM3.png" alt="EM" width="750" /></p>
</div>
<div id="step-3" class="section level3 hasAnchor">
<h3>Step 3<a href="gsea_mod3.html#step-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Navigate the enrichment map to gain a better understanding of a EnrichmentMap network.</p>
<p>General layout of Cytoscape panel: In addition to the main window where the network is displayed, there are 2 panels: the Control Panel on the left side and the Table Panel at the bottom of the window.</p>
<p>Steps:</p>
<p><strong>3a</strong>. In the Cytocape menu bar, select <em>View</em> and <em>Always Show Graphic details</em>. It will turn the squared nodes into circles and the gene-set labels will be visible.</p>
<p><img src="Module3/gsea/images/EM5.png" alt="EM" width="350" /></p>
<p><strong>3b</strong>: Zoom in or out using + or - in toolbar or scroll button on mouse until you are able to read the labels comfortably.</p>
<p><img src="Module3/gsea/images/EM6.png" alt="EM" width="350" /></p>
<p><strong>3c</strong>: Use the birds eye view (located at the bottom of the control panel) to navigate around the network by moving the blue rectangle using the mouse or trackpad.</p>
<p><img src="Module3/gsea/images/EM7.png" alt="EM" width="750" /></p>
<p><strong>3d</strong>: Click on an individual node of interest.</p>
<p>For this example, you could use <em>TGF-BETA RECEPTOR SIGNALING ACTIVATES SMADS</em>.</p>
<div class="rmd-tip">
<p>
If you are unable to locate <em>TGF-BETA RECEPTOR SIGNALING ACTIVATES
SMADS</em>, type TGF-BETA RECEPTOR SIGNALING ACTIVATES SMADS in the
search box (quotes are important). Selected nodes appear yellow (or
highlighted) in the network.
</p>
</div>
<p><strong>3e</strong>. In the Table Panel in the <em>EM Heat map</em> tab change:</p>
<ul>
<li><p>Expressions: <em>Row Norm</em></p></li>
<li><p>Compress: <em>-None-</em></p></li>
</ul>
<p><img src="Module3/gsea/images/EM8.png" alt="EM" width="750" /></p>
<div class="rmd-tip">
<p>
Genes in the heatmap that are highlighted yellow (rank column)
represent genes that are part of the leading edge for this gene set,
i.e.contributed the most to the enriched phenotype.<br> Leading edge
genes will only be highlighted if an individual node has been selected
and the Enrichment Map was created from GSEA results.<br><br>
<em>Troubleshooting</em>: if you dont see the sort column highlighted
in yellow, reselect the node of interest and click on the GSEARanking
Data Set 1 text in the EM Heatmap tab.
</p>
</div>
</div>
<div id="step-4" class="section level3 hasAnchor">
<h3>Step 4<a href="gsea_mod3.html#step-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Use Filters to automatically select nodes on the map: Move the blue nodes to the left side of the window and the red nodes to the right side of the window.</p>
<p><strong>4a</strong>. Locate the <em>Filter</em> tab on the side bar of the <em>Control Panel</em>.</p>
<p><strong>4b</strong>. Click on the + sign to view the menu and select <em>Column Filter</em>.</p>
<p><strong>4c</strong>. From the <em>Choose column </em> box, select <em>Node: NES(PAAD_Basal_vs_Classical)</em> and set filter values from -2.242 and 0 inclusive.</p>
<p><strong>4d</strong>. The blue nodes are now automatically selected. Zoom out to be able to look at the entire network and drag all blue nodes to the left side of the screen.</p>
<p><img src="Module3/gsea/images/EM9.png" alt="EM" width="750" /></p>
<p><strong>4e</strong>. Optional. Change <em>is</em> to <em>is not</em> to select the red nodes.</p>
<p><img src="Module3/gsea/images/EM10.png" alt="EM" width="750" /></p>
<div class="rmd-note">
<p>
The red pathways (nodes) are specific to the Basal subtype. They were
listed in the <em>pos</em> table of the GSEA results. The enrichment
score (ES) values in this table are all positive values.<br>
</p>
<p>
The blue pathways are specific to the Classical subtype and were
listed in the <em>neg</em> table of the GSEA results. The ES values in
this table are all negative values. <br><br>
</p>
<p>
This is the information we used as the filtering criteria.
</p>
</div>
</div>
</div>
<div id="exercise-2---post-analysis-add-drug-target-gene-sets-to-the-network" class="section level2 hasAnchor">
<h2>Exercise 2 - Post analysis (add drug target gene-sets to the network)<a href="gsea_mod3.html#exercise-2---post-analysis-add-drug-target-gene-sets-to-the-network" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="step-5" class="section level3 hasAnchor">
<h3>Step 5<a href="gsea_mod3.html#step-5" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Add drug target gene-sets to the network (Add Signature Gene-Sets).</p>
<p><strong>5a</strong>. In Control Panel, go to the EnrichmentMap tab and click on Options located above the Data Sets: box. Select Add Signature Gene Sets. A window named EnrichmentMap: Add Signature Gene Sets (Post-Analysis) is now opened.</p>
<p><img src="Module3/gsea/images/drug1.png" alt="EM" width="750" /></p>
<p><strong>5b</strong>. Using the Load from File button, select the <em>Human_DrugBank_approved_symbol_June_01_2024_selected.gmt</em> file that you saved on your computer.</p>
<p><img src="Module3/gsea/images/drug2.png" alt="EM" width="75%" /></p>
<p><img src="Module3/gsea/images/drug3.png" alt="EM" width="75%" /></p>
<p><strong>5c</strong>. Click on Finish.</p>
<div class="rmd-note">
<p>
Two additional nodes are now added to the network and visible as grey
diamonds.
</p>
<p>
Dotted orange edges represent their overlap with the nodes of our
network.
</p>
<p>
These additional nodes represent gene targets of some approved drugs
and these genes are either specific of the basal type (dotted orange
edges connected to red nodes) or specific of the classical type (dotted
orange edges connected to blue nodes).
</p>
<p>
The remaining five drugs that do not pass the threshold in this map
are other drugs currently used in treatment of pancreatic cancer.
</p>
</div>
<p><img src="Module3/gsea/images/drug4.png" alt="EM" width="750" /></p>
<div class="rmd-tip">
<p>
more info using this link:
<a href="https://enrichmentmap.readthedocs.io/en/latest/PostAnalysis.html" class="uri">https://enrichmentmap.readthedocs.io/en/latest/PostAnalysis.html</a>
</p>
</div>
</div>
</div>
<div id="exercise-3---autoannotate-the-network" class="section level2 hasAnchor">
<h2>Exercise 3 - Autoannotate the Network<a href="gsea_mod3.html#exercise-3---autoannotate-the-network" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="step-6" class="section level3 hasAnchor">
<h3>Step 6<a href="gsea_mod3.html#step-6" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>By default, Enrichment map will Auto-annotate the network with cluster labels.</p>
<div class="rmd-note">
<p>
The Apps WordCloud, ClusterMaker and Autoannotate have to be
installed. (they should have been installed during the pre-workshop set
up)
</p>
</div>
<div class="rmd-note">
<p>
if you ran step 5, <br>
</p>
<p>
<strong>delete the drug targets diamond nodes and associated edge
before performing step 6</strong>:<br> * select the 4 nodes and
associated dotted orange edges by browsing the mouse and <br> * click
delete on your keyboard or<br> * in the Cytoscape menu, Edit,
Delete Selected Nodes and Edges.<br><br>
</p>
<p>
<strong>Alternately, in the Enrichment Map Input Panel in the
Datasets box, un-select
Human_Drugbank_approved_symbol_June_01_2024_selected to hide the post
analysis nodes.</strong>
</p>
</div>
<p>The annotations are shown as larger node labels.</p>
<p><img src="Module3/gsea/images/EM13.png" alt="EM" width="750" /></p>
<p><strong>6a</strong>. To modify these precomputed annotations find the Auto annotate Display panel on the right or Auto annotate input panel left.</p>
<p><img src="Module3/gsea/images/EM12.png" alt="EM" width="750" /></p>
<p>Unhide labels and shapes to see the underlying annotation for the network.</p>
<p><img src="Module3/gsea/images/EM14.png" alt="EM" width="750" /></p>
<div class="rmd-note">
<p>
The network is now subdivided into clusters that are represented by
ellipses. Each of these clusters are composed of pathways (nodes)
interconnected by many common genes. These pathways represent similar
biological processes. The app WordCloud take all the labels of the
pathways in one cluster and summarize them as a unique cluster label
displayed at the top of each ellipse.
</p>
</div>
<div class="rmd-tip">
<p>
<strong>Tip 1</strong>: further editing and formatting can be
performed on the AutoAnnote results using the <em>AutoAnnotate
Display</em> in the <em>Results Panels</em> located at the right side of
the window.<br> For example, it is possible to change Ellipse to
Rectangle, uncheck <em>Scale font by cluster size</em> and increase the
<em>Font Scale</em> using the scaling bar. It is also possible to reduce
the length of the cluster label by checking the Word Wrap option.
</p>
<p>
<strong>Tip 2</strong>: The AutoAnnotate window on the left side in
Result Panel contains the list of all clusters. Clicking on a cluster
label will highlight in yellow all nodes in this cluster. It is then
easy to move the nodes using the mouse to avoid cluster overlaps.
</p>
</div>
</div>
</div>
<div id="exercise-4-optional---explore-results-in-genemania-or-string" class="section level2 hasAnchor">
<h2>Exercise 4 (Optional) - Explore results in GeneMANIA or STRING<a href="gsea_mod3.html#exercise-4-optional---explore-results-in-genemania-or-string" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Each node in the Enrichment map represents a biological process or pathway. It consists of a collection of genes. Often we want to know how the genes in that group interact. There are many different ways you can investigate the underlying interactions for the given group. Some involve searching online databases and others are directly integrated into cytoscape.</p>
<ul>
<li><a href="https://genemania.org/">GeneMANIA</a> - an integrative database of gene connections including co-expression, protein interactions, genetic interactions, pathways and more. <strong>Cytoscape App</strong></li>
<li><a href="https://string-db.org/">String</a> - an integrative database of gene connections including co-expression, protein interactions, genetic interactions, pathways and more. <strong>Cytoscape App</strong></li>
<li><a href="https://www.pathwaycommons.org/">Pathway Commons</a> - a intergrative database of pathways. (There is a beta feature in EM to show your pathway in the painter app, a pathway common web page that overlays your expression data on the given pathway. Still in beta testing and requires expression data to work correctly so wont work for this example)</li>
</ul>
<div id="step-7" class="section level3 hasAnchor">
<h3>Step 7<a href="gsea_mod3.html#step-7" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Visualize genes in a pathway/node of interest using the apps STRING and GeneMANIA. This will create a protein-protein interaction network using the genes included in the pathway. Note: We will go more in depth into <a href="genemania_cytoscape.html#genemania_cytoscape">GeneMANIA in module 5</a></p>
<p><strong>7a</strong>: Click on an individual node of interest.</p>
<p>For this example, you could use <em>cellular response to BMP</em>.</p>
<div class="rmd-tip">
<p>
If you are unable to locate <em>cellular response to BMP</em>, type
cellular response to BMP in the search box (quotes are important). The
selected node appears yellow (or highlighted) in the network. If you
have annotated your network, it should be included in the <em>restricted
smad bmp</em> cluster.
</p>
</div>
<p><strong>7b</strong>: Right Click on the node of interest to diplay the option menu. Select <em>Apps</em>,&gt; <em>EnrichmentMap - Show in STRING</em><br></p>
<p><img src="Module3/gsea/images/Stringa.png" alt="workflow" width="500" /></p>
<div class="rmd-tip">
<p>
Patience. :) . It might take a few seconds for the <em>String Protein
Query</em> window to open.
</p>
</div>
<ul>
<li>A <em>STRING Protein Query</em> box appears.</li>
<li>Select <em>Select genes with expression</em>.</li>
<li>Click on <em>OK</em>.</li>
</ul>
<p><img src="Module3/gsea/images/Stringb.png" alt="workflow" width="500" /></p>
<ul>
<li>The resulting network will look something like this.</li>
</ul>
<p><img src="Module3/gsea/images/Stringc.png" alt="workflow" width="750" /></p>
<div class="rmd-question">
<p>
Explore the features and data of each Cytoscape app.<br>What happens
to the network if you change the initial parameters like <em>Confidence
cutoff</em> or <em>Max Additional interactors</em> <br><br>
</p>
<p>
<img src="Module3/gsea/images/String_additional_parameters.png" alt="workflow" width="750" />
</p>
</div>
<p><strong>7c</strong>:Go back to enrichment map network.</p>
<ul>
<li>In Control Panel (left side of the window), select the Network tab and click on the Enrichment Map network as shown in below screenshot.</li>
</ul>
<p><img src="Module3/gsea/images/find_network.png" alt="workflow" width="500" /></p>
<p><strong>7d</strong>: Search again for the node labelled <em>cellular response to BMP</em> as in Step 7a.</p>
<ul>
<li>Right Click on the node of interest to diplay the option menu. Select <em>Apps</em>,&gt; <em>EnrichmentMap - Show in GeneMANIA</em><br></li>
</ul>
<p><img src="Module3/gsea/images/Stringa-genemania.png" alt="workflow" width="500" /></p>
<ul>
<li>A <em>GeneMANIA Query</em> box appears.</li>
<li>select <em>Select genes with expression</em>.</li>
<li>Click on <em>OK</em>.</li>
</ul>
<p><img src="Module3/gsea/images/Genemaniab.png" alt="workflow" width="500" /></p>
<ul>
<li>A pop up will appear indicating that it is currenlty querying GeneMANIA</li>
</ul>
<p><img src="Module3/gsea/images/genemaniaquery.png" alt="workflow" width="500" /></p>
<ul>
<li>The resulting network will look similiar to the below screenshot.</li>
</ul>
<p><img src="Module3/gsea/images/Genemaniac.png" alt="workflow" width="750" /></p>
<div class="rmd-tip">
<p>
It is possible to view gene expression data for the nodes in the
STRING network. See the section
<a href="https://enrichmentmap.readthedocs.io/en/latest/Integration.html" class="uri">https://enrichmentmap.readthedocs.io/en/latest/Integration.html</a> and try
it out after the workshop.
</p>
</div>
<div class="rmd-caution">
<p>
SAVE YOUR SESSION FILE!
</p>
</div>
<hr />
</div>
</div>
<div id="bonus---automation.-3" class="section level2 hasAnchor">
<h2>Bonus - Automation.<a href="gsea_mod3.html#bonus---automation.-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Run analysis directly from R for easy integration into existing pipelines.</p>
<div class="rmd-bonus">
<p>
Instead of creating an Enrichment map manually through the user
interface you can create an enrichment map directly using the <a
href="https://www.bioconductor.org/packages/release/bioc/html/RCy3.html">RCy3
bioconductor package</a> or through direct rest calls with <a
href="https://apps.cytoscape.org/apps/cyrest">Cytoscape cyrest</a>.
</p>
<p>
Follow the step by step instructions on how to run from R here -
<a href="https://risserlin.github.io/CBW_pathways_workshop_R_notebooks/create-enrichment-map-from-r-with-gsea-results.html" class="uri">https://risserlin.github.io/CBW_pathways_workshop_R_notebooks/create-enrichment-map-from-r-with-gsea-results.html</a>
</p>
<p>
First, make sure your environment is set up correctly by following
there instructions -
<a href="https://risserlin.github.io/CBW_pathways_workshop_R_notebooks/setup.html" class="uri">https://risserlin.github.io/CBW_pathways_workshop_R_notebooks/setup.html</a>
</p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="gprofiler_mod3.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="automation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": "https://github.com/BaderLab/CBW_Pathways_2023/commit/master/3.2-Module3lab_gsea.Rmd",
"text": null
},
"view": {
"link": "https://github.com/BaderLab/CBW_Pathways_2023/master/3.2-Module3lab_gsea.Rmd",
"text": null
},
"download": ["CBW_Pathways.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true
}
});
});
</script>

</body>

</html>
