<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Module 08 -1 lab - scRNA pathway analysis using g:Profiler {#scRNA gprofiler-lab} | Pathway and Network Analysis of -Omics Data ( June 2023 )</title>
  <meta name="description" content="Course covers the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information." />
  <meta name="generator" content="bookdown 0.33 and GitBook 2.6.7" />

  <meta property="og:title" content="Module 08 -1 lab - scRNA pathway analysis using g:Profiler {#scRNA gprofiler-lab} | Pathway and Network Analysis of -Omics Data ( June 2023 )" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Course covers the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Module 08 -1 lab - scRNA pathway analysis using g:Profiler {#scRNA gprofiler-lab} | Pathway and Network Analysis of -Omics Data ( June 2023 )" />
  
  <meta name="twitter:description" content="Course covers the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information." />
  

<meta name="author" content="Gary Bader, Ruth Isserlin, Chaitra Sarathy, Veronique Voisin" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/favicon.ico" type="image/x-icon" />
<link rel="prev" href="ass_automation.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://bioinformatics.ca/workshops/current-workshops/"><img src="./images/bioinfologo.png" width="280"></a><a href="./">CBW Pathways Workshop</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Canadian Bioinformatics Workshops</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html"><i class="fa fa-check"></i>Welcome <a id="welcome"></a></a>
<ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#meet-your-faculty"><i class="fa fa-check"></i>Meet your Faculty</a>
<ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#gary-bader"><i class="fa fa-check"></i>Gary Bader</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#lincoln-stein"><i class="fa fa-check"></i>Lincoln Stein</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#veronique-voisin"><i class="fa fa-check"></i>Veronique Voisin</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#ruth-isserlin"><i class="fa fa-check"></i>Ruth Isserlin</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#chaitra-sarathy-phd"><i class="fa fa-check"></i>Chaitra Sarathy, PhD</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#michelle-brazas-phd"><i class="fa fa-check"></i>Michelle Brazas, PhD</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#class-materials"><i class="fa fa-check"></i>Class Materials</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#schedule"><i class="fa fa-check"></i>Workshop Schedule</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#pre-workshop"><i class="fa fa-check"></i>Pre-Workshop Materials and Laptop Setup Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#laptop-setup-instructions"><i class="fa fa-check"></i>Laptop Setup Instructions</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#basic-programs"><i class="fa fa-check"></i>Basic programs</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#cytoscape-installation"><i class="fa fa-check"></i>Cytoscape Installation</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#gsea-installation"><i class="fa fa-check"></i>GSEA Installation</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#pre-workshop-tutorials"><i class="fa fa-check"></i>Pre-workshop Tutorials</a>
<ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#cytoscape-preparation-tutorials"><i class="fa fa-check"></i>Cytoscape Preparation tutorials</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#pre-workshop-readings-and-lectures"><i class="fa fa-check"></i>Pre-workshop Readings and Lectures</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#additional-tutorials"><i class="fa fa-check"></i>Additional tutorials <a id="additionaltutorials"></a></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i>Module 1 - Introduction to Pathway and Network Analysis (Gary Bader)</a></li>
<li class="chapter" data-level="" data-path="module-2-finding-over-represented-pathways-veronique-voisin.html"><a href="module-2-finding-over-represented-pathways-veronique-voisin.html"><i class="fa fa-check"></i>Module 2: Finding Over-represented Pathways (Veronique Voisin)</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html"><i class="fa fa-check"></i>Module 2 lab - g:Profiler</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#introduction"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#goal-of-the-exercise-1"><i class="fa fa-check"></i>Goal of the exercise 1</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#data"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#exercise-1"><i class="fa fa-check"></i>Exercise 1 - run g:Profiler</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-1---launch-gprofiler."><i class="fa fa-check"></i>Step 1 - Launch g:Profiler.</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-2---input-query"><i class="fa fa-check"></i>Step 2 - input query</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-3---adjust-parameters."><i class="fa fa-check"></i>Step 3 - Adjust parameters.</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-4---run-query"><i class="fa fa-check"></i>Step 4 - Run query</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-5---explore-the-results."><i class="fa fa-check"></i>Step 5 - Explore the results.</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-6-expand-the-stats-tab"><i class="fa fa-check"></i>Step 6: Expand the stats tab</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-7-save-the-results"><i class="fa fa-check"></i>Step 7: Save the results</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-8-optional-but-recommended"><i class="fa fa-check"></i>Step 8 (Optional but recommended)</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-9-optional-by-recommended"><i class="fa fa-check"></i>Step 9 (Optional by recommended)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#exercise-2-load-and-use-a-custom-.gmt-file-and-run-the-query"><i class="fa fa-check"></i>Exercise 2: Load and use a custom .gmt file and run the query</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#optional-steps"><i class="fa fa-check"></i>Optional steps</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#optional-1"><i class="fa fa-check"></i><strong>Optional 1</strong>:</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#option-2"><i class="fa fa-check"></i><strong>Option 2</strong>:</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#option-3"><i class="fa fa-check"></i><strong>Option 3</strong> :</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#bonus---automation."><i class="fa fa-check"></i>Bonus - Automation.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html"><i class="fa fa-check"></i>Module 2 lab - GSEA</a>
<ul>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#introduction-1"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#goal-of-the-exercise"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#data-1"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#background"><i class="fa fa-check"></i>Background</a>
<ul>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#how-to-generate-a-rank-file."><i class="fa fa-check"></i>How to generate a rank file.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#start-the-exercise"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step1."><i class="fa fa-check"></i>Step1.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-2."><i class="fa fa-check"></i>Step 2.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step3."><i class="fa fa-check"></i>Step3.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-4."><i class="fa fa-check"></i>Step 4.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-5."><i class="fa fa-check"></i>Step 5.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#additional_information"><i class="fa fa-check"></i>Additional information</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#bonus---automation.-1"><i class="fa fa-check"></i>Bonus - Automation.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-3-network-visualization-and-analysis-with-cytoscape.html"><a href="module-3-network-visualization-and-analysis-with-cytoscape.html"><i class="fa fa-check"></i>Module 3: Network Visualization and Analysis with Cytoscape</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html"><i class="fa fa-check"></i>Module 3 Lab: g:profiler Visualization</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#goal-of-the-exercise-2"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#data-2"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#enrichmentmap"><i class="fa fa-check"></i>EnrichmentMap</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#description-of-this-exercise"><i class="fa fa-check"></i>Description of this exercise</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#start-the-exercise-1"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#exercise-1a---compare-different-gprofiler-geneset-size-results"><i class="fa fa-check"></i>Exercise 1a - compare different gprofiler geneset size results</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#step-1"><i class="fa fa-check"></i>Step 1</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#step-2"><i class="fa fa-check"></i>Step 2</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#step3-explore-the-results"><i class="fa fa-check"></i>Step3: Explore the results:</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#explore-detailed-results"><i class="fa fa-check"></i>Explore Detailed results</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#exercise-1b---is-specifying-the-gmt-file-important"><i class="fa fa-check"></i>Exercise 1b - Is specifying the gmt file important?</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#exercise-1c---create-em-from-results-using-baderlab-genesets"><i class="fa fa-check"></i>Exercise 1c - create EM from results using Baderlab genesets</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#exercise-1c-optional---investigate-individual-pathways-in-genemania-or-string"><i class="fa fa-check"></i>Exercise 1c (optional) - investigate individual pathways in GeneMANIA or String</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#genemania"><i class="fa fa-check"></i>GeneMANIA</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#string"><i class="fa fa-check"></i>String</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#bonus---automation.-2"><i class="fa fa-check"></i>Bonus - Automation.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html"><i class="fa fa-check"></i>Module 3 Lab: GSEA Visualization</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#goal-of-the-exercise-3"><i class="fa fa-check"></i>Goal of the exercise:</a></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#data-3"><i class="fa fa-check"></i>Data</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#enrichmentmap-1"><i class="fa fa-check"></i>EnrichmentMap</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#exercise-1---gsea-output-and-enrichmentmap"><i class="fa fa-check"></i>Exercise 1 - GSEA output and EnrichmentMap</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-1-1"><i class="fa fa-check"></i>Step 1</a></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-2-1"><i class="fa fa-check"></i>Step 2</a></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-3"><i class="fa fa-check"></i>Step 3</a></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-4"><i class="fa fa-check"></i>Step 4</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#exercise-2---post-analysis-add-drug-target-gene-sets-to-the-network"><i class="fa fa-check"></i>Exercise 2 - Post analysis (add drug target gene-sets to the network)</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-5"><i class="fa fa-check"></i>Step 5</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#exercise-3---autoannotate-the-network"><i class="fa fa-check"></i>Exercise 3 - Autoannotate the Network</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-6"><i class="fa fa-check"></i>Step 6</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#exercise-4-optional---explore-results-in-genemania-or-string"><i class="fa fa-check"></i>Exercise 4 (Optional) - Explore results in GeneMANIA or STRING</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-7"><i class="fa fa-check"></i>Step 7</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#bonus---automation.-3"><i class="fa fa-check"></i>Bonus - Automation.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html"><i class="fa fa-check"></i>Module 3 Lab: (Bonus) Automation</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#goal-of-the-exercise-4"><i class="fa fa-check"></i>Goal of the exercise:</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#set-up---option-1---install-rrstudio"><i class="fa fa-check"></i>Set Up - Option 1 - Install R/Rstudio</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#set-up---option-2---docker-image-with-rrstudio"><i class="fa fa-check"></i>Set Up - Option 2 - Docker image with R/Rstudio</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#what-is-docker"><i class="fa fa-check"></i>What is docker?</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#docker---basic-term-definition"><i class="fa fa-check"></i>Docker - Basic term definition</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#container"><i class="fa fa-check"></i>Container</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#image"><i class="fa fa-check"></i>Image</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#docker-volumes"><i class="fa fa-check"></i>Docker Volumes</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#r_docker"><i class="fa fa-check"></i>Install Docker</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#windows"><i class="fa fa-check"></i>Windows</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#macos-linux"><i class="fa fa-check"></i>MacOS / Linux</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#create-your-first-notebook-using-docker"><i class="fa fa-check"></i>Create your first notebook using Docker</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#start-coding"><i class="fa fa-check"></i>Start coding!</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#start-using-automation"><i class="fa fa-check"></i>Start using automation</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-1---launch-rstudio"><i class="fa fa-check"></i>Step 1 - launch RStudio</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-2---create-a-new-project"><i class="fa fa-check"></i>Step 2 - create a new project</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-3---open-example-up-rnotebook"><i class="fa fa-check"></i>Step 3 - Open example up RNotebook</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-4---define-notebook-parameters"><i class="fa fa-check"></i>Step 4 - Define Notebook parameters</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-5---step-through-notebook-to-run-the-analysis"><i class="fa fa-check"></i>Step 5 - Step through notebook to run the analysis</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#exercises"><i class="fa fa-check"></i>Exercises</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#additional-resources"><i class="fa fa-check"></i>Additional resources</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-4-in-depth-analysis-of-networks-and-pathways.html"><a href="module-4-in-depth-analysis-of-networks-and-pathways.html"><i class="fa fa-check"></i>Module 4: In-depth Analysis of Networks and Pathways</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html"><i class="fa fa-check"></i>Module 4 Lab: ReactomeFI</a>
<ul>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#goal-of-this-practical-lab"><i class="fa fa-check"></i>Goal of this practical lab</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#data-download-the-following-files-on-your-computer-before-starting-the-practical-lab."><i class="fa fa-check"></i>Data: download the following files on your computer before starting the practical lab.</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#exercise-1-use-the-reactome-functional-interaction-fi-network"><i class="fa fa-check"></i>Exercise 1: Use the Reactome Functional Interaction (FI) Network</a>
<ul>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#question-1-describe-the-size-and-composition-of-the-network"><i class="fa fa-check"></i>Question 1: Describe the size and composition of the network?</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#question-2-after-clustering-how-many-modules-are-there"><i class="fa fa-check"></i>Question 2: After clustering, how many modules are there?</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#query-information-about-the-interaction-between-2-genes"><i class="fa fa-check"></i>Query information about the interaction between 2 genes:</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#question-3-what-are-the-most-significant-pathways-in-each-module"><i class="fa fa-check"></i>Question 3: What are the most significant pathways in each module?</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#set-the-size-of-the-nodes-proportional-to-the-mutation-frequencies-in-each-cancer"><i class="fa fa-check"></i>Set the size of the nodes proportional to the mutation frequencies in each cancer</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#play-around-with-the-styles-change-transparency-and-colors"><i class="fa fa-check"></i>Play around with the styles: change transparency and colors</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#create-a-pie-chart"><i class="fa fa-check"></i>Create a pie chart</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#create-a-subnetwork"><i class="fa fa-check"></i>Create a subnetwork</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#fetch-cancer-drugs-on-the-created-subnetwork"><i class="fa fa-check"></i>Fetch Cancer drugs on the created subnetwork</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#save-the-network-as-an-image-for-publication"><i class="fa fa-check"></i>Save the network as an image for publication</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#exercise-2a-explore-reactome-pathways"><i class="fa fa-check"></i>Exercise 2a: Explore Reactome Pathways</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#exercise-2b-pathway-enrichment-analysis-using-a-simple-gene-list"><i class="fa fa-check"></i>Exercise 2b: Pathway enrichment analysis using a simple gene list</a>
<ul>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#question-1-what-are-the-most-significant-biological-pathways-based-on-the-fdr"><i class="fa fa-check"></i>Question 1: What are the most significant biological pathways based on the FDR?</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#answer-to-question-1"><i class="fa fa-check"></i>Answer to Question 1</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#exercise-2c-pathway-based-analysis-using-a-rank-gene-list-gsea"><i class="fa fa-check"></i>Exercise 2c: Pathway-based analysis using a rank gene list (GSEA)</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#exercise-3-perform-scrna-seq-data-analysis-and-visualization"><i class="fa fa-check"></i>Exercise 3: Perform scRNA-seq Data Analysis and Visualization</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#exercise-4-automation-using-a-jupiter-notebook-advanced-users"><i class="fa fa-check"></i>Exercise 4: Automation using a Jupiter Notebook (advanced users)</a></li>
<li class="chapter" data-level="" data-path="ReactomeFI.html"><a href="ReactomeFI.html#reference-guide-bonus-exercises"><i class="fa fa-check"></i>Reference guide /bonus exercises:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html"><i class="fa fa-check"></i>Module 5: Regulatory Network Analysis</a>
<ul>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html#lecture"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html#practical-lab-1-chip_seq-data---great-and-meme-chip"><i class="fa fa-check"></i>Practical lab 1: chIP_seq data - GREAT and MEME-chIP</a></li>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html#practical-lab-2-gene-list---iregulon-and-enrichrenrichmentmap"><i class="fa fa-check"></i>Practical lab 2: gene list - iREgulon and enrichr/EnrichmentMap</a></li>
<li class="chapter" data-level="" data-path="intro-regulatory-networks.html"><a href="intro-regulatory-networks.html#additional-slides-about-the-tools-segway-and-behst-presented-during-the-lecture"><i class="fa fa-check"></i>Additional slides about the tools Segway and BEHST presented during the lecture</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html"><i class="fa fa-check"></i>Module 5 Lab 1: Gene Regulation and Motif Analysis Practical Lab /chIP-seq</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#goal-of-this-practical-lab-1"><i class="fa fa-check"></i>Goal of this practical lab</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#dataset-used-during-this-practical-lab"><i class="fa fa-check"></i>Dataset used during this practical lab</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-1---run-pathway-analysis-using-great"><i class="fa fa-check"></i>Exercise 1 - Run pathway analysis using GREAT</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#perform-pathway-enrichment"><i class="fa fa-check"></i>Perform pathway enrichment</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#explore-the-results."><i class="fa fa-check"></i>Explore the results.</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#perform-pathway-enrichment---proximal-approach"><i class="fa fa-check"></i>Perform pathway enrichment - Proximal approach</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#explore-the-results.---proximal-analysis"><i class="fa fa-check"></i>Explore the results. - proximal analysis</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-2---build-an-enrichment-map-to-visualize-great-results"><i class="fa fa-check"></i>Exercise 2 - Build an enrichment map to visualize GREAT results</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-3-optional-practice-building-enrichment-maps-and-auto-annotation"><i class="fa fa-check"></i>Exercise 3 (optional): Practice building enrichment maps and auto-annotation</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#optional-exercise-3a-autoannotate-the-enrichment-map"><i class="fa fa-check"></i>Optional exercise 3a: AutoAnnotate the enrichment map:</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#optional-exercise-3b-repeat-the-process-of-building-an-enrichment-map-using-the-proximal-data-proximal_gobp_greatexportall.tsv."><i class="fa fa-check"></i>Optional exercise 3b: Repeat the process of building an enrichment map using the proximal data (Proximal_GOBP_greatExportAll.tsv).</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#optional-exercise-3c-repeat-the-process-by-building-both-the-proximal-and-distal-enrichment-maps-at-the-same-time."><i class="fa fa-check"></i>Optional exercise 3c: Repeat the process by building both the Proximal and Distal enrichment maps at the same time.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-4-add-runx1-targets-and-runx1-ko-genes-on-the-distal-enrichment-map."><i class="fa fa-check"></i>Exercise 4: Add RUNX1 targets and RUNX1 KO genes on the distal enrichment map.</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#step-4a-post-analysis"><i class="fa fa-check"></i>step 4a: post analysis:</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#step-4b-optional-change-the-edge-style-of-the-signature-gene-sets"><i class="fa fa-check"></i>Step 4b Optional: Change the edge style of the signature gene-sets:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-5-learning-how-to-run-meme-chip-from-the-meme-suite-httpsmeme-suite.orgmemetoolsmeme-chip"><i class="fa fa-check"></i>Exercise 5: Learning how to run MEME-chip from the MEME suite (https://meme-suite.org/meme/tools/meme-chip)</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#format-the-data"><i class="fa fa-check"></i>Format the Data</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-5a-download-sequences-from-.bed-coordinates"><i class="fa fa-check"></i>Exercise 5a: Download sequences from .bed coordinates</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-5b-run-meme-chip"><i class="fa fa-check"></i>Exercise 5b: Run MEME-chIP</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#exercise-6-optional-get-the-iregulon-runx1-targets-and-find-the-mouse-orthologs-using-gorth-from-gprofiler-to-create-the-gmt-file-used-in-exercise-4."><i class="fa fa-check"></i>Exercise 6 (optional): Get the iRegulon RUNX1 targets and find the mouse orthologs using g:Orth (from g:Profiler) to create the gmt file used in Exercise 4.</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_chipseq_lab.html"><a href="regulatory_network_chipseq_lab.html#end-of-lab"><i class="fa fa-check"></i>End of Lab</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html"><i class="fa fa-check"></i>Module 5 Lab 2: Gene Regulation and Motif Analysis Practical Lab / iRegulon</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#iregulon-lab"><i class="fa fa-check"></i>iRegulon lab</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#goal"><i class="fa fa-check"></i>Goal</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#exercise-1.-detect-regulons-from-co-expressed-genes"><i class="fa fa-check"></i>Exercise 1. Detect regulons from co-expressed genes</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#skills-learned-in-this-exercise"><i class="fa fa-check"></i>Skills learned in this exercise:</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#steps"><i class="fa fa-check"></i>Steps</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#exercise-2.-create-a-metatargetome-using-iregulon-and-merge-2-networks-in-cytoscape."><i class="fa fa-check"></i>Exercise 2. Create a metatargetome using iRegulon and merge 2 networks in Cytoscape.</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#steps-1"><i class="fa fa-check"></i>Steps</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#use-or-precomputed-iregulon-results"><i class="fa fa-check"></i>Use or precomputed iRegulon results:</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#notes-about-iregulon"><i class="fa fa-check"></i>Notes about iRegulon:</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#notes-about-cytoscape"><i class="fa fa-check"></i>Notes about Cytoscape:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#exercise-3.-use-enrichr-with-the-prostate-gene-list."><i class="fa fa-check"></i>Exercise 3. Use Enrichr with the prostate gene list.</a>
<ul>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#goal-1"><i class="fa fa-check"></i>Goal</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#steps-2"><i class="fa fa-check"></i>Steps</a></li>
<li class="chapter" data-level="" data-path="regulatory_network_lab.html"><a href="regulatory_network_lab.html#end-of-practical-lab"><i class="fa fa-check"></i>end of practical lab</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-6-gene-function-prediction.html"><a href="module-6-gene-function-prediction.html"><i class="fa fa-check"></i>Module 6: Gene Function Prediction</a></li>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html"><i class="fa fa-check"></i>Module 6 Lab: GeneMANIA (Cytoscape version)</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#goal-of-this-practical-lab-2"><i class="fa fa-check"></i>Goal of this practical lab</a></li>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#exercise-1-questions-and-steps-to-follow"><i class="fa fa-check"></i>EXERCISE 1: QUESTIONS AND STEPS TO FOLLOW</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#steps-3"><i class="fa fa-check"></i>STEPS</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#exercise-1-answers"><i class="fa fa-check"></i>EXERCISE 1 ANSWERS</a></li>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#exercise-2-questions-and-steps-to-follow"><i class="fa fa-check"></i>EXERCISE 2: QUESTIONS AND STEPS TO FOLLOW</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#steps-4"><i class="fa fa-check"></i>STEPS</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#exercise-3-questions-and-steps-to-follow"><i class="fa fa-check"></i>EXERCISE 3: QUESTIONS AND STEPS TO FOLLOW</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#steps-5"><i class="fa fa-check"></i>STEPS</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_cytoscape.html"><a href="genemania_cytoscape.html#some-definitions"><i class="fa fa-check"></i>SOME DEFINITIONS:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html"><i class="fa fa-check"></i>Module 6 Lab: GeneMANIA (web version)</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#goal-of-this-practical-lab-3"><i class="fa fa-check"></i>Goal of this practical lab</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1-questions-and-steps-to-follow-1"><i class="fa fa-check"></i>EXERCISE 1: QUESTIONS AND STEPS TO FOLLOW</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#steps-6"><i class="fa fa-check"></i>STEPS</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1-answers-detailed-explanation-and-screenshots"><i class="fa fa-check"></i>EXERCISE 1 ANSWERS: DETAILED EXPLANATION AND SCREENSHOTS</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---steps-1-4"><i class="fa fa-check"></i>EXERCISE 1 - STEPS 1-4</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---step-5"><i class="fa fa-check"></i>EXERCISE 1 - STEP 5</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---step-6"><i class="fa fa-check"></i>EXERCISE 1 - STEP 6</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---step-7"><i class="fa fa-check"></i>Exercise 1 - STEP 7</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---step-8"><i class="fa fa-check"></i>Exercise 1 - STEP 8</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---step-9"><i class="fa fa-check"></i>Exercise 1 - STEP 9</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---step-10-layouts"><i class="fa fa-check"></i>Exercise 1 - STEP 10 (layouts)</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-1---step-11-save-an-image"><i class="fa fa-check"></i>Exercise 1 - STEP 11 (save an image)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2-questions-and-steps-to-follow-1"><i class="fa fa-check"></i>EXERCISE 2: QUESTIONS AND STEPS TO FOLLOW</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#steps-7"><i class="fa fa-check"></i>STEPS</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2-answers-detailed-steps-and-screenshots"><i class="fa fa-check"></i>EXERCISE 2 ANSWERS: DETAILED STEPS AND SCREENSHOTS</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---steps-1-to-4"><i class="fa fa-check"></i>Exercise 2 - STEPS 1 to 4</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-5"><i class="fa fa-check"></i>Exercise 2 - STEP 5</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-6."><i class="fa fa-check"></i>Exercise 2 - STEP 6.</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-7"><i class="fa fa-check"></i>Exercise 2 - STEP 7</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-8"><i class="fa fa-check"></i>Exercise 2 - STEP 8</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-9"><i class="fa fa-check"></i>Exercise 2 - STEP 9</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-10"><i class="fa fa-check"></i>Exercise 2 - STEP 10</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-11"><i class="fa fa-check"></i>Exercise 2 - STEP 11</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-12"><i class="fa fa-check"></i>Exercise 2 - STEP 12</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-2---step-13."><i class="fa fa-check"></i>Exercise 2 - STEP 13.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3-questions-and-steps-to-follow-1"><i class="fa fa-check"></i>EXERCISE 3: QUESTIONS AND STEPS TO FOLLOW</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#steps-8"><i class="fa fa-check"></i>STEPS</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3-more-details-and-screenshots"><i class="fa fa-check"></i>Exercise 3: MORE DETAILS AND SCREENSHOTS</a>
<ul>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3---steps-1---3"><i class="fa fa-check"></i>Exercise 3 - STEPS 1 - 3</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3---step-4"><i class="fa fa-check"></i>Exercise 3 - STEP 4</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3---step-5"><i class="fa fa-check"></i>Exercise 3 - STEP 5</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3---steps-6"><i class="fa fa-check"></i>Exercise 3 - STEPS 6</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3---step-7"><i class="fa fa-check"></i>Exercise 3 - STEP 7</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3---step-8"><i class="fa fa-check"></i>Exercise 3 - STEP 8</a></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#exercise-3---step-9"><i class="fa fa-check"></i>Exercise 3 - STEP 9</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genemania_web.html"><a href="genemania_web.html#some-definitions-1"><i class="fa fa-check"></i>SOME DEFINITIONS:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html"><i class="fa fa-check"></i>Integrated Assignment</a>
<ul>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#final-slides"><i class="fa fa-check"></i>Final slides</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#goal-2"><i class="fa fa-check"></i>Goal</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#dataset-1"><i class="fa fa-check"></i>DATASET 1</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#background-1"><i class="fa fa-check"></i>Background</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#data-processing"><i class="fa fa-check"></i>Data processing</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-1-run-gprofiler"><i class="fa fa-check"></i>PART 1: run g:Profiler</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-2-save-as-generic-enrichment-map-output-be"><i class="fa fa-check"></i>PART 2: save as Generic Enrichment Map output (BE)</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-3-save-as-generic-enrichment-map-output-ne"><i class="fa fa-check"></i>PART 3: save as Generic Enrichment Map output (NE)</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-4-create-an-enrichment-map"><i class="fa fa-check"></i>PART 4: create an enrichment map</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#answers-gprofiler"><i class="fa fa-check"></i>Answers g:Profiler</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-5-gsea-run-and-create-an-enrichment-map"><i class="fa fa-check"></i>PART 5: GSEA (run and create an enrichment map)</a>
<ul>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#answers-gsea"><i class="fa fa-check"></i>Answers GSEA</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-6-iregulon"><i class="fa fa-check"></i>PART 6: iRegulon</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#dataset-2"><i class="fa fa-check"></i>DATASET 2</a></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-1-reactomefi"><i class="fa fa-check"></i>PART 1: ReactomeFI</a>
<ul>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#answers-reactome-fi"><i class="fa fa-check"></i>Answers REACTOME FI</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#part-2-genemania"><i class="fa fa-check"></i>PART 2: GeneMANIA</a>
<ul>
<li class="chapter" data-level="" data-path="integrated_assignment.html"><a href="integrated_assignment.html#answers-genemania"><i class="fa fa-check"></i>Answers GeneMANIA</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="ass_automation.html"><a href="ass_automation.html"><i class="fa fa-check"></i>Integrated Assignment Bonus - Automation</a>
<ul>
<li class="chapter" data-level="" data-path="ass_automation.html"><a href="ass_automation.html#goal-of-the-exercise-5"><i class="fa fa-check"></i>Goal of the exercise:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><i class="fa fa-check"></i>Module 08 -1 lab - scRNA pathway analysis using g:Profiler {#scRNA gprofiler-lab}</a>
<ul>
<li class="chapter" data-level="" data-path="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#introduction-2"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#goal-of-the-exercise-1-1"><i class="fa fa-check"></i>Goal of the exercise 1</a></li>
<li class="chapter" data-level="" data-path="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#data-4"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#exercise-1"><i class="fa fa-check"></i>Exercise 1 - run g:Profiler</a>
<ul>
<li class="chapter" data-level="" data-path="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-1---launch-gprofiler.-1"><i class="fa fa-check"></i>Step 1 - Launch g:Profiler.</a></li>
<li class="chapter" data-level="" data-path="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-2---input-query-1"><i class="fa fa-check"></i>Step 2 - input query</a></li>
<li class="chapter" data-level="" data-path="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-3---adjust-parameters.-1"><i class="fa fa-check"></i>Step 3 - Adjust parameters.</a></li>
<li class="chapter" data-level="" data-path="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-4---run-query-1"><i class="fa fa-check"></i>Step 4 - Run query</a></li>
<li class="chapter" data-level="" data-path="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-5---explore-the-results.-1"><i class="fa fa-check"></i>Step 5 - Explore the results.</a></li>
<li class="chapter" data-level="" data-path="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-6-expand-the-stats-tab-1"><i class="fa fa-check"></i>Step 6: Expand the stats tab</a></li>
<li class="chapter" data-level="" data-path="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-7-save-the-results-1"><i class="fa fa-check"></i>Step 7: Save the results</a></li>
<li class="chapter" data-level="" data-path="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-8-optional-but-recommended-1"><i class="fa fa-check"></i>Step 8 (Optional but recommended)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#exercise-2"><i class="fa fa-check"></i>Exercise 2 - Cytoscape/EnrichmentMap</a></li>
<li class="chapter" data-level="" data-path="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#goal-of-the-exercise-6"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#data-5"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#enrichmentmap-2"><i class="fa fa-check"></i>EnrichmentMap</a></li>
<li class="chapter" data-level="" data-path="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#description-of-this-exercise-1"><i class="fa fa-check"></i>Description of this exercise</a></li>
<li class="chapter" data-level="" data-path="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#start-the-exercise-2"><i class="fa fa-check"></i>Start the exercise</a>
<ul>
<li class="chapter" data-level="" data-path="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-1-2"><i class="fa fa-check"></i>Step 1</a></li>
<li class="chapter" data-level="" data-path="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-2-2"><i class="fa fa-check"></i>Step 2</a></li>
<li class="chapter" data-level="" data-path="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-3-explore-detailed-results"><i class="fa fa-check"></i>Step 3: Explore Detailed results</a></li>
<li class="chapter" data-level="" data-path="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-4-optional-autoannotate-the-enrichment-map"><i class="fa fa-check"></i>Step 4 [OPTIONAL]: AutoAnnotate the enrichment map</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#exercise-3"><i class="fa fa-check"></i>Exercise 3 [OPTIONAL]- Cytoscape/EnrichmentMap using multiple datasets</a></li>
<li class="chapter" data-level="" data-path="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#goal-of-the-exercise-7"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#data-6"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#start-the-exercise-3"><i class="fa fa-check"></i>Start the exercise</a>
<ul>
<li class="chapter" data-level="" data-path="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-1-3"><i class="fa fa-check"></i>Step 1</a></li>
<li class="chapter" data-level="" data-path="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html"><a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-2-3"><i class="fa fa-check"></i>Step 2</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a  href=""><img src="./images/sponsors_2023.png" width="200"></a><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Pathway and Network Analysis of -Omics Data ( June 2023 )</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab" class="section level1 hasAnchor">
<h1>Module 08 -1 lab - scRNA pathway analysis using g:Profiler {#scRNA gprofiler-lab}<a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><strong>This work is licensed under a <a href="http://creativecommons.org/licenses/by-sa/3.0/deed.en_US">Creative Commons Attribution-ShareAlike 3.0 Unported License</a>. This means that you are able to copy, share and modify the work, as long as the result is distributed under the same license.</strong></p>
<div id="introduction-2" class="section level2 hasAnchor">
<h2>Introduction<a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#introduction-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Performing Over-Representation Analysis (ORA) with <a href="https://biit.cs.ut.ee/gprofiler/gost">g:Profiler</a>.
Visualizing the results using Cytoscape/Enrichmentmap.</p>
<p>The practical lab contains 2 exercises. The first exercise uses <a href="https://biit.cs.ut.ee/gprofiler/gost">g:Profiler</a> to perform gene-set enrichment analysis.</p>
</div>
<div id="goal-of-the-exercise-1-1" class="section level2 hasAnchor">
<h2>Goal of the exercise 1<a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#goal-of-the-exercise-1-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Learn how to run <em>g:GOSt Functional profiling</em> from the g:Profiler website and explore the results.</p>
</div>
<div id="data-4" class="section level2 hasAnchor">
<h2>Data<a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#data-4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>g:Profiler requires a list of genes, one per line, in a text file or spreadsheet,
ready to copy and paste into a web page: for this, we use genes identified in single cell RNA data from Module07. 14 cell clusters (0 to 14) were identified with the glioblastoma data we are working with (Richards et al, Nat Cancer, 2021).</p>
<p><img src="Can_Module8/images/scRNA1.png" alt="workflow" width="100%" /></p>
<p>The 14 clusters were further further classified into 5 cell types.</p>
<p><img src="Can_Module8/images/scRNA2.png" alt="workflow" width="100%" /></p>
<p>The gene lists were obtained from differential gene expression (DGE) analyses comparing cells from each cluster vs.the rest of the cells using Seurats function FindAllMarkers(, only.pos=T, min.pct = 0, return.thresh = 1, logfc.threshold = 0). For each cluster, the top 250 genes with FDR value equal or less than 0.05 were retrieved. All genes present in at least 1 cluster will be used as background (16066 genes) for the pathway enrichment analysis.</p>
<p><img src="Can_Module8/images/scRNA2a.png" alt="workflow" width="100%" /></p>
<p>DGE: Table (top genes of cluster 3 versus all clusters)
<img src="Can_Module8/images/scRNA2b.png" alt="workflow" width="100%" /></p>
<p>link to file: <a href="./Can_Module8/data/Richards_NatCancer_2021_DGE_GlobalClustering_SCT_wilcox.tsv.bz2">Richards_NatCancer_2021_DGE_GlobalClustering_SCT_wilcox.tsv.bz2</a></p>
</div>
<div id="exercise-1" class="section level2 hasAnchor">
<h2>Exercise 1 - run g:Profiler<a href="gprofiler-lab.html#exercise-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For this exercise, our goal is to copy and paste the list of genes into g:Profiler, adjust some parameters (e.g selecting the pathway databases), run the query and explore the results.</p>
<p>g:Profiler performs a gene-set enrichment analysis using a hypergeometric test (Fishers exact test). The Gene Ontology Biological Process, Reactome and Wiki pathways are going to be used as pathway databases. The results are displayed as a table or downloadable as an Generic Enrichment Map (GEM) output file.</p>
<p>Before starting this exercise, download the required files:</p>
<div class="rmd-datadownload">
<p>
Right click on link below and select Save Link As.
</p>
<p>
Place it in a folder on your computer : for example create a
pathway_analysis folder and save all the files needed for this module in
this directory.
</p>
</div>
<ul>
<li><a href="./Can_Module8/data/cluster3.txt">cluster3.txt</a></li>
<li><a href="./Can_Module8/data/background.txt">background.txt</a></li>
</ul>
<p>We recommend saving all these files in a personal project data folder before starting. We also recommend creating an additional result data folder to save the files generated while performing the protocol.</p>
<div id="step-1---launch-gprofiler.-1" class="section level3 hasAnchor">
<h3>Step 1 - Launch g:Profiler.<a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-1---launch-gprofiler.-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Open the g:Profiler website at <a href="https://biit.cs.ut.ee/gprofiler/gost">g:Profiler</a> in your web browser.</p>
</div>
<div id="step-2---input-query-1" class="section level3 hasAnchor">
<h3>Step 2 - input query<a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-2---input-query-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Paste the gene list (<a href="./Can_Module8/data/cluster3.txt">cluster3.txt</a>) into the Query field in top-left corner of the screen.</p>
<p><img src="Can_Module8/images/gp1.png" /></p>
<div class="rmd-note">
<p>
The gene list can be space-separated or one per line.<br> The
organism for the analysis, Homo sapiens, is selected by default.<br> The
input list can contain a mix of gene and protein IDs, symbols and
accession numbers.<br> Duplicated and unrecognized IDs will be removed
automatically, and ambiguous symbols can be refined in an interactive
dialogue after submitting the query.
</p>
</div>
<div class="rmd-tip">
<p>
open the file in a simple text editor such as Notepad or Textedit to
copy the list of genes. <br> Or right click on the file name above and
select <strong>Open link in new tab</strong>
</p>
</div>
</div>
<div id="step-3---adjust-parameters.-1" class="section level3 hasAnchor">
<h3>Step 3 - Adjust parameters.<a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-3---adjust-parameters.-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>3a. Click on the <em>Advanced options</em> tab (black rectangle) to expand it.</p>
<ul>
<li><p>Upload the custom background: Set <em>Statistical domain scope</em> to <em>Custom</em> and <em>Upload</em> the <a href="./Can_Module8/data/background.txt">background.txt</a> text file.</p></li>
<li><p>Set <em>Significance threshold</em> to Benjamini-Hochberg FDR</p></li>
<li><p><em>User threshold</em> - select 0.05 if you want g:Profiler to return only pathways that are significant (FDR &lt; 0.05).</p></li>
</ul>
<p align="center">
<img src="Can_Module8/images/gp2.png" alt="workflow" width="100%" />
</p>
<p>3b. Click on the <em>Data sources</em> tab (black rectangle) to expand it.</p>
<ul>
<li>UnSelect all gene-set databases by clicking the clear all button.</li>
<li>In the <em>Gene Ontology</em> category, check <em>GO Biological Process</em> and <em>No electronic GO annotations</em>.</li>
</ul>
<div class="rmd-note">
<p>
<em>No electronic GO annotations</em> option will discard less
reliable GO annotations (inferred from electronic annotations (IEAs))
that are not manually reviewed.
</p>
</div>
<div class="rmd-tip">
<p>
if g:Profiler does not return any results uncheck the <em>No
electronic GO annotation</em> option to expand annotations used in the
test.
</p>
</div>
<ul>
<li>In the <em>biological pathways</em> category, check <em>Reactome</em> and <em>WikiPathways</em>.</li>
</ul>
<p align="center">
<img src="Can_Module8/images/gp3.png" alt="workflow" width="100%" />
</p>
</div>
<div id="step-4---run-query-1" class="section level3 hasAnchor">
<h3>Step 4 - Run query<a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-4---run-query-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Click on the <em>Run query</em> button to run g:Profiler.</p>
<p>Scroll down page to see results.</p>
<div class="rmd-tip">
<p>
If after clicking on <em>Run query</em> button the analysis completes
but there is the following message above results file - <em>Select the
Ensembl ID with the most GO annotations (all)</em>. For each ambiguous
gene select its correct mapping. Ambiguous mapping is often caused by
multiple ensembl ids for a given gene and are easy to resolve as a user.
To choose the correct mapping, check the option that has the correct
gene name and/or then that has the most GO annotations. Rerun query.
</p>
</div>
</div>
<div id="step-5---explore-the-results.-1" class="section level3 hasAnchor">
<h3>Step 5 - Explore the results.<a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-5---explore-the-results.-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Step 5a:</p>
<ul>
<li>After the query has run, the results are displayed at the bottom of the page, below the input parameters.</li>
<li>By default, the Results tab is selected. A global graph displays gene-sets that passed the significance threshold of 0.05 for the 3 gene-set databases that we have selected, GO Biological Process(<a href="GO:BP" class="uri">GO:BP</a>) and Reactome(REAC) and Wikipathways(WP). Numbers in parentheses are indicating the number of gene-sets that passed the threshold (0 gene-sets passed the 0.05 threshold for Reactome).</li>
</ul>
<p><img src="Can_Module8/images/gp5.png" alt="workflow" width="100%" /></p>
<p>Step5b:</p>
<ul>
<li>Click on Detailed Results to view the results in more depth. Two tables are displayed, one for each of the data sources selected. (If more than 2 data sources are selected there will be additional tables for each datasource) Each row of the table contains:
<ul>
<li><strong>Term name</strong> - gene-set name</li>
<li><strong>Term ID</strong> - gene-est identifier</li>
<li><strong>Padj</strong> - FDR value</li>
<li><strong>-log10(Padj)</strong> - enrichment score calculated using the formula -log10(padj)</li>
<li>variable number of gene columns (One for each gene in the query set) - If the gene is present in the current gene-set its cell is colored. For any data source besides GO the cell is colored black if the gene is found in the gene-set. For the GO data source cells are colored according to the annotation evidence code. Expand the legend tab for detailed coloring mapping of GO evidence codes.</li>
</ul></li>
<li>Above the <a href="GO:BP" class="uri">GO:BP</a> result table, locate the slide bar that enables to select for the minimum and maximum number of genes in the tested gene-sets (Term size).</li>
<li>Change the maximum <em>Term size</em> from 10000 to <strong>500</strong> and change the minimum <em>Term size</em> to <strong>10</strong> and observe the results in the detailed stats panel:</li>
</ul>
<p><img src="Can_Module8/images/gp8a.png" alt="workflow" width="100%" /></p>
<ul>
<li>Without filtering term size, the top terms were GO terms containing that could contain 4000 or 5000 genes and that were located high in the GO hierarchy (parent term).
<ul>
<li>With filtering the maximum term size to 500, the top list contains pathways of larger interpretative values. However, please note that the adjusted pvalues was calculated using all gene-sets without size filtering.</li>
</ul></li>
</ul>
<p>The first table displays the gene-sets significantly enriched at FDR 0.05 for the <a href="GO:BP" class="uri">GO:BP</a> database.The second table displays the results corresponding to the Reactome database and the third table displays the results corresponding to the Wikipathways database.</p>
</div>
<div id="step-6-expand-the-stats-tab-1" class="section level3 hasAnchor">
<h3>Step 6: Expand the stats tab<a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-6-expand-the-stats-tab-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Expand the <em>stats</em> tab by clicking on the double arrow located at the right of the tab.</p>
<p align="center">
<img src="Can_Module8/images/gp7a.png" alt="workflow" width="75%" />
</p>
<p>It displays the gene set size (T), the size of our gene list (Q) , the number of genes that overlap between our gene list and the tested gene-set (TnQ) as well as the number of genes in the background (U).</p>
</div>
<div id="step-7-save-the-results-1" class="section level3 hasAnchor">
<h3>Step 7: Save the results<a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-7-save-the-results-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>7a. In the <em>Detailed Results</em> panel, select GEM . It will save the results in a text file in the Generic Enrichment Map format that we will use to visualize using Cytoscape.</p>
<ul>
<li>Click on the GEM button. A file is downloaded on your computer. (change the name to Cluster3.gem.txt)</li>
</ul>
<p>7b: Open the file that you saved using Microsoft Office Excel or in an equivalent software.</p>
<p>Observe the results included in this file:</p>
<ol style="list-style-type: decimal">
<li>Name of each gene-set</li>
<li>Description of each gene-set</li>
<li>significance of the overlap (pvalue)</li>
<li>significance of the overlap (adjusted pvalue/qvalue)</li>
<li>Phenotype</li>
<li>Genes included in each gene-set</li>
</ol>
<div class="rmd-question">
<p>
Which term has the best corrected p-value? <br> Which genes in our
list are included in this term? <br> Observe that same genes can be
present in several lines (pathways are related when they contain a lof
of genes in common).
</p>
</div>
<div class="rmd-note">
<p>
The table is formatted for the input into Cytoscape EnrichmentMap. It
is called the <a
href="https://enrichmentmap.readthedocs.io/en/latest/FileFormats.html#generic-results-files"><em>generic
format</em></a>. The p-value and FDR columns contain identical values
because g:Profiler directly outputs the FDR (= corrected p-value)
meaning that the p-value column is already the FDR. Phenotype 1 means
that each pathway will be represented by red nodes on the enrichment map
(presented during next module).
</p>
</div>
<p><img src="Can_Module8/images/gp15a.png" alt="workflow" width="100%" /></p>
<p>The terms <em>myelin</em> and <em>axon ensheathment</em> are the most significant gene-sets (=the lowest FDR value). Many gene-sets from the top of this list are related to each other and have genes in common.</p>
<p><img src="Can_Module8/images/gp15b.png" alt="workflow" width="50%" /></p>
<hr />
</div>
<div id="step-8-optional-but-recommended-1" class="section level3 hasAnchor">
<h3>Step 8 (Optional but recommended)<a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-8-optional-but-recommended-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>8a. Download the pathway database files.</p>
<ul>
<li>Go to the top of the page and expand the Data sources tab. Click on the combined name.gmt link located at bottom of this tab. It will download a file named <em>combined name.gmt</em> containing a pathway database gmt file with all the available sources.</li>
</ul>
<p align="center">
<img src="Can_Module8/images/gp12.png" alt="workflow" width="60%" />
</p>
<div class="rmd-note">
<p>
you will be using this optional gprofiler_full_hsapiens.name.gmt file
in Cytoscape EnrichmentMap.
</p>
</div>
<hr />
</div>
</div>
<div id="exercise-2" class="section level2 hasAnchor">
<h2>Exercise 2 - Cytoscape/EnrichmentMap<a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#exercise-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
</div>
<div id="goal-of-the-exercise-6" class="section level2 hasAnchor">
<h2>Goal of the exercise<a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#goal-of-the-exercise-6" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Create an enrichment map and navigate through the network</strong></p>
<p>During this exercise, you will learn how to create an enrichment map from gene-set enrichment results. The enrichment results chosen for this exercise are generated using g:Profiler but an enrichment map can be created directly from output from <a href="http://software.broadinstitute.org/gsea/index.jsp">GSEA</a>,
<a href="https://biit.cs.ut.ee/gprofiler/gost">g:Profiler</a>,
<a href="http://great.stanford.edu/public/html/">GREAT</a>,
<a href="http://apps.cytoscape.org/apps/bingo">BinGo</a>, <a href="https://amp.pharm.mssm.edu/Enrichr/">Enrichr</a> or alternately from any gene-set tool using the generic enrichment results format.</p>
</div>
<div id="data-5" class="section level2 hasAnchor">
<h2>Data<a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#data-5" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The data used in this exercise is pathway enrichment result from the list of genes that we found in cluster 3 in <a href="gprofiler-lab.html#exercise-1">exercise 1</a>.
Pathway enrichment analysis has been run using g:Profiler and the results have been downloaded as a GEM format.</p>
</div>
<div id="enrichmentmap-2" class="section level2 hasAnchor">
<h2>EnrichmentMap<a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#enrichmentmap-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li><p>A circle (node) is a gene-set (pathway) enriched in genes that we used as input in g:Profiler (frequently mutated genes).</p></li>
<li><p>edges (lines) represent genes in common between 2 pathways (nodes).</p></li>
<li><p>A cluster of nodes represent overlapping and related pathways and may represent a common biological process.</p></li>
<li><p>Clicking on a node will display the genes included in each pathway.</p></li>
</ul>
</div>
<div id="description-of-this-exercise-1" class="section level2 hasAnchor">
<h2>Description of this exercise<a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#description-of-this-exercise-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We run and saved g:Profiler result.
An enrichment map represents the result of enrichment analysis as a network where significantly enriched gene-sets that share a lot of genes in common will form identifiable clusters. The visualization of the results as these biological themes will ease the interpretation of the results.</p>
<p>The goal of this exercise is to learn how to:</p>
<ol style="list-style-type: decimal">
<li>upload g:Profiler results into Cytoscape EnrichmentMap to create a map.</li>
<li>learn how to navigate through Cytoscape EnrichmentMap and interpret the results.</li>
</ol>
</div>
<div id="start-the-exercise-2" class="section level2 hasAnchor">
<h2>Start the exercise<a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#start-the-exercise-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Two files are needed for this exercise:</p>
<ol style="list-style-type: decimal">
<li>Enrichment result: <a href="./Can_Module8/data/Cluster3_noEIA_gProfiler.gem.txt">Cluster3_noEIA_gProfiler.gem.txt</a></li>
</ol>
<ul>
<li>In g:Profiler, the parameters that we used to generate this file were:
<ul>
<li>GO_BP no electronic annotation,</li>
<li>Reactome,</li>
<li>Wikipathways</li>
<li>Benjamini-HochBerg FDR 0.05</li>
<li>gene-set size from 10 to 500
Note: this file is similar to the one that you have created in exercise 1. Use this link to follow exercise 2.</li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>Pathway database 1 (.gmt):<a href="./Can_Module8/data/gprofiler_full_hsapiens.name.gmt">gprofiler_full_hsapiens.name.gmt</a></li>
</ol></li>
</ul>
<div class="rmd-datadownload">
<p>
Right click on link below and select Save Link As.
</p>
<p>
Place it in the corresponding module directory of your
pathway_analysis folder on your computer.
</p>
</div>
<div id="step-1-2" class="section level3 hasAnchor">
<h3>Step 1<a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-1-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Launch Cytoscape and open the EnrichmentMap App</p>
<p>1a. Double click on Cytoscape icon</p>
<p>1b. Open EnrichmentMap App</p>
<ul>
<li><p>In the Cytoscape top menu bar:</p></li>
<li><p>Click on Apps -&gt; EnrichmentMap</p></li>
</ul>
<p align="center">
<img src="Can_Module8/images/EM1.png" alt="workflow" width="60%" />
</p>
<ul>
<li>A Create Enrichment Map window is now opened.</li>
</ul>
</div>
<div id="step-2-2" class="section level3 hasAnchor">
<h3>Step 2<a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-2-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Create an enrichment map from 1 dataset and with a gmt file.</p>
<p>2a. In the Create Enrichment Map window, drag and drop the enrichment file <a href="./Can_Module8/data/Cluster3_noEIA_gProfiler.gem.txt">Cluster3_noEIA_gProfiler.gem.txt</a>.
Tip: if drag and drop does not work, you can click  next to enrichments and upload the file. The analysis type needs to be set to generic/gprofiler.</p>
<p><img src="Can_Module8/images/gem0.png" alt="workflow" width="100%" /></p>
<p>2b. On the right side, go to the <em>GMT</em> field, click on the 3 radio button () and locate the file gprofiler_full_hsapiens.name.gmt that you have saved on your computer to upload it.</p>
<p>2c. Locate the <em>FDR q-value cutoff</em> field and set the value to 0.01</p>
<p>2d. Click on <em>Build</em>.</p>
<p><img src="Can_Module8/images/gem1.png" alt="workflow" width="100%" /></p>
<ul>
<li>a status bar should pop up showing progress of the Enrichment map build.</li>
</ul>
<p align="center">
<img src="Can_Module8/images/gp16.png" alt="workflow" width="60%"/>
</p>
</div>
<div id="step-3-explore-detailed-results" class="section level3 hasAnchor">
<h3>Step 3: Explore Detailed results<a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-3-explore-detailed-results" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>In the Cytoscape menu bar, select View and Show Graphic Details to display node labels.</li>
</ul>
<div class="rmd-caution">
<p>
Make sure you have unselected Publication Ready in the
EnrichmentMap control panel.
</p>
</div>
<ul>
<li><p>Zoom in to be able to read the labels and navigate the network using the bird eye view (blue rectangle).</p>
<ul>
<li>Select a node and visualize the <em>Table Panel</em>
<ul>
<li>Click on a node; Click on Dummy column. Genes with a green box are genes in the Cluster3 gene list and the selected pathway.</li>
</ul></li>
</ul></li>
</ul>
</div>
<div id="step-4-optional-autoannotate-the-enrichment-map" class="section level3 hasAnchor">
<h3>Step 4 [OPTIONAL]: AutoAnnotate the enrichment map<a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-4-optional-autoannotate-the-enrichment-map" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><p>move the the nodes and clusters apart of each other by selecting them and moving them around.</p></li>
<li><p>In the Cytoscape menu bar, select Apps &gt; AutoAnnotate &gt; New Annotation Set</p></li>
<li><p>An AutoAnnotate: Create Annotation Set window opens. In Advanced tab, check Create singleton clusters and click on Create Annotations.</p></li>
</ul>
<p><img src="Can_Module8/images/EM2.png" alt="workflow" width="60%"/></p>
<p>Tips for formatting:</p>
<ul>
<li><p>In the <em>AutoAnnotate Display</em> window located on the right side, uncheck <em>Scale font by cluster size</em> and check <em>Word Wrap</em>.</p>
<p><img src="Can_Module8/images/EM3.png" alt="workflow" width="60%"/></p></li>
</ul>
<p>Tip: if you are having difficulty separating nodes/clusters, you can hold shift and click and drag a square around a nodes of interest to highlight them, then move them all at once.</p>
<div class="rmd-caution">
<p>
SAVE YOUR CYTOSCAPE SESSION (.cys) FILE !
</p>
</div>
</div>
</div>
<div id="exercise-3" class="section level2 hasAnchor">
<h2>Exercise 3 [OPTIONAL]- Cytoscape/EnrichmentMap using multiple datasets<a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#exercise-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
</div>
<div id="goal-of-the-exercise-7" class="section level2 hasAnchor">
<h2>Goal of the exercise<a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#goal-of-the-exercise-7" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Create an enrichment map and navigate through the network</strong></p>
<p>During this exercise, you will learn how to create an enrichment map from multiple gene-set enrichment results generated using g:Profiler.</p>
</div>
<div id="data-6" class="section level2 hasAnchor">
<h2>Data<a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#data-6" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li><p>The data used in this exercise is the enrichment results from the list of genes of clusters that we found in clusters 0, 1, 3, 4, 5, 7, and 10 from the single cell RNAseq data.</p></li>
<li><p>Pathway enrichment analysis has been run using g:Profiler and the results have been downloaded as a GEM format.</p></li>
<li><p>The gene lists were obtained from differential gene expression analyses comparing cells from each cluster vs.the rest of the cells using Seurats function FindAllMarkers(, only.pos=T, min.pct = 0, return.thresh = 1, logfc.threshold = 0). For each cluster, the top 250 genes with FDR value equal or less than 0.05 were retrieved.</p></li>
<li><p>In g:Profiler, the parameters that we used to generate this file were:</p></li>
<li><p>GO_BP no electronic annotation,</p></li>
<li><p>Reactome,</p></li>
<li><p>Wikipathways</p></li>
<li><p>Benjamini-HochBerg FDR 0.05</p></li>
<li><p>gene-set size from 10 to 500</p></li>
<li><p>Top 50 pathways were selected for further analysis.</p></li>
</ul>
</div>
<div id="start-the-exercise-3" class="section level2 hasAnchor">
<h2>Start the exercise<a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#start-the-exercise-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Download the files needed for this exercise on your computer:</p>
<ul>
<li><a href="./Can_Module8/data/Cluster0_gProfiler50.gem.txt">Cluster0_gProfiler50.gem.txt</a></li>
<li><a href="./Can_Module8/data/Cluster1_gProfiler50.gem.txt">Cluster1_gProfiler50.gem.txt</a></li>
<li><a href="./Can_Module8/data/Cluster3_gProfiler50.gem.txt">Cluster3_gProfiler50.gem.txt</a></li>
<li><a href="./Can_Module8/data/Cluster4_gProfiler50.gem.txt">Cluster4_gProfiler50.gem.txt</a></li>
<li><a href="./Can_Module8/data/Cluster5_gProfiler50.gem.txt">Cluster5_gProfiler50.gem.txt</a></li>
<li><a href="./Can_Module8/data/Cluster7_gProfiler50.gem.txt">Cluster7_gProfiler50.gem.txt</a></li>
<li><a href="./Can_Module8/data/Cluster10_gProfiler50.gem.txt">Cluster10_gProfiler50.gem.txt</a></li>
</ul>
<p>Launch Cytoscape and open the EnrichmentMap App</p>
<div id="step-1-3" class="section level3 hasAnchor">
<h3>Step 1<a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-1-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>1a. Double click on Cytoscape icon</p>
<p>1b. Open EnrichmentMap App</p>
<ul>
<li><p>In the Cytoscape top menu bar:</p></li>
<li><p>Click on Apps -&gt; EnrichmentMap</p></li>
</ul>
<p align="center">
<img src="Can_Module8/images/EM1.png" alt="workflow" width="60%" />
</p>
<ul>
<li>A Create Enrichment Map window is now opened.</li>
</ul>
</div>
<div id="step-2-3" class="section level3 hasAnchor">
<h3>Step 2<a href="module-08--1-lab---scrna-pathway-analysis-using-gprofiler-scrna-gprofiler-lab.html#step-2-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Create an enrichment map from multiple datasets.</p>
<p>2a. In the Create Enrichment Map window, drag and drop the enrichment files</p>
<ul>
<li><a href="./Can_Module8/data/Cluster0_gProfiler50.gem.txt">Cluster0_gProfiler50.gem.txt</a></li>
<li><a href="./Can_Module8/data/Cluster1_gProfiler50.gem.txt">Cluster1_gProfiler50.gem.txt</a></li>
<li><a href="./Can_Module8/data/Cluster3_gProfiler50.gem.txt">Cluster3_gProfiler50.gem.txt</a></li>
<li><a href="./Can_Module8/data/Cluster4_gProfiler50.gem.txt">Cluster4_gProfiler50.gem.txt</a></li>
<li><a href="./Can_Module8/data/Cluster5_gProfiler50.gem.txt">Cluster5_gProfiler50.gem.txt</a></li>
<li><a href="./Can_Module8/data/Cluster7_gProfiler50.gem.txt">Cluster7_gProfiler50.gem.txt</a></li>
<li><a href="./Can_Module8/data/Cluster10_gProfiler50.gem.txt">Cluster10_gProfiler50.gem.txt</a></li>
</ul>
<p>2b. Locate the <em>FDR q-value cutoff</em> field and set the value to 0.05</p>
<p>2c. Click on <em>Build</em>.</p>
<p><img src="Can_Module8/images/em2_1.png" alt="workflow" width="100%" /></p>
<ul>
<li>a status bar should pop up showing progress of the Enrichment map build.</li>
</ul>
<p>2d. Once the map is build, locate the EnrichmentMap tab on the right and set <em>Chart Data</em> to <em>Color by Data Set</em>. You can also check publication ready to remove node labels.</p>
<p align="center">
<img src="Can_Module8/images/em2_2.png" alt="workflow" width="100%" />
</p>
<p>2e. Change the color of each data set so it corresponds to the single cell RNAseq UPMAP plot</p>
<ul>
<li>Locate the EnrichmentMap tab on the right and click on <em>Change colors</em></li>
</ul>
<p align="center">
<img src="Can_Module8/images/em2_4.png" alt="workflow" width="100%" />
</p>
<ul>
<li>Adjust the colors so it corresponds approximately to the single cell RNAseq UMAP plot (see top of the document for reference).</li>
</ul>
<p align="center">
<img src="Can_Module8/images/em2_5.png" alt="workflow" width="50%" />
</p>
<ul>
<li>View of the map:</li>
</ul>
<p align="center">
<img src="Can_Module8/images/em2_6.png" alt="workflow" width="100%" />
</p>
<p>On this view, the nodes of the same colors were grouped by selecting and moving the nodes and clusters of interest.</p>
<p>Tip: To finalize the map, it could be further auto-annotated (see step execise 2). If there are too many annotated clusters, the clusters can be merged (select the nodes of interest, right click on a blanc space, select Apps &gt; AutoAnnotate - Create Cluster).</p>
<p align="center">
<img src="Can_Module8/images/em2_7.png" alt="workflow" width="100%" />
</p>
Legend:
<p align="center">
<img src="Can_Module8/images/em2_8.png" alt="workflow" width="40%" />
</p>
<div id="section-2" class="section level60">
<p class="heading"></p>
<p>Cytoscape file:</p>
<ul>
<li><a href="./Can_Module8/data/scRNAgprofiler.cys">scRNAgprofiler.cys</a></li>
</ul>

<div id="refs" class="references csl-bib-body hanging-indent">
<div class="csl-entry">
Merkel, Dirk. 2014. <span>Docker: Lightweight Linux Containers for Consistent Development and Deployment.</span> <em>Linux Journal</em> 2014 (239): 2.
</div>
</div>
</div>
</div>
</div>
</div>






            </section>

          </div>
        </div>
      </div>
<a href="ass_automation.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": "https://github.com/BaderLab/CBW_Pathways_2023/commit/master/8.1-Can_Module8lab_gprofiler.Rmd",
"text": null
},
"view": {
"link": "https://github.com/BaderLab/CBW_Pathways_2023/master/8.1-Can_Module8lab_gprofiler.Rmd",
"text": null
},
"download": ["CBW_Pathways.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true
}
});
});
</script>

</body>

</html>
